<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.6.39">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>3&nbsp; Skin: Check patient sexes – Blass &amp; Keskin (2025): Melanoma NeoVax analysis: Skin single-cell RNAseq</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./Skin_scRNA_Part4.html" rel="next">
<link href="./Skin_scRNA_Part2.html" rel="prev">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-e26003cea8cd680ca0c55a263523d882.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap-41a4eb7d26fa9333a8f88b12b94df59e.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


</head>

<body class="nav-sidebar floating">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./Skin_scRNA_Part3.html"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Skin: Check patient sexes</span></a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="./">Blass &amp; Keskin (2025): Melanoma NeoVax analysis: Skin single-cell RNAseq</a> 
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Introduction</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Skin_scRNA_Part1.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Skin: QC &amp; Filtering</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Skin_scRNA_Part2.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Skin: Normalization, doublet discrimination, integration, clustering</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Skin_scRNA_Part3.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Skin: Check patient sexes</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Skin_scRNA_Part4.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Skin: Post-clustering QC and characterization</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Skin_scRNA_Part5.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">Skin: Post-clustering comprehensive marker identification</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Skin_scRNA_Part6.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">Skin: Sub-clustering on T/NK lineage</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Skin_scRNA_Part7.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">Skin: Assisting cell type annotation with enrichR</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Skin_scRNA_Part8.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">8</span>&nbsp; <span class="chapter-title">Skin: Classification of CD4 and CD8 T cell states within T/NK lineage</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Skin_scRNA_Part9.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">9</span>&nbsp; <span class="chapter-title">Skin: Differential expression analysis, Post vs Pre 3rd vaccination</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Skin_scRNA_Part10.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">10</span>&nbsp; <span class="chapter-title">Skin: Deep phenotyping, cluster annotation, and exploration of Post vs Pre 3rd vaccination differences in cell proportions and expression</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Skin_scRNA_Part11.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">11</span>&nbsp; <span class="chapter-title">Skin: Calculating Post vs Pre 3rd vaccination differences in pathway enrichments with fgsea</span></span></a>
  </div>
</li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#set-up-seurat-workspace" id="toc-set-up-seurat-workspace" class="nav-link active" data-scroll-target="#set-up-seurat-workspace"><span class="header-section-number">3.1</span> Set up Seurat workspace</a></li>
  <li><a href="#load-previously-saved-clustered-object" id="toc-load-previously-saved-clustered-object" class="nav-link" data-scroll-target="#load-previously-saved-clustered-object"><span class="header-section-number">3.2</span> Load previously saved clustered object</a></li>
  <li><a href="#infer-sex-per-cell-using-chrx-and-chry-gene-counts-using-cellxy" id="toc-infer-sex-per-cell-using-chrx-and-chry-gene-counts-using-cellxy" class="nav-link" data-scroll-target="#infer-sex-per-cell-using-chrx-and-chry-gene-counts-using-cellxy"><span class="header-section-number">3.3</span> Infer sex per cell using chrX and chrY gene counts using <code>cellXY</code></a></li>
  <li><a href="#assign-known-sexes-to-each-sample" id="toc-assign-known-sexes-to-each-sample" class="nav-link" data-scroll-target="#assign-known-sexes-to-each-sample"><span class="header-section-number">3.4</span> Assign known sexes to each sample</a></li>
  <li><a href="#summarize-cell-wise-sex-predictions-per-sample-and-compare-to-known-labels" id="toc-summarize-cell-wise-sex-predictions-per-sample-and-compare-to-known-labels" class="nav-link" data-scroll-target="#summarize-cell-wise-sex-predictions-per-sample-and-compare-to-known-labels"><span class="header-section-number">3.5</span> Summarize cell-wise sex predictions per Sample and compare to known labels</a></li>
  <li><a href="#since-everything-matches-we-now-add-sex-as-a-metadata-label" id="toc-since-everything-matches-we-now-add-sex-as-a-metadata-label" class="nav-link" data-scroll-target="#since-everything-matches-we-now-add-sex-as-a-metadata-label"><span class="header-section-number">3.6</span> Since everything matches we now add Sex as a metadata label</a></li>
  <li><a href="#plot-umap-of-sex" id="toc-plot-umap-of-sex" class="nav-link" data-scroll-target="#plot-umap-of-sex"><span class="header-section-number">3.7</span> Plot UMAP of sex</a></li>
  <li><a href="#save-the-updated-object" id="toc-save-the-updated-object" class="nav-link" data-scroll-target="#save-the-updated-object"><span class="header-section-number">3.8</span> Save the updated object</a></li>
  <li><a href="#get-session-info" id="toc-get-session-info" class="nav-link" data-scroll-target="#get-session-info"><span class="header-section-number">3.9</span> Get session info</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title"><span id="sec-skin-patient-sex" class="quarto-section-identifier"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Skin: Check patient sexes</span></span></h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<section id="set-up-seurat-workspace" class="level2" data-number="3.1">
<h2 data-number="3.1" class="anchored" data-anchor-id="set-up-seurat-workspace"><span class="header-section-number">3.1</span> Set up Seurat workspace</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load libraries</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(data.table)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(devtools)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(presto)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(glmGamPoi)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(sctransform)</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Seurat) </span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(miQC)   </span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(SeuratWrappers)</span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(flexmix)</span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(SingleCellExperiment)</span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(SummarizedExperiment)</span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readxl)</span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(fishpond)</span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Matrix)</span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(speckle)</span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(scater)</span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(vctrs)</span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(alevinQC)</span>
<span id="cb1-22"><a href="#cb1-22" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(harmony)</span>
<span id="cb1-23"><a href="#cb1-23" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(scDblFinder)</span>
<span id="cb1-24"><a href="#cb1-24" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(cellXY)</span>
<span id="cb1-25"><a href="#cb1-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-26"><a href="#cb1-26" aria-hidden="true" tabindex="-1"></a><span class="co"># Set global options for Seurat v5 objects</span></span>
<span id="cb1-27"><a href="#cb1-27" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">Seurat.object.assay.version =</span> <span class="st">'v5'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="load-previously-saved-clustered-object" class="level2" data-number="3.2">
<h2 data-number="3.2" class="anchored" data-anchor-id="load-previously-saved-clustered-object"><span class="header-section-number">3.2</span> Load previously saved clustered object</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>merged.<span class="fl">18279.</span>skin.singlets <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"Skin_scRNA_Part2.rds"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="infer-sex-per-cell-using-chrx-and-chry-gene-counts-using-cellxy" class="level2" data-number="3.3">
<h2 data-number="3.3" class="anchored" data-anchor-id="infer-sex-per-cell-using-chrx-and-chry-gene-counts-using-cellxy"><span class="header-section-number">3.3</span> Infer sex per cell using chrX and chrY gene counts using <code>cellXY</code></h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>merged.<span class="fl">18279.</span>skin.singlets[[<span class="st">'RNA'</span>]] <span class="ot">&lt;-</span> <span class="fu">JoinLayers</span>(merged.<span class="fl">18279.</span>skin.singlets[[<span class="st">'RNA'</span>]])</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>xyPredict <span class="ot">&lt;-</span> <span class="fu">classifySex</span>(<span class="at">x=</span>merged.<span class="fl">18279.</span>skin.singlets<span class="sc">@</span>assays<span class="sc">$</span>RNA<span class="sc">$</span>counts,<span class="at">genome=</span><span class="st">"Hs"</span>, <span class="at">qc=</span>F)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in asMethod(object): sparse-&gt;dense coercion: allocating vector of size
15.2 GiB</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>28cell/s are unable to be classified
              due to an abundance of zeroes on X and Y chromosome genes</code></pre>
</div>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>merged.<span class="fl">18279.</span>skin.singlets <span class="ot">&lt;-</span> <span class="fu">AddMetaData</span>(merged.<span class="fl">18279.</span>skin.singlets,xyPredict<span class="sc">$</span>prediction,<span class="at">col.name=</span><span class="st">"CellPredictedSex"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="assign-known-sexes-to-each-sample" class="level2" data-number="3.4">
<h2 data-number="3.4" class="anchored" data-anchor-id="assign-known-sexes-to-each-sample"><span class="header-section-number">3.4</span> Assign known sexes to each sample</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>knownSex <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">cbind</span>(<span class="st">"Sample"</span> <span class="ot">=</span> <span class="fu">unique</span>(merged.<span class="fl">18279.</span>skin.singlets<span class="sc">$</span>Sample), <span class="st">"Sex"</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="fu">rep</span>(<span class="st">"Male"</span>,<span class="dv">2</span>),<span class="fu">rep</span>(<span class="st">"Male"</span>,<span class="dv">2</span>),<span class="fu">rep</span>(<span class="st">"Female"</span>,<span class="dv">2</span>),<span class="fu">rep</span>(<span class="st">"Male"</span>,<span class="dv">2</span>),<span class="fu">rep</span>(<span class="st">"Female"</span>,<span class="dv">2</span>),<span class="fu">rep</span>(<span class="st">"Male"</span>,<span class="dv">2</span>),<span class="fu">rep</span>(<span class="st">"Female"</span>,<span class="dv">1</span>),<span class="fu">rep</span>(<span class="st">"Male"</span>,<span class="dv">2</span>),<span class="fu">rep</span>(<span class="st">"Male"</span>,<span class="dv">2</span>))))</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>knownSex</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                           Sample    Sex
1   P101_Skin_Pre3rd_2.5mgIpi_RNA   Male
2  P101_Skin_Post3rd_2.5mgIpi_RNA   Male
3   P103_Skin_Pre3rd_2.5mgIpi_RNA   Male
4  P103_Skin_Post3rd_2.5mgIpi_RNA   Male
5  P104_Skin_Post3rd_2.5mgIpi_RNA Female
6   P104_Skin_Pre3rd_2.5mgIpi_RNA Female
7  P105_Skin_Post3rd_2.5mgIpi_RNA   Male
8   P105_Skin_Pre3rd_2.5mgIpi_RNA   Male
9   P106_Skin_Pre3rd_2.5mgIpi_RNA Female
10 P106_Skin_Post3rd_2.5mgIpi_RNA Female
11    P108_Skin_Pre3rd_5mgIpi_RNA   Male
12   P108_Skin_Post3rd_5mgIpi_RNA   Male
13    P109_Skin_Pre3rd_5mgIpi_RNA Female
14    P110_Skin_Pre3rd_5mgIpi_RNA   Male
15   P110_Skin_Post3rd_5mgIpi_RNA   Male
16    P111_Skin_Pre3rd_5mgIpi_RNA   Male
17   P111_Skin_Post3rd_5mgIpi_RNA   Male</code></pre>
</div>
</div>
</section>
<section id="summarize-cell-wise-sex-predictions-per-sample-and-compare-to-known-labels" class="level2" data-number="3.5">
<h2 data-number="3.5" class="anchored" data-anchor-id="summarize-cell-wise-sex-predictions-per-sample-and-compare-to-known-labels"><span class="header-section-number">3.5</span> Summarize cell-wise sex predictions per Sample and compare to known labels</h2>
<p>If more than 80% of the individual cell sex predictions are consistent for a given Sample, we call that Sample as that sex, then match to known labels</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Show snippet first</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames_to_column</span>(merged.<span class="fl">18279.</span>skin.singlets<span class="sc">@</span>meta.data,<span class="at">var=</span><span class="st">"bc"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(bc,CellPredictedSex,Sample) <span class="sc">%&gt;%</span></span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(Sample) <span class="sc">%&gt;%</span></span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarize</span>(<span class="at">nMale =</span> <span class="fu">sum</span>(CellPredictedSex<span class="sc">==</span><span class="st">"Male"</span>), </span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a>        <span class="at">nFemale =</span> <span class="fu">sum</span>(CellPredictedSex<span class="sc">==</span><span class="st">"Female"</span>), </span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a>        <span class="at">nCells =</span> <span class="fu">n</span>()</span>
<span id="cb9-9"><a href="#cb9-9" aria-hidden="true" tabindex="-1"></a>    )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 17 × 4
   Sample                         nMale nFemale nCells
   &lt;chr&gt;                          &lt;int&gt;   &lt;int&gt;  &lt;int&gt;
 1 P101_Skin_Post3rd_2.5mgIpi_RNA  1655      62   1718
 2 P101_Skin_Pre3rd_2.5mgIpi_RNA    417       9    426
 3 P103_Skin_Post3rd_2.5mgIpi_RNA  3436     278   3726
 4 P103_Skin_Pre3rd_2.5mgIpi_RNA    782      36    818
 5 P104_Skin_Post3rd_2.5mgIpi_RNA   957    4465   5434
 6 P104_Skin_Pre3rd_2.5mgIpi_RNA    136    2956   3092
 7 P105_Skin_Post3rd_2.5mgIpi_RNA  3213     215   3429
 8 P105_Skin_Pre3rd_2.5mgIpi_RNA   2387     139   2526
 9 P106_Skin_Post3rd_2.5mgIpi_RNA   249    1405   1655
10 P106_Skin_Pre3rd_2.5mgIpi_RNA     45    1076   1121
11 P108_Skin_Post3rd_5mgIpi_RNA    2589     100   2689
12 P108_Skin_Pre3rd_5mgIpi_RNA      906      31    937
13 P109_Skin_Pre3rd_5mgIpi_RNA       44    1331   1375
14 P110_Skin_Post3rd_5mgIpi_RNA    1408      72   1481
15 P110_Skin_Pre3rd_5mgIpi_RNA     1160      31   1191
16 P111_Skin_Post3rd_5mgIpi_RNA    1270      72   1342
17 P111_Skin_Pre3rd_5mgIpi_RNA      406      31    437</code></pre>
</div>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Now determine sex per sample and count matching vs non-matching labels</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames_to_column</span>(merged.<span class="fl">18279.</span>skin.singlets<span class="sc">@</span>meta.data,<span class="at">var=</span><span class="st">"bc"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(bc,CellPredictedSex,Sample) <span class="sc">%&gt;%</span></span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(Sample) <span class="sc">%&gt;%</span></span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarize</span>(<span class="at">nMale =</span> <span class="fu">sum</span>(CellPredictedSex<span class="sc">==</span><span class="st">"Male"</span>), </span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a>        <span class="at">nFemale =</span> <span class="fu">sum</span>(CellPredictedSex<span class="sc">==</span><span class="st">"Female"</span>), </span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a>        <span class="at">nCells =</span> <span class="fu">n</span>()</span>
<span id="cb11-9"><a href="#cb11-9" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb11-10"><a href="#cb11-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb11-11"><a href="#cb11-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">PredictedSex =</span> <span class="fu">case_when</span>(</span>
<span id="cb11-12"><a href="#cb11-12" aria-hidden="true" tabindex="-1"></a>        nMale <span class="sc">/</span> nCells <span class="sc">&gt;</span> <span class="fl">0.8</span> <span class="sc">~</span> <span class="st">"Male"</span>,</span>
<span id="cb11-13"><a href="#cb11-13" aria-hidden="true" tabindex="-1"></a>        nFemale <span class="sc">/</span> nCells <span class="sc">&gt;</span> <span class="fl">0.8</span> <span class="sc">~</span> <span class="st">"Female"</span>,</span>
<span id="cb11-14"><a href="#cb11-14" aria-hidden="true" tabindex="-1"></a>        T <span class="sc">~</span> <span class="st">"other"</span></span>
<span id="cb11-15"><a href="#cb11-15" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb11-16"><a href="#cb11-16" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb11-17"><a href="#cb11-17" aria-hidden="true" tabindex="-1"></a>    <span class="fu">inner_join</span>(knownSex,<span class="at">by=</span><span class="st">"Sample"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb11-18"><a href="#cb11-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarize</span>(<span class="at">nEqual =</span> <span class="fu">sum</span>(PredictedSex<span class="sc">==</span>Sex), <span class="at">nNotEqual =</span> <span class="fu">sum</span>(PredictedSex <span class="sc">!=</span> Sex))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 1 × 2
  nEqual nNotEqual
   &lt;int&gt;     &lt;int&gt;
1     17         0</code></pre>
</div>
</div>
</section>
<section id="since-everything-matches-we-now-add-sex-as-a-metadata-label" class="level2" data-number="3.6">
<h2 data-number="3.6" class="anchored" data-anchor-id="since-everything-matches-we-now-add-sex-as-a-metadata-label"><span class="header-section-number">3.6</span> Since everything matches we now add Sex as a metadata label</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>knownSexByCell <span class="ot">&lt;-</span> <span class="fu">enframe</span>(merged.<span class="fl">18279.</span>skin.singlets<span class="sc">$</span>Sample,<span class="at">name=</span><span class="st">"bc"</span>,<span class="at">value=</span><span class="st">"Sample"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">inner_join</span>(knownSex,<span class="at">by=</span><span class="st">"Sample"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pull</span>(Sex)</span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a>merged.<span class="fl">18279.</span>skin.singlets <span class="ot">&lt;-</span> <span class="fu">AddMetaData</span>(merged.<span class="fl">18279.</span>skin.singlets,knownSexByCell,<span class="at">col.name=</span><span class="st">"Sex"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="plot-umap-of-sex" class="level2" data-number="3.7">
<h2 data-number="3.7" class="anchored" data-anchor-id="plot-umap-of-sex"><span class="header-section-number">3.7</span> Plot UMAP of sex</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(merged.<span class="fl">18279.</span>skin.singlets,<span class="at">reduction =</span> <span class="st">"umap.harmony"</span>, <span class="at">group.by =</span> <span class="st">"Sex"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Skin_scRNA_Part3_files/figure-html/unnamed-chunk-7-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="save-the-updated-object" class="level2" data-number="3.8">
<h2 data-number="3.8" class="anchored" data-anchor-id="save-the-updated-object"><span class="header-section-number">3.8</span> Save the updated object</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(merged.<span class="fl">18279.</span>skin.singlets,<span class="at">file=</span><span class="st">"Skin_scRNA_Part3.rds"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="get-session-info" class="level2" data-number="3.9">
<h2 data-number="3.9" class="anchored" data-anchor-id="get-session-info"><span class="header-section-number">3.9</span> Get session info</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>R version 4.3.1 (2023-06-16)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Rocky Linux 8.10 (Green Obsidian)

Matrix products: default
BLAS/LAPACK: /usr/lib64/libopenblasp-r0.3.15.so;  LAPACK version 3.9.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

time zone: America/New_York
tzcode source: system (glibc)

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] cellXY_0.99.0               scDblFinder_1.14.0         
 [3] harmony_1.2.0               alevinQC_1.16.1            
 [5] vctrs_0.6.5                 patchwork_1.3.0            
 [7] scater_1.28.0               scuttle_1.10.3             
 [9] speckle_1.0.0               Matrix_1.6-4               
[11] fishpond_2.6.2              readxl_1.4.3               
[13] SingleCellExperiment_1.22.0 SummarizedExperiment_1.30.2
[15] Biobase_2.60.0              GenomicRanges_1.52.1       
[17] GenomeInfoDb_1.36.4         IRanges_2.34.1             
[19] S4Vectors_0.38.2            BiocGenerics_0.46.0        
[21] MatrixGenerics_1.12.3       matrixStats_1.2.0          
[23] flexmix_2.3-19              lattice_0.22-5             
[25] SeuratWrappers_0.3.19       miQC_1.8.0                 
[27] lubridate_1.9.3             forcats_1.0.0              
[29] stringr_1.5.1               dplyr_1.1.4                
[31] purrr_1.0.2                 readr_2.1.5                
[33] tidyr_1.3.1                 tibble_3.2.1               
[35] ggplot2_3.4.4               tidyverse_2.0.0            
[37] Seurat_5.1.0                SeuratObject_5.0.2         
[39] sp_2.1-3                    sctransform_0.4.1          
[41] glmGamPoi_1.12.2            presto_1.0.0               
[43] Rcpp_1.0.12                 devtools_2.4.5             
[45] usethis_2.2.2               data.table_1.15.0          

loaded via a namespace (and not attached):
  [1] fs_1.6.3                  spatstat.sparse_3.0-3    
  [3] bitops_1.0-7              httr_1.4.7               
  [5] RColorBrewer_1.1-3        profvis_0.3.8            
  [7] tools_4.3.1               utf8_1.2.4               
  [9] R6_2.5.1                  DT_0.31                  
 [11] lazyeval_0.2.2            uwot_0.1.16              
 [13] urlchecker_1.0.1          withr_3.0.0              
 [15] GGally_2.2.1              gridExtra_2.3            
 [17] progressr_0.14.0          cli_3.6.2                
 [19] spatstat.explore_3.2-6    fastDummies_1.7.3        
 [21] labeling_0.4.3            spatstat.data_3.0-4      
 [23] ggridges_0.5.6            pbapply_1.7-2            
 [25] Rsamtools_2.16.0          R.utils_2.12.3           
 [27] parallelly_1.37.0         sessioninfo_1.2.2        
 [29] limma_3.56.2              RSQLite_2.3.5            
 [31] BiocIO_1.10.0             generics_0.1.3           
 [33] gtools_3.9.5              ica_1.0-3                
 [35] spatstat.random_3.2-2     ggbeeswarm_0.7.2         
 [37] fansi_1.0.6               abind_1.4-5              
 [39] R.methodsS3_1.8.2         lifecycle_1.0.4          
 [41] yaml_2.3.8                edgeR_3.42.4             
 [43] recipes_1.0.10            Rtsne_0.17               
 [45] blob_1.2.4                grid_4.3.1               
 [47] dqrng_0.3.2               promises_1.2.1           
 [49] crayon_1.5.2              shinydashboard_0.7.2     
 [51] miniUI_0.1.1.1            beachmat_2.16.0          
 [53] cowplot_1.1.3             KEGGREST_1.40.1          
 [55] metapod_1.8.0             pillar_1.9.0             
 [57] knitr_1.45                rjson_0.2.21             
 [59] xgboost_1.7.7.1           future.apply_1.11.1      
 [61] codetools_0.2-19          leiden_0.4.3.1           
 [63] glue_1.7.0                remotes_2.4.2.1          
 [65] png_0.1-8                 spam_2.10-0              
 [67] org.Mm.eg.db_3.18.0       cellranger_1.1.0         
 [69] gtable_0.3.4              cachem_1.0.8             
 [71] gower_1.0.1               xfun_0.42                
 [73] prodlim_2023.08.28        S4Arrays_1.2.0           
 [75] mime_0.12                 survival_3.5-8           
 [77] timeDate_4032.109         iterators_1.0.14         
 [79] hardhat_1.3.1             lava_1.7.3               
 [81] statmod_1.5.0             bluster_1.10.0           
 [83] ellipsis_0.3.2            fitdistrplus_1.1-11      
 [85] ipred_0.9-14              ROCR_1.0-11              
 [87] nlme_3.1-164              bit64_4.0.5              
 [89] RcppAnnoy_0.0.22          irlba_2.3.5.1            
 [91] rpart_4.1.23              vipor_0.4.7              
 [93] KernSmooth_2.23-22        DBI_1.2.2                
 [95] colorspace_2.1-0          nnet_7.3-19              
 [97] tidyselect_1.2.0          bit_4.0.5                
 [99] compiler_4.3.1            BiocNeighbors_1.18.0     
[101] DelayedArray_0.26.7       plotly_4.10.4            
[103] rtracklayer_1.60.1        scales_1.3.0             
[105] lmtest_0.9-40             digest_0.6.34            
[107] goftest_1.2-3             spatstat.utils_3.0-4     
[109] rmarkdown_2.25            XVector_0.40.0           
[111] htmltools_0.5.7           pkgconfig_2.0.3          
[113] sparseMatrixStats_1.12.2  fastmap_1.1.1            
[115] rlang_1.1.3               htmlwidgets_1.6.4        
[117] shiny_1.8.0               DelayedMatrixStats_1.22.6
[119] farver_2.1.1              zoo_1.8-12               
[121] jsonlite_1.8.8            BiocParallel_1.34.2      
[123] ModelMetrics_1.2.2.2      R.oo_1.26.0              
[125] BiocSingular_1.16.0       RCurl_1.98-1.14          
[127] magrittr_2.0.3            modeltools_0.2-23        
[129] GenomeInfoDbData_1.2.10   dotCall64_1.1-1          
[131] munsell_0.5.0             viridis_0.6.5            
[133] reticulate_1.35.0         pROC_1.18.5              
[135] stringi_1.8.3             zlibbioc_1.46.0          
[137] MASS_7.3-60.0.1           org.Hs.eg.db_3.18.0      
[139] plyr_1.8.9                pkgbuild_1.4.3           
[141] ggstats_0.5.1             parallel_4.3.1           
[143] listenv_0.9.1             ggrepel_0.9.5            
[145] deldir_2.0-2              Biostrings_2.68.1        
[147] splines_4.3.1             tensor_1.5               
[149] hms_1.1.3                 locfit_1.5-9.8           
[151] igraph_2.0.2              spatstat.geom_3.2-8      
[153] RcppHNSW_0.6.0            reshape2_1.4.4           
[155] ScaledMatrix_1.8.1        pkgload_1.3.4            
[157] XML_3.99-0.16.1           evaluate_0.23            
[159] scran_1.28.2              BiocManager_1.30.22      
[161] foreach_1.5.2             tzdb_0.4.0               
[163] httpuv_1.6.14             RANN_2.6.1               
[165] polyclip_1.10-6           future_1.33.1            
[167] scattermore_1.2           rsvd_1.0.5               
[169] xtable_1.8-4              restfulr_0.0.15          
[171] svMisc_1.2.3              RSpectra_0.16-1          
[173] later_1.3.2               class_7.3-22             
[175] viridisLite_0.4.2         AnnotationDbi_1.64.1     
[177] GenomicAlignments_1.36.0  memoise_2.0.1            
[179] beeswarm_0.4.0            tximport_1.28.0          
[181] cluster_2.1.6             timechange_0.3.0         
[183] globals_0.16.2            caret_6.0-94             </code></pre>
</div>
</div>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="./Skin_scRNA_Part2.html" class="pagination-link" aria-label="Skin: Normalization, doublet discrimination, integration, clustering">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Skin: Normalization, doublet discrimination, integration, clustering</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="./Skin_scRNA_Part4.html" class="pagination-link" aria-label="Skin: Post-clustering QC and characterization">
        <span class="nav-page-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Skin: Post-clustering QC and characterization</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->




</body></html>