<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.6.39">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>2&nbsp; Skin: Normalization, doublet discrimination, integration, clustering â€“ Blass &amp; Keskin (2025): Melanoma NeoVax analysis: Skin single-cell RNAseq</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./Skin_scRNA_Part3.html" rel="next">
<link href="./Skin_scRNA_Part1.html" rel="prev">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-e26003cea8cd680ca0c55a263523d882.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap-41a4eb7d26fa9333a8f88b12b94df59e.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


</head>

<body class="nav-sidebar floating">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./Skin_scRNA_Part2.html"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Skin: Normalization, doublet discrimination, integration, clustering</span></a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="./">Blass &amp; Keskin (2025): Melanoma NeoVax analysis: Skin single-cell RNAseq</a> 
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Introduction</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Skin_scRNA_Part1.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Skin: QC &amp; Filtering</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Skin_scRNA_Part2.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Skin: Normalization, doublet discrimination, integration, clustering</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Skin_scRNA_Part3.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Skin: Check patient sexes</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Skin_scRNA_Part4.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Skin: Post-clustering QC and characterization</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Skin_scRNA_Part5.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">Skin: Post-clustering comprehensive marker identification</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Skin_scRNA_Part6.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">Skin: Sub-clustering on T/NK lineage</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Skin_scRNA_Part7.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">Skin: Assisting cell type annotation with enrichR</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Skin_scRNA_Part8.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">8</span>&nbsp; <span class="chapter-title">Skin: Classification of CD4 and CD8 T cell states within T/NK lineage</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Skin_scRNA_Part9.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">9</span>&nbsp; <span class="chapter-title">Skin: Differential expression analysis, Post vs Pre 3rd vaccination</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Skin_scRNA_Part10.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">10</span>&nbsp; <span class="chapter-title">Skin: Deep phenotyping, cluster annotation, and exploration of Post vs Pre 3rd vaccination differences in cell proportions and expression</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Skin_scRNA_Part11.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">11</span>&nbsp; <span class="chapter-title">Skin: Calculating Post vs Pre 3rd vaccination differences in pathway enrichments with fgsea</span></span></a>
  </div>
</li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#set-up-seurat-workspace" id="toc-set-up-seurat-workspace" class="nav-link active" data-scroll-target="#set-up-seurat-workspace"><span class="header-section-number">2.1</span> Set up Seurat workspace</a></li>
  <li><a href="#load-previously-saved-object" id="toc-load-previously-saved-object" class="nav-link" data-scroll-target="#load-previously-saved-object"><span class="header-section-number">2.2</span> Load previously saved object</a></li>
  <li><a href="#normalize-and-scale-data" id="toc-normalize-and-scale-data" class="nav-link" data-scroll-target="#normalize-and-scale-data"><span class="header-section-number">2.3</span> Normalize and scale data</a></li>
  <li><a href="#run-pca" id="toc-run-pca" class="nav-link" data-scroll-target="#run-pca"><span class="header-section-number">2.4</span> Run PCA</a></li>
  <li><a href="#plot-unintegrated-umap" id="toc-plot-unintegrated-umap" class="nav-link" data-scroll-target="#plot-unintegrated-umap"><span class="header-section-number">2.5</span> Plot unintegrated UMAP</a></li>
  <li><a href="#call-preliminary-clusters-for-the-purposes-of-doublet-discrimination" id="toc-call-preliminary-clusters-for-the-purposes-of-doublet-discrimination" class="nav-link" data-scroll-target="#call-preliminary-clusters-for-the-purposes-of-doublet-discrimination"><span class="header-section-number">2.6</span> Call preliminary clusters for the purposes of doublet discrimination</a></li>
  <li><a href="#identify-and-remove-doublets" id="toc-identify-and-remove-doublets" class="nav-link" data-scroll-target="#identify-and-remove-doublets"><span class="header-section-number">2.7</span> Identify and remove doublets</a>
  <ul class="collapse">
  <li><a href="#combine-rna-layers" id="toc-combine-rna-layers" class="nav-link" data-scroll-target="#combine-rna-layers"><span class="header-section-number">2.7.1</span> Combine RNA layers</a></li>
  <li><a href="#convert-seurat-to-sce-and-check-coldata" id="toc-convert-seurat-to-sce-and-check-coldata" class="nav-link" data-scroll-target="#convert-seurat-to-sce-and-check-coldata"><span class="header-section-number">2.7.2</span> Convert seurat to sce and check colData</a></li>
  <li><a href="#run-scdblfinder" id="toc-run-scdblfinder" class="nav-link" data-scroll-target="#run-scdblfinder"><span class="header-section-number">2.7.3</span> Run scDblFinder</a></li>
  <li><a href="#inspect-results" id="toc-inspect-results" class="nav-link" data-scroll-target="#inspect-results"><span class="header-section-number">2.7.4</span> Inspect results</a></li>
  <li><a href="#save-doublet-classifications-into-main-seurat-object" id="toc-save-doublet-classifications-into-main-seurat-object" class="nav-link" data-scroll-target="#save-doublet-classifications-into-main-seurat-object"><span class="header-section-number">2.7.5</span> Save doublet classifications into main Seurat object</a></li>
  <li><a href="#count-singlets-and-doublets" id="toc-count-singlets-and-doublets" class="nav-link" data-scroll-target="#count-singlets-and-doublets"><span class="header-section-number">2.7.6</span> Count singlets and doublets</a></li>
  <li><a href="#plot-singletsdoublets-in-umap-space" id="toc-plot-singletsdoublets-in-umap-space" class="nav-link" data-scroll-target="#plot-singletsdoublets-in-umap-space"><span class="header-section-number">2.7.7</span> Plot singlets/doublets in UMAP space</a></li>
  <li><a href="#subset-object-to-remove-doublets-and-count-remaining-cells" id="toc-subset-object-to-remove-doublets-and-count-remaining-cells" class="nav-link" data-scroll-target="#subset-object-to-remove-doublets-and-count-remaining-cells"><span class="header-section-number">2.7.8</span> Subset object to remove doublets and count remaining cells</a></li>
  </ul></li>
  <li><a href="#remove-cells-with-very-high-ncount_rna-values-set-other-final-qc-filters" id="toc-remove-cells-with-very-high-ncount_rna-values-set-other-final-qc-filters" class="nav-link" data-scroll-target="#remove-cells-with-very-high-ncount_rna-values-set-other-final-qc-filters"><span class="header-section-number">2.8</span> Remove cells with very high nCount_RNA values, set other final QC filters</a></li>
  <li><a href="#re-compute-pca" id="toc-re-compute-pca" class="nav-link" data-scroll-target="#re-compute-pca"><span class="header-section-number">2.9</span> Re-compute PCA</a></li>
  <li><a href="#run-harmony-integration" id="toc-run-harmony-integration" class="nav-link" data-scroll-target="#run-harmony-integration"><span class="header-section-number">2.10</span> Run Harmony integration</a></li>
  <li><a href="#identify-clusters-after-integration-using-a-range-of-resolution" id="toc-identify-clusters-after-integration-using-a-range-of-resolution" class="nav-link" data-scroll-target="#identify-clusters-after-integration-using-a-range-of-resolution"><span class="header-section-number">3</span> Identify clusters after integration using a range of resolution</a></li>
  <li><a href="#plot-clusters" id="toc-plot-clusters" class="nav-link" data-scroll-target="#plot-clusters"><span class="header-section-number">4</span> Plot clusters</a>
  <ul class="collapse">
  <li><a href="#plot-one-as-example" id="toc-plot-one-as-example" class="nav-link" data-scroll-target="#plot-one-as-example"><span class="header-section-number">4.1</span> Plot one as example</a></li>
  <li><a href="#plot-metadata-in-umap-space" id="toc-plot-metadata-in-umap-space" class="nav-link" data-scroll-target="#plot-metadata-in-umap-space"><span class="header-section-number">4.2</span> Plot metadata in UMAP space</a></li>
  <li><a href="#save-clustered-object" id="toc-save-clustered-object" class="nav-link" data-scroll-target="#save-clustered-object"><span class="header-section-number">4.3</span> Save clustered object</a></li>
  <li><a href="#get-session-info" id="toc-get-session-info" class="nav-link" data-scroll-target="#get-session-info"><span class="header-section-number">4.4</span> Get session info</a></li>
  </ul></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title"><span id="sec-skin-normalizaation-clustering" class="quarto-section-identifier"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Skin: Normalization, doublet discrimination, integration, clustering</span></span></h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<section id="set-up-seurat-workspace" class="level2" data-number="2.1">
<h2 data-number="2.1" class="anchored" data-anchor-id="set-up-seurat-workspace"><span class="header-section-number">2.1</span> Set up Seurat workspace</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load libraries</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(data.table)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(devtools)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(presto)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(BiocParallel)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(glmGamPoi)</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(sctransform)</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Seurat)</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(miQC)</span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(SeuratWrappers)</span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(flexmix)</span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(SingleCellExperiment)</span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(SummarizedExperiment)</span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readxl)</span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(fishpond)</span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Matrix)</span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(speckle)</span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(scater)</span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(vctrs)</span>
<span id="cb1-22"><a href="#cb1-22" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(alevinQC)</span>
<span id="cb1-23"><a href="#cb1-23" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(harmony)</span>
<span id="cb1-24"><a href="#cb1-24" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(scDblFinder)</span>
<span id="cb1-25"><a href="#cb1-25" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(cellXY)</span>
<span id="cb1-26"><a href="#cb1-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-27"><a href="#cb1-27" aria-hidden="true" tabindex="-1"></a><span class="co"># Set global options for Seurat v5 objects</span></span>
<span id="cb1-28"><a href="#cb1-28" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">Seurat.object.assay.version =</span> <span class="st">'v5'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="load-previously-saved-object" class="level2" data-number="2.2">
<h2 data-number="2.2" class="anchored" data-anchor-id="load-previously-saved-object"><span class="header-section-number">2.2</span> Load previously saved object</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>merged.<span class="fl">18279.</span>skin <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"Skin_scRNA_Part1.rds"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="normalize-and-scale-data" class="level2" data-number="2.3">
<h2 data-number="2.3" class="anchored" data-anchor-id="normalize-and-scale-data"><span class="header-section-number">2.3</span> Normalize and scale data</h2>
<p>Regress out mitochondrial contribution</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>merged.<span class="fl">18279.</span>skin <span class="ot">&lt;-</span> <span class="fu">NormalizeData</span>(merged.<span class="fl">18279.</span>skin)</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>merged.<span class="fl">18279.</span>skin <span class="ot">&lt;-</span> <span class="fu">FindVariableFeatures</span>(merged.<span class="fl">18279.</span>skin, </span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">assay=</span><span class="st">"RNA"</span>, </span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">layer=</span><span class="st">"data"</span>, </span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">selection.method =</span> <span class="st">"vst"</span>, </span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">nfeatures =</span> <span class="dv">5000</span>)</span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a>merged.<span class="fl">18279.</span>skin <span class="ot">&lt;-</span> <span class="fu">ScaleData</span>(merged.<span class="fl">18279.</span>skin, <span class="at">vars.to.regress =</span> <span class="st">"percent.mt"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="run-pca" class="level2" data-number="2.4">
<h2 data-number="2.4" class="anchored" data-anchor-id="run-pca"><span class="header-section-number">2.4</span> Run PCA</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>merged.<span class="fl">18279.</span>skin <span class="ot">&lt;-</span> <span class="fu">RunPCA</span>(merged.<span class="fl">18279.</span>skin, <span class="at">npcs =</span> <span class="dv">200</span>, <span class="at">verbose =</span> <span class="cn">FALSE</span>)</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ElbowPlot</span>(merged.<span class="fl">18279.</span>skin, <span class="at">ndims =</span> <span class="dv">200</span>, <span class="at">reduction =</span> <span class="st">"pca"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Skin_scRNA_Part2_files/figure-html/unnamed-chunk-4-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="plot-unintegrated-umap" class="level2" data-number="2.5">
<h2 data-number="2.5" class="anchored" data-anchor-id="plot-unintegrated-umap"><span class="header-section-number">2.5</span> Plot unintegrated UMAP</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>merged.<span class="fl">18279.</span>skin <span class="ot">&lt;-</span> <span class="fu">RunUMAP</span>(merged.<span class="fl">18279.</span>skin, </span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>                        <span class="at">reduction =</span> <span class="st">"pca"</span>, </span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>                        <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">50</span>, </span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a>                        <span class="at">reduction.name =</span> <span class="st">"umap.unintegrated"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: The default method for RunUMAP has changed from calling Python UMAP via reticulate to the R-native UWOT using the cosine metric
To use Python UMAP via reticulate, set umap.method to 'umap-learn' and metric to 'correlation'
This message will be shown once per session</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>15:12:49 UMAP embedding parameters a = 0.9922 b = 1.112</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "dist" in cache; using the first, from namespace 'BiocGenerics'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'spam'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>15:12:49 Read 46891 rows and found 50 numeric columns</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>15:12:49 Using Annoy for neighbor search, n_neighbors = 30</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "dist" in cache; using the first, from namespace 'BiocGenerics'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'spam'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>15:12:49 Building Annoy index with metric = cosine, n_trees = 50</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>0%   10   20   30   40   50   60   70   80   90   100%</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>[----|----|----|----|----|----|----|----|----|----|</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>**************************************************|
15:13:05 Writing NN index file to temp file /tmp/Rtmp5cppMq/file39fc2036ef469f
15:13:05 Searching Annoy index using 1 thread, search_k = 3000
15:13:27 Annoy recall = 100%
15:13:28 Commencing smooth kNN distance calibration using 1 thread with target n_neighbors = 30
15:13:33 Initializing from normalized Laplacian + noise (using RSpectra)
15:13:38 Commencing optimization for 200 epochs, with 2042658 positive edges
15:14:13 Optimization finished</code></pre>
</div>
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(merged.<span class="fl">18279.</span>skin, <span class="at">reduction =</span> <span class="st">"umap.unintegrated"</span>, <span class="at">group.by =</span> <span class="fu">c</span>(<span class="st">"Site"</span>, <span class="st">"Patient"</span>, <span class="st">"Timepoint"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Skin_scRNA_Part2_files/figure-html/unnamed-chunk-5-1.png" class="img-fluid figure-img" width="1152"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="call-preliminary-clusters-for-the-purposes-of-doublet-discrimination" class="level2" data-number="2.6">
<h2 data-number="2.6" class="anchored" data-anchor-id="call-preliminary-clusters-for-the-purposes-of-doublet-discrimination"><span class="header-section-number">2.6</span> Call preliminary clusters for the purposes of doublet discrimination</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a>merged.<span class="fl">18279.</span>skin <span class="ot">&lt;-</span> <span class="fu">FindNeighbors</span>(merged.<span class="fl">18279.</span>skin, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">50</span>, <span class="at">reduction =</span> <span class="st">"pca"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Computing nearest neighbor graph</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Computing SNN</code></pre>
</div>
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a>merged.<span class="fl">18279.</span>skin <span class="ot">&lt;-</span> <span class="fu">FindClusters</span>(merged.<span class="fl">18279.</span>skin, </span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a>                             <span class="at">resolution =</span> <span class="fl">0.4</span>, </span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a>                             <span class="at">algorithm =</span> <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 46891
Number of edges: 1576348

Running Louvain algorithm with multilevel refinement...
Maximum modularity in 10 random starts: 0.9313
Number of communities: 22
Elapsed time: 13 seconds</code></pre>
</div>
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(merged.<span class="fl">18279.</span>skin, <span class="at">reduction =</span> <span class="st">"umap.unintegrated"</span>, <span class="at">label =</span> T)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Skin_scRNA_Part2_files/figure-html/unnamed-chunk-6-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="identify-and-remove-doublets" class="level2" data-number="2.7">
<h2 data-number="2.7" class="anchored" data-anchor-id="identify-and-remove-doublets"><span class="header-section-number">2.7</span> Identify and remove doublets</h2>
<p>This uses raw counts as input</p>
<section id="combine-rna-layers" class="level3" data-number="2.7.1">
<h3 data-number="2.7.1" class="anchored" data-anchor-id="combine-rna-layers"><span class="header-section-number">2.7.1</span> Combine RNA layers</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a>merged.<span class="fl">18279.</span>skin[[<span class="st">'RNA'</span>]] <span class="ot">&lt;-</span> <span class="fu">JoinLayers</span>(merged.<span class="fl">18279.</span>skin[[<span class="st">'RNA'</span>]])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="convert-seurat-to-sce-and-check-coldata" class="level3" data-number="2.7.2">
<h3 data-number="2.7.2" class="anchored" data-anchor-id="convert-seurat-to-sce-and-check-coldata"><span class="header-section-number">2.7.2</span> Convert seurat to sce and check colData</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a>merged.<span class="fl">18279.</span>skin.sce <span class="ot">&lt;-</span> <span class="fu">as.SingleCellExperiment</span>(merged.<span class="fl">18279.</span>skin, <span class="at">assay =</span> <span class="st">"RNA"</span>)</span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a>merged.<span class="fl">18279.</span>skin.sce</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>class: SingleCellExperiment 
dim: 61217 46891 
metadata(0):
assays(2): counts logcounts
rownames(61217): 5-8S-rRNA 5S-rRNA ... ZZEF1 ZZZ3
rowData names(0):
colnames(46891): P101_Skin_Pre3rd_2.5mgIpi_RNA_AAACCTGCACCATGTA
  P101_Skin_Pre3rd_2.5mgIpi_RNA_CACATAGCAGGCTCAC ...
  P111_Skin_Post3rd_5mgIpi_RNA_TGGCCACCAGGCTGAA
  P111_Skin_Post3rd_5mgIpi_RNA_TGAGCCGAGACGCGTT
colData names(15): orig.ident nCount_RNA ... seurat_clusters ident
reducedDimNames(2): PCA UMAP.UNINTEGRATED
mainExpName: RNA
altExpNames(0):</code></pre>
</div>
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colData</span>(merged.<span class="fl">18279.</span>skin.sce)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>DataFrame with 46891 rows and 15 columns
                                                  orig.ident nCount_RNA
                                                 &lt;character&gt;  &lt;numeric&gt;
P101_Skin_Pre3rd_2.5mgIpi_RNA_AAACCTGCACCATGTA SeuratProject    26987.6
P101_Skin_Pre3rd_2.5mgIpi_RNA_CACATAGCAGGCTCAC SeuratProject    26321.6
P101_Skin_Pre3rd_2.5mgIpi_RNA_GCACATAAGTGCCATT SeuratProject    29916.6
P101_Skin_Pre3rd_2.5mgIpi_RNA_TTAGTTCAGCACCGTC SeuratProject    28724.6
P101_Skin_Pre3rd_2.5mgIpi_RNA_GGAACTTGTCGCGGTT SeuratProject    25775.8
...                                                      ...        ...
P111_Skin_Post3rd_5mgIpi_RNA_CAGGTGCGTCGCTTTA  SeuratProject        532
P111_Skin_Post3rd_5mgIpi_RNA_GTCACGACACATTTCT  SeuratProject        502
P111_Skin_Post3rd_5mgIpi_RNA_GAGACCTAGGGTCTCC  SeuratProject        618
P111_Skin_Post3rd_5mgIpi_RNA_TGGCCACCAGGCTGAA  SeuratProject        508
P111_Skin_Post3rd_5mgIpi_RNA_TGAGCCGAGACGCGTT  SeuratProject        511
                                               nFeature_RNA percent.mt
                                                  &lt;integer&gt;  &lt;numeric&gt;
P101_Skin_Pre3rd_2.5mgIpi_RNA_AAACCTGCACCATGTA         5591    3.39147
P101_Skin_Pre3rd_2.5mgIpi_RNA_CACATAGCAGGCTCAC         5500    3.17440
P101_Skin_Pre3rd_2.5mgIpi_RNA_GCACATAAGTGCCATT         6213    6.09781
P101_Skin_Pre3rd_2.5mgIpi_RNA_TTAGTTCAGCACCGTC         5440    4.31684
P101_Skin_Pre3rd_2.5mgIpi_RNA_GGAACTTGTCGCGGTT         4171    2.80602
...                                                     ...        ...
P111_Skin_Post3rd_5mgIpi_RNA_CAGGTGCGTCGCTTTA           423   1.879699
P111_Skin_Post3rd_5mgIpi_RNA_GTCACGACACATTTCT           484   0.796813
P111_Skin_Post3rd_5mgIpi_RNA_GAGACCTAGGGTCTCC           508   1.941748
P111_Skin_Post3rd_5mgIpi_RNA_TGGCCACCAGGCTGAA           452   1.279528
P111_Skin_Post3rd_5mgIpi_RNA_TGAGCCGAGACGCGTT           386   0.391389
                                               miQC.probability   miQC.keep
                                                      &lt;numeric&gt; &lt;character&gt;
P101_Skin_Pre3rd_2.5mgIpi_RNA_AAACCTGCACCATGTA        0.0454933        keep
P101_Skin_Pre3rd_2.5mgIpi_RNA_CACATAGCAGGCTCAC        0.0462136        keep
P101_Skin_Pre3rd_2.5mgIpi_RNA_GCACATAAGTGCCATT        0.0893537        keep
P101_Skin_Pre3rd_2.5mgIpi_RNA_TTAGTTCAGCACCGTC        0.0520042        keep
P101_Skin_Pre3rd_2.5mgIpi_RNA_GGAACTTGTCGCGGTT        0.0498276        keep
...                                                         ...         ...
P111_Skin_Post3rd_5mgIpi_RNA_CAGGTGCGTCGCTTTA         0.0148312        keep
P111_Skin_Post3rd_5mgIpi_RNA_GTCACGACACATTTCT         0.0225911        keep
P111_Skin_Post3rd_5mgIpi_RNA_GAGACCTAGGGTCTCC         0.0154750        keep
P111_Skin_Post3rd_5mgIpi_RNA_TGGCCACCAGGCTGAA         0.0180636        keep
P111_Skin_Post3rd_5mgIpi_RNA_TGAGCCGAGACGCGTT         0.0260453        keep
                                                   Patient        Site
                                               &lt;character&gt; &lt;character&gt;
P101_Skin_Pre3rd_2.5mgIpi_RNA_AAACCTGCACCATGTA        P101        Skin
P101_Skin_Pre3rd_2.5mgIpi_RNA_CACATAGCAGGCTCAC        P101        Skin
P101_Skin_Pre3rd_2.5mgIpi_RNA_GCACATAAGTGCCATT        P101        Skin
P101_Skin_Pre3rd_2.5mgIpi_RNA_TTAGTTCAGCACCGTC        P101        Skin
P101_Skin_Pre3rd_2.5mgIpi_RNA_GGAACTTGTCGCGGTT        P101        Skin
...                                                    ...         ...
P111_Skin_Post3rd_5mgIpi_RNA_CAGGTGCGTCGCTTTA         P111        Skin
P111_Skin_Post3rd_5mgIpi_RNA_GTCACGACACATTTCT         P111        Skin
P111_Skin_Post3rd_5mgIpi_RNA_GAGACCTAGGGTCTCC         P111        Skin
P111_Skin_Post3rd_5mgIpi_RNA_TGGCCACCAGGCTGAA         P111        Skin
P111_Skin_Post3rd_5mgIpi_RNA_TGAGCCGAGACGCGTT         P111        Skin
                                                 Timepoint   IpiCohort
                                               &lt;character&gt; &lt;character&gt;
P101_Skin_Pre3rd_2.5mgIpi_RNA_AAACCTGCACCATGTA      Pre3rd    2.5mgIpi
P101_Skin_Pre3rd_2.5mgIpi_RNA_CACATAGCAGGCTCAC      Pre3rd    2.5mgIpi
P101_Skin_Pre3rd_2.5mgIpi_RNA_GCACATAAGTGCCATT      Pre3rd    2.5mgIpi
P101_Skin_Pre3rd_2.5mgIpi_RNA_TTAGTTCAGCACCGTC      Pre3rd    2.5mgIpi
P101_Skin_Pre3rd_2.5mgIpi_RNA_GGAACTTGTCGCGGTT      Pre3rd    2.5mgIpi
...                                                    ...         ...
P111_Skin_Post3rd_5mgIpi_RNA_CAGGTGCGTCGCTTTA      Post3rd      5mgIpi
P111_Skin_Post3rd_5mgIpi_RNA_GTCACGACACATTTCT      Post3rd      5mgIpi
P111_Skin_Post3rd_5mgIpi_RNA_GAGACCTAGGGTCTCC      Post3rd      5mgIpi
P111_Skin_Post3rd_5mgIpi_RNA_TGGCCACCAGGCTGAA      Post3rd      5mgIpi
P111_Skin_Post3rd_5mgIpi_RNA_TGAGCCGAGACGCGTT      Post3rd      5mgIpi
                                                     Assay
                                               &lt;character&gt;
P101_Skin_Pre3rd_2.5mgIpi_RNA_AAACCTGCACCATGTA         RNA
P101_Skin_Pre3rd_2.5mgIpi_RNA_CACATAGCAGGCTCAC         RNA
P101_Skin_Pre3rd_2.5mgIpi_RNA_GCACATAAGTGCCATT         RNA
P101_Skin_Pre3rd_2.5mgIpi_RNA_TTAGTTCAGCACCGTC         RNA
P101_Skin_Pre3rd_2.5mgIpi_RNA_GGAACTTGTCGCGGTT         RNA
...                                                    ...
P111_Skin_Post3rd_5mgIpi_RNA_CAGGTGCGTCGCTTTA          RNA
P111_Skin_Post3rd_5mgIpi_RNA_GTCACGACACATTTCT          RNA
P111_Skin_Post3rd_5mgIpi_RNA_GAGACCTAGGGTCTCC          RNA
P111_Skin_Post3rd_5mgIpi_RNA_TGGCCACCAGGCTGAA          RNA
P111_Skin_Post3rd_5mgIpi_RNA_TGAGCCGAGACGCGTT          RNA
                                                               Sample
                                                          &lt;character&gt;
P101_Skin_Pre3rd_2.5mgIpi_RNA_AAACCTGCACCATGTA P101_Skin_Pre3rd_2.5..
P101_Skin_Pre3rd_2.5mgIpi_RNA_CACATAGCAGGCTCAC P101_Skin_Pre3rd_2.5..
P101_Skin_Pre3rd_2.5mgIpi_RNA_GCACATAAGTGCCATT P101_Skin_Pre3rd_2.5..
P101_Skin_Pre3rd_2.5mgIpi_RNA_TTAGTTCAGCACCGTC P101_Skin_Pre3rd_2.5..
P101_Skin_Pre3rd_2.5mgIpi_RNA_GGAACTTGTCGCGGTT P101_Skin_Pre3rd_2.5..
...                                                               ...
P111_Skin_Post3rd_5mgIpi_RNA_CAGGTGCGTCGCTTTA  P111_Skin_Post3rd_5m..
P111_Skin_Post3rd_5mgIpi_RNA_GTCACGACACATTTCT  P111_Skin_Post3rd_5m..
P111_Skin_Post3rd_5mgIpi_RNA_GAGACCTAGGGTCTCC  P111_Skin_Post3rd_5m..
P111_Skin_Post3rd_5mgIpi_RNA_TGGCCACCAGGCTGAA  P111_Skin_Post3rd_5m..
P111_Skin_Post3rd_5mgIpi_RNA_TGAGCCGAGACGCGTT  P111_Skin_Post3rd_5m..
                                               RNA_snn_res.0.4 seurat_clusters
                                                      &lt;factor&gt;        &lt;factor&gt;
P101_Skin_Pre3rd_2.5mgIpi_RNA_AAACCTGCACCATGTA              9               9 
P101_Skin_Pre3rd_2.5mgIpi_RNA_CACATAGCAGGCTCAC              13              13
P101_Skin_Pre3rd_2.5mgIpi_RNA_GCACATAAGTGCCATT              12              12
P101_Skin_Pre3rd_2.5mgIpi_RNA_TTAGTTCAGCACCGTC              1               1 
P101_Skin_Pre3rd_2.5mgIpi_RNA_GGAACTTGTCGCGGTT              3               3 
...                                                        ...             ...
P111_Skin_Post3rd_5mgIpi_RNA_CAGGTGCGTCGCTTTA               4               4 
P111_Skin_Post3rd_5mgIpi_RNA_GTCACGACACATTTCT               11              11
P111_Skin_Post3rd_5mgIpi_RNA_GAGACCTAGGGTCTCC               11              11
P111_Skin_Post3rd_5mgIpi_RNA_TGGCCACCAGGCTGAA               4               4 
P111_Skin_Post3rd_5mgIpi_RNA_TGAGCCGAGACGCGTT               4               4 
                                                  ident
                                               &lt;factor&gt;
P101_Skin_Pre3rd_2.5mgIpi_RNA_AAACCTGCACCATGTA       9 
P101_Skin_Pre3rd_2.5mgIpi_RNA_CACATAGCAGGCTCAC       13
P101_Skin_Pre3rd_2.5mgIpi_RNA_GCACATAAGTGCCATT       12
P101_Skin_Pre3rd_2.5mgIpi_RNA_TTAGTTCAGCACCGTC       1 
P101_Skin_Pre3rd_2.5mgIpi_RNA_GGAACTTGTCGCGGTT       3 
...                                                 ...
P111_Skin_Post3rd_5mgIpi_RNA_CAGGTGCGTCGCTTTA        4 
P111_Skin_Post3rd_5mgIpi_RNA_GTCACGACACATTTCT        11
P111_Skin_Post3rd_5mgIpi_RNA_GAGACCTAGGGTCTCC        11
P111_Skin_Post3rd_5mgIpi_RNA_TGGCCACCAGGCTGAA        4 
P111_Skin_Post3rd_5mgIpi_RNA_TGAGCCGAGACGCGTT        4 </code></pre>
</div>
</div>
</section>
<section id="run-scdblfinder" class="level3" data-number="2.7.3">
<h3 data-number="2.7.3" class="anchored" data-anchor-id="run-scdblfinder"><span class="header-section-number">2.7.3</span> Run scDblFinder</h3>
<p>Set the <code>dbr.sd</code> very high to better allow thresholds to be set based on misclassification rates per sample</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a>merged.<span class="fl">18279.</span>skin.sce <span class="ot">&lt;-</span> <span class="fu">scDblFinder</span>(merged.<span class="fl">18279.</span>skin.sce,</span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a>                    <span class="at">samples =</span> <span class="st">"Sample"</span>,</span>
<span id="cb30-3"><a href="#cb30-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">dbr.sd =</span> <span class="dv">1</span>,</span>
<span id="cb30-4"><a href="#cb30-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">clusters =</span> <span class="st">"seurat_clusters"</span>,</span>
<span id="cb30-5"><a href="#cb30-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">BPPARAM =</span> <span class="fu">MulticoreParam</span>(<span class="dv">4</span>,<span class="at">RNGseed=</span><span class="dv">123</span>)</span>
<span id="cb30-6"><a href="#cb30-6" aria-hidden="true" tabindex="-1"></a>                )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="inspect-results" class="level3" data-number="2.7.4">
<h3 data-number="2.7.4" class="anchored" data-anchor-id="inspect-results"><span class="header-section-number">2.7.4</span> Inspect results</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Look at the classes</span></span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(merged.<span class="fl">18279.</span>skin.sce<span class="sc">$</span>seurat_clusters, merged.<span class="fl">18279.</span>skin.sce<span class="sc">$</span>scDblFinder.class)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>    
     singlet doublet
  0     8034     407
  1     5222    1273
  2     6077     347
  3     1537    1756
  4     3038      17
  5     2570     206
  6     1923     152
  7     1668     228
  8     1762      97
  9     1613     242
  10    1548      97
  11    1464      34
  12    1197     149
  13     549     295
  14     572     148
  15     518     177
  16     511     109
  17     294     158
  18     286      72
  19     184     112
  20     141      46
  21      60       1</code></pre>
</div>
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(merged.<span class="fl">18279.</span>skin.sce<span class="sc">$</span>Sample, merged.<span class="fl">18279.</span>skin.sce<span class="sc">$</span>scDblFinder.class)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                                
                                 singlet doublet
  P101_Skin_Post3rd_2.5mgIpi_RNA    2005     333
  P101_Skin_Pre3rd_2.5mgIpi_RNA      549      73
  P103_Skin_Post3rd_2.5mgIpi_RNA    4675     665
  P103_Skin_Pre3rd_2.5mgIpi_RNA     1049     165
  P104_Skin_Post3rd_2.5mgIpi_RNA    6422     974
  P104_Skin_Pre3rd_2.5mgIpi_RNA     3619     672
  P105_Skin_Post3rd_2.5mgIpi_RNA    3996     672
  P105_Skin_Pre3rd_2.5mgIpi_RNA     3017     324
  P106_Skin_Post3rd_2.5mgIpi_RNA    1939     275
  P106_Skin_Pre3rd_2.5mgIpi_RNA     2015     208
  P108_Skin_Post3rd_5mgIpi_RNA      3108     493
  P108_Skin_Pre3rd_5mgIpi_RNA       1148     129
  P109_Skin_Pre3rd_5mgIpi_RNA       1870     274
  P110_Skin_Post3rd_5mgIpi_RNA      1658     321
  P110_Skin_Pre3rd_5mgIpi_RNA       1381     212
  P111_Skin_Post3rd_5mgIpi_RNA      1724     225
  P111_Skin_Pre3rd_5mgIpi_RNA        593     108</code></pre>
</div>
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Look at the scores</span></span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(merged.<span class="fl">18279.</span>skin.sce<span class="sc">$</span>scDblFinder.score)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>     Min.   1st Qu.    Median      Mean   3rd Qu.      Max. 
0.0001124 0.0242574 0.0842477 0.2207687 0.2681302 0.9999770 </code></pre>
</div>
</div>
</section>
<section id="save-doublet-classifications-into-main-seurat-object" class="level3" data-number="2.7.5">
<h3 data-number="2.7.5" class="anchored" data-anchor-id="save-doublet-classifications-into-main-seurat-object"><span class="header-section-number">2.7.5</span> Save doublet classifications into main Seurat object</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a>merged.<span class="fl">18279.</span>skin<span class="sc">$</span>doublet_classification <span class="ot">&lt;-</span> merged.<span class="fl">18279.</span>skin.sce<span class="sc">$</span>scDblFinder.class</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="count-singlets-and-doublets" class="level3" data-number="2.7.6">
<h3 data-number="2.7.6" class="anchored" data-anchor-id="count-singlets-and-doublets"><span class="header-section-number">2.7.6</span> Count singlets and doublets</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb38"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(merged.<span class="fl">18279.</span>skin<span class="sc">$</span>doublet_classification)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
singlet doublet 
  40768    6123 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb40"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(merged.<span class="fl">18279.</span>skin<span class="sc">$</span>doublet_classification, merged.<span class="fl">18279.</span>skin<span class="sc">$</span>seurat_clusters)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>         
             0    1    2    3    4    5    6    7    8    9   10   11   12   13
  singlet 8034 5222 6077 1537 3038 2570 1923 1668 1762 1613 1548 1464 1197  549
  doublet  407 1273  347 1756   17  206  152  228   97  242   97   34  149  295
         
            14   15   16   17   18   19   20   21
  singlet  572  518  511  294  286  184  141   60
  doublet  148  177  109  158   72  112   46    1</code></pre>
</div>
</div>
</section>
<section id="plot-singletsdoublets-in-umap-space" class="level3" data-number="2.7.7">
<h3 data-number="2.7.7" class="anchored" data-anchor-id="plot-singletsdoublets-in-umap-space"><span class="header-section-number">2.7.7</span> Plot singlets/doublets in UMAP space</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb42"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(merged.<span class="fl">18279.</span>skin,<span class="at">reduction =</span> <span class="st">"umap.unintegrated"</span>, <span class="at">group.by =</span> <span class="st">"doublet_classification"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Skin_scRNA_Part2_files/figure-html/unnamed-chunk-13-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="subset-object-to-remove-doublets-and-count-remaining-cells" class="level3" data-number="2.7.8">
<h3 data-number="2.7.8" class="anchored" data-anchor-id="subset-object-to-remove-doublets-and-count-remaining-cells"><span class="header-section-number">2.7.8</span> Subset object to remove doublets and count remaining cells</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb43"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a>merged.<span class="fl">18279.</span>skin.singlets <span class="ot">&lt;-</span> <span class="fu">subset</span>(merged.<span class="fl">18279.</span>skin, doublet_classification <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"singlet"</span>))</span>
<span id="cb43-2"><a href="#cb43-2" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(merged.<span class="fl">18279.</span>skin.singlets)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 61217 40768</code></pre>
</div>
<div class="sourceCode cell-code" id="cb45"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Count remaining cells per initial cluster</span></span>
<span id="cb45-2"><a href="#cb45-2" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(merged.<span class="fl">18279.</span>skin.singlets<span class="sc">$</span>seurat_clusters)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
   0    1    2    3    4    5    6    7    8    9   10   11   12   13   14   15 
8034 5222 6077 1537 3038 2570 1923 1668 1762 1613 1548 1464 1197  549  572  518 
  16   17   18   19   20   21 
 511  294  286  184  141   60 </code></pre>
</div>
</div>
</section>
</section>
<section id="remove-cells-with-very-high-ncount_rna-values-set-other-final-qc-filters" class="level2" data-number="2.8">
<h2 data-number="2.8" class="anchored" data-anchor-id="remove-cells-with-very-high-ncount_rna-values-set-other-final-qc-filters"><span class="header-section-number">2.8</span> Remove cells with very high nCount_RNA values, set other final QC filters</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb47"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a>merged.<span class="fl">18279.</span>skin.singlets <span class="ot">&lt;-</span> <span class="fu">subset</span>(merged.<span class="fl">18279.</span>skin.singlets,</span>
<span id="cb47-2"><a href="#cb47-2" aria-hidden="true" tabindex="-1"></a>                                <span class="at">subset =</span> nCount_RNA <span class="sc">&lt;</span> <span class="dv">40000</span> <span class="sc">&amp;</span> nCount_RNA <span class="sc">&gt;</span> <span class="dv">1500</span> <span class="sc">&amp;</span> nFeature_RNA <span class="sc">&gt;</span> <span class="dv">750</span>)</span>
<span id="cb47-3"><a href="#cb47-3" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(merged.<span class="fl">18279.</span>skin.singlets)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 61217 33397</code></pre>
</div>
</div>
</section>
<section id="re-compute-pca" class="level2" data-number="2.9">
<h2 data-number="2.9" class="anchored" data-anchor-id="re-compute-pca"><span class="header-section-number">2.9</span> Re-compute PCA</h2>
<p>Re-scale data now that it has been subset</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb49"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb49-1"><a href="#cb49-1" aria-hidden="true" tabindex="-1"></a>merged.<span class="fl">18279.</span>skin.singlets[[<span class="st">"RNA"</span>]] <span class="ot">&lt;-</span> <span class="fu">split</span>(merged.<span class="fl">18279.</span>skin.singlets[[<span class="st">"RNA"</span>]], <span class="at">f =</span> merged.<span class="fl">18279.</span>skin.singlets<span class="sc">$</span>Sample)</span>
<span id="cb49-2"><a href="#cb49-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-3"><a href="#cb49-3" aria-hidden="true" tabindex="-1"></a>merged.<span class="fl">18279.</span>skin.singlets <span class="ot">&lt;-</span> <span class="fu">FindVariableFeatures</span>(merged.<span class="fl">18279.</span>skin.singlets, </span>
<span id="cb49-4"><a href="#cb49-4" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">assay =</span> <span class="st">"RNA"</span>, </span>
<span id="cb49-5"><a href="#cb49-5" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">layer =</span> <span class="st">"data"</span>, </span>
<span id="cb49-6"><a href="#cb49-6" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">selection.method =</span> <span class="st">"vst"</span>, </span>
<span id="cb49-7"><a href="#cb49-7" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">nfeatures =</span> <span class="dv">5000</span>)</span>
<span id="cb49-8"><a href="#cb49-8" aria-hidden="true" tabindex="-1"></a>merged.<span class="fl">18279.</span>skin.singlets <span class="ot">&lt;-</span> <span class="fu">ScaleData</span>(merged.<span class="fl">18279.</span>skin.singlets, <span class="at">vars.to.regress =</span> <span class="st">"percent.mt"</span>)</span>
<span id="cb49-9"><a href="#cb49-9" aria-hidden="true" tabindex="-1"></a>merged.<span class="fl">18279.</span>skin.singlets <span class="ot">&lt;-</span> <span class="fu">RunPCA</span>(merged.<span class="fl">18279.</span>skin.singlets, <span class="at">npcs =</span> <span class="dv">200</span>, <span class="at">verbose =</span> <span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="run-harmony-integration" class="level2" data-number="2.10">
<h2 data-number="2.10" class="anchored" data-anchor-id="run-harmony-integration"><span class="header-section-number">2.10</span> Run Harmony integration</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb50"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a>merged.<span class="fl">18279.</span>skin.singlets <span class="ot">&lt;-</span> <span class="fu">IntegrateLayers</span>(merged.<span class="fl">18279.</span>skin.singlets, </span>
<span id="cb50-2"><a href="#cb50-2" aria-hidden="true" tabindex="-1"></a>                                <span class="at">method =</span> HarmonyIntegration, </span>
<span id="cb50-3"><a href="#cb50-3" aria-hidden="true" tabindex="-1"></a>                                <span class="at">orig.reduction =</span> <span class="st">"pca"</span>,</span>
<span id="cb50-4"><a href="#cb50-4" aria-hidden="true" tabindex="-1"></a>                                <span class="at">new.reduction =</span> <span class="st">"integrated.harmony"</span></span>
<span id="cb50-5"><a href="#cb50-5" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: HarmonyMatrix is deprecated and will be removed in the future from the
API in the future</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Warning: The parameters do_pca and npcs are deprecated. They will be ignored for this function call and please remove parameters do_pca and npcs and pass to harmony cell_embeddings directly.
This warning is displayed once per session.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Warning: The parameter tau is deprecated. It will be ignored for this function call and please remove parameter tau in future function calls. Advanced users can set value of parameter tau by using parameter .options and function harmony_options().
This warning is displayed once per session.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Warning: The parameter block.size is deprecated. It will be ignored for this function call and please remove parameter block.size in future function calls. Advanced users can set value of parameter block.size by using parameter .options and function harmony_options().
This warning is displayed once per session.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Warning: The parameter max.iter.harmony is replaced with parameter max_iter. It will be ignored for this function call and please use parameter max_iter in future function calls.
This warning is displayed once per session.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Warning: The parameter max.iter.cluster is deprecated. It will be ignored for this function call and please remove parameter max.iter.cluster in future function calls. Advanced users can set value of parameter max.iter.cluster by using parameter .options and function harmony_options().
This warning is displayed once per session.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Warning: The parameter epsilon.cluster is deprecated. It will be ignored for this function call and please remove parameter epsilon.cluster in future function calls. Advanced users can set value of parameter epsilon.cluster by using parameter .options and function harmony_options().
This warning is displayed once per session.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Warning: The parameter epsilon.harmony is deprecated. It will be ignored for this function call and please remove parameter epsilon.harmony in future function calls. If users want to control if harmony would stop early or not, use parameter early_stop. Advanced users can set value of parameter epsilon.harmony by using parameter .options and function harmony_options().
This warning is displayed once per session.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Transposing data matrix</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Using automatic lambda estimation</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Initializing state using k-means centroids initialization</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Harmony 1/10</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Harmony 2/10</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Harmony 3/10</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Harmony converged after 3 iterations</code></pre>
</div>
</div>
</section>
<section id="identify-clusters-after-integration-using-a-range-of-resolution" class="level1" data-number="3">
<h1 data-number="3"><span class="header-section-number">3</span> Identify clusters after integration using a range of resolution</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb66"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb66-1"><a href="#cb66-1" aria-hidden="true" tabindex="-1"></a>merged.<span class="fl">18279.</span>skin.singlets <span class="ot">&lt;-</span> <span class="fu">FindNeighbors</span>(merged.<span class="fl">18279.</span>skin.singlets, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">50</span>, <span class="at">reduction =</span> <span class="st">"integrated.harmony"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Computing nearest neighbor graph</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Computing SNN</code></pre>
</div>
<div class="sourceCode cell-code" id="cb69"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb69-1"><a href="#cb69-1" aria-hidden="true" tabindex="-1"></a>merged.<span class="fl">18279.</span>skin.singlets <span class="ot">&lt;-</span> <span class="fu">FindClusters</span>(merged.<span class="fl">18279.</span>skin.singlets, </span>
<span id="cb69-2"><a href="#cb69-2" aria-hidden="true" tabindex="-1"></a>                             <span class="at">resolution =</span> <span class="fu">seq</span>(<span class="fl">0.1</span>, <span class="dv">2</span>, <span class="at">by =</span> <span class="fl">0.1</span>), </span>
<span id="cb69-3"><a href="#cb69-3" aria-hidden="true" tabindex="-1"></a>                             <span class="at">algorithm =</span> <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 33397
Number of edges: 1159203

Running Louvain algorithm with multilevel refinement...
Maximum modularity in 10 random starts: 0.9718
Number of communities: 12
Elapsed time: 10 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 33397
Number of edges: 1159203

Running Louvain algorithm with multilevel refinement...
Maximum modularity in 10 random starts: 0.9535
Number of communities: 15
Elapsed time: 10 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 33397
Number of edges: 1159203

Running Louvain algorithm with multilevel refinement...
Maximum modularity in 10 random starts: 0.9373
Number of communities: 19
Elapsed time: 9 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 33397
Number of edges: 1159203

Running Louvain algorithm with multilevel refinement...
Maximum modularity in 10 random starts: 0.9273
Number of communities: 22
Elapsed time: 8 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 33397
Number of edges: 1159203

Running Louvain algorithm with multilevel refinement...
Maximum modularity in 10 random starts: 0.9186
Number of communities: 23
Elapsed time: 8 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 33397
Number of edges: 1159203

Running Louvain algorithm with multilevel refinement...
Maximum modularity in 10 random starts: 0.9099
Number of communities: 23
Elapsed time: 10 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 33397
Number of edges: 1159203

Running Louvain algorithm with multilevel refinement...
Maximum modularity in 10 random starts: 0.9013
Number of communities: 26
Elapsed time: 9 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 33397
Number of edges: 1159203

Running Louvain algorithm with multilevel refinement...
Maximum modularity in 10 random starts: 0.8943
Number of communities: 27
Elapsed time: 9 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 33397
Number of edges: 1159203

Running Louvain algorithm with multilevel refinement...
Maximum modularity in 10 random starts: 0.8877
Number of communities: 27
Elapsed time: 9 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 33397
Number of edges: 1159203

Running Louvain algorithm with multilevel refinement...
Maximum modularity in 10 random starts: 0.8813
Number of communities: 28
Elapsed time: 9 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 33397
Number of edges: 1159203

Running Louvain algorithm with multilevel refinement...
Maximum modularity in 10 random starts: 0.8755
Number of communities: 30
Elapsed time: 10 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 33397
Number of edges: 1159203

Running Louvain algorithm with multilevel refinement...
Maximum modularity in 10 random starts: 0.8702
Number of communities: 32
Elapsed time: 9 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 33397
Number of edges: 1159203

Running Louvain algorithm with multilevel refinement...
Maximum modularity in 10 random starts: 0.8650
Number of communities: 32
Elapsed time: 9 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 33397
Number of edges: 1159203

Running Louvain algorithm with multilevel refinement...
Maximum modularity in 10 random starts: 0.8602
Number of communities: 34
Elapsed time: 8 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 33397
Number of edges: 1159203

Running Louvain algorithm with multilevel refinement...
Maximum modularity in 10 random starts: 0.8554
Number of communities: 35
Elapsed time: 8 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 33397
Number of edges: 1159203

Running Louvain algorithm with multilevel refinement...
Maximum modularity in 10 random starts: 0.8511
Number of communities: 36
Elapsed time: 9 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 33397
Number of edges: 1159203

Running Louvain algorithm with multilevel refinement...
Maximum modularity in 10 random starts: 0.8467
Number of communities: 40
Elapsed time: 9 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 33397
Number of edges: 1159203

Running Louvain algorithm with multilevel refinement...
Maximum modularity in 10 random starts: 0.8425
Number of communities: 40
Elapsed time: 9 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 33397
Number of edges: 1159203

Running Louvain algorithm with multilevel refinement...
Maximum modularity in 10 random starts: 0.8385
Number of communities: 40
Elapsed time: 8 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 33397
Number of edges: 1159203

Running Louvain algorithm with multilevel refinement...
Maximum modularity in 10 random starts: 0.8345
Number of communities: 40
Elapsed time: 9 seconds</code></pre>
</div>
<div class="sourceCode cell-code" id="cb71"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb71-1"><a href="#cb71-1" aria-hidden="true" tabindex="-1"></a>merged.<span class="fl">18279.</span>skin.singlets <span class="ot">&lt;-</span> <span class="fu">RunUMAP</span>(merged.<span class="fl">18279.</span>skin.singlets,</span>
<span id="cb71-2"><a href="#cb71-2" aria-hidden="true" tabindex="-1"></a>                        <span class="at">reduction =</span> <span class="st">"integrated.harmony"</span>,</span>
<span id="cb71-3"><a href="#cb71-3" aria-hidden="true" tabindex="-1"></a>                        <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">50</span>,</span>
<span id="cb71-4"><a href="#cb71-4" aria-hidden="true" tabindex="-1"></a>                        <span class="at">reduction.name =</span> <span class="st">"umap.harmony"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>15:45:38 UMAP embedding parameters a = 0.9922 b = 1.112</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "dist" in cache; using the first, from namespace 'BiocGenerics'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'spam'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>15:45:38 Read 33397 rows and found 50 numeric columns</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>15:45:38 Using Annoy for neighbor search, n_neighbors = 30</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "dist" in cache; using the first, from namespace 'BiocGenerics'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'spam'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>15:45:38 Building Annoy index with metric = cosine, n_trees = 50</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>0%   10   20   30   40   50   60   70   80   90   100%</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>[----|----|----|----|----|----|----|----|----|----|</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>**************************************************|
15:45:49 Writing NN index file to temp file /tmp/Rtmp5cppMq/file39fc203f045530
15:45:49 Searching Annoy index using 1 thread, search_k = 3000
15:46:03 Annoy recall = 100%
15:46:04 Commencing smooth kNN distance calibration using 1 thread with target n_neighbors = 30
15:46:08 Initializing from normalized Laplacian + noise (using RSpectra)
15:46:20 Commencing optimization for 200 epochs, with 1459272 positive edges
15:46:45 Optimization finished</code></pre>
</div>
<div class="sourceCode cell-code" id="cb83"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb83-1"><a href="#cb83-1" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(merged.<span class="fl">18279.</span>skin.singlets<span class="sc">$</span>seurat_clusters)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
   0    1    2    3    4    5    6    7    8    9   10   11   12   13   14   15 
3299 2474 1763 1626 1613 1538 1465 1421 1372 1329 1288 1210 1082 1036 1032  932 
  16   17   18   19   20   21   22   23   24   25   26   27   28   29   30   31 
 920  874  821  562  557  543  534  509  484  408  360  308  280  262  252  241 
  32   33   34   35   36   37   38   39 
 180  174  168  167  116   87   77   33 </code></pre>
</div>
</div>
</section>
<section id="plot-clusters" class="level1" data-number="4">
<h1 data-number="4"><span class="header-section-number">4</span> Plot clusters</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb85"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb85-1"><a href="#cb85-1" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(merged.<span class="fl">18279.</span>skin.singlets,</span>
<span id="cb85-2"><a href="#cb85-2" aria-hidden="true" tabindex="-1"></a>        <span class="at">reduction =</span> <span class="st">"umap.harmony"</span>, </span>
<span id="cb85-3"><a href="#cb85-3" aria-hidden="true" tabindex="-1"></a>        <span class="at">label =</span> <span class="cn">TRUE</span>, </span>
<span id="cb85-4"><a href="#cb85-4" aria-hidden="true" tabindex="-1"></a>        <span class="at">group.by =</span> <span class="fu">paste0</span>(<span class="st">"RNA_snn_res."</span>,<span class="fu">seq</span>(<span class="fl">0.1</span>,<span class="dv">2</span>,<span class="fl">0.1</span>)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Skin_scRNA_Part2_files/figure-html/unnamed-chunk-19-1.png" class="img-fluid figure-img" width="3840"></p>
</figure>
</div>
</div>
</div>
<section id="plot-one-as-example" class="level2" data-number="4.1">
<h2 data-number="4.1" class="anchored" data-anchor-id="plot-one-as-example"><span class="header-section-number">4.1</span> Plot one as example</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb86"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb86-1"><a href="#cb86-1" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(merged.<span class="fl">18279.</span>skin.singlets,</span>
<span id="cb86-2"><a href="#cb86-2" aria-hidden="true" tabindex="-1"></a>        <span class="at">reduction =</span> <span class="st">"umap.harmony"</span>, </span>
<span id="cb86-3"><a href="#cb86-3" aria-hidden="true" tabindex="-1"></a>        <span class="at">label =</span> <span class="cn">TRUE</span>, </span>
<span id="cb86-4"><a href="#cb86-4" aria-hidden="true" tabindex="-1"></a>        <span class="at">group.by =</span> <span class="st">"RNA_snn_res.1"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Skin_scRNA_Part2_files/figure-html/unnamed-chunk-20-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="plot-metadata-in-umap-space" class="level2" data-number="4.2">
<h2 data-number="4.2" class="anchored" data-anchor-id="plot-metadata-in-umap-space"><span class="header-section-number">4.2</span> Plot metadata in UMAP space</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb87"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb87-1"><a href="#cb87-1" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(merged.<span class="fl">18279.</span>skin.singlets,<span class="at">reduction =</span> <span class="st">"umap.harmony"</span>, <span class="at">group.by =</span> <span class="st">"Patient"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Skin_scRNA_Part2_files/figure-html/unnamed-chunk-21-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb88"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb88-1"><a href="#cb88-1" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(merged.<span class="fl">18279.</span>skin.singlets,<span class="at">reduction =</span> <span class="st">"umap.harmony"</span>, <span class="at">group.by =</span> <span class="st">"Site"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Skin_scRNA_Part2_files/figure-html/unnamed-chunk-21-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb89"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb89-1"><a href="#cb89-1" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(merged.<span class="fl">18279.</span>skin.singlets,<span class="at">reduction =</span> <span class="st">"umap.harmony"</span>, <span class="at">group.by =</span> <span class="st">"Timepoint"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Skin_scRNA_Part2_files/figure-html/unnamed-chunk-21-3.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb90"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb90-1"><a href="#cb90-1" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(merged.<span class="fl">18279.</span>skin.singlets,<span class="at">reduction =</span> <span class="st">"umap.harmony"</span>, <span class="at">group.by =</span> <span class="st">"IpiCohort"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Skin_scRNA_Part2_files/figure-html/unnamed-chunk-21-4.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb91"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb91-1"><a href="#cb91-1" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(merged.<span class="fl">18279.</span>skin.singlets,<span class="at">reduction =</span> <span class="st">"umap.harmony"</span>, <span class="at">group.by =</span> <span class="st">"Sample"</span>) <span class="sc">+</span> <span class="fu">NoLegend</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Skin_scRNA_Part2_files/figure-html/unnamed-chunk-21-5.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb92"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb92-1"><a href="#cb92-1" aria-hidden="true" tabindex="-1"></a><span class="fu">FeaturePlot</span>(merged.<span class="fl">18279.</span>skin.singlets,<span class="at">reduction =</span> <span class="st">"umap.harmony"</span>,<span class="at">features=</span><span class="st">"nCount_RNA"</span>,<span class="at">order=</span>T)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Skin_scRNA_Part2_files/figure-html/unnamed-chunk-21-6.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb93"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb93-1"><a href="#cb93-1" aria-hidden="true" tabindex="-1"></a><span class="fu">FeaturePlot</span>(merged.<span class="fl">18279.</span>skin.singlets,<span class="at">reduction =</span> <span class="st">"umap.harmony"</span>,<span class="at">features=</span><span class="st">"nFeature_RNA"</span>,<span class="at">order=</span>T)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Skin_scRNA_Part2_files/figure-html/unnamed-chunk-21-7.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb94"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb94-1"><a href="#cb94-1" aria-hidden="true" tabindex="-1"></a><span class="fu">FeaturePlot</span>(merged.<span class="fl">18279.</span>skin.singlets,<span class="at">reduction =</span> <span class="st">"umap.harmony"</span>,<span class="at">features=</span><span class="st">"percent.mt"</span>,<span class="at">order=</span>T)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Skin_scRNA_Part2_files/figure-html/unnamed-chunk-21-8.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="save-clustered-object" class="level2" data-number="4.3">
<h2 data-number="4.3" class="anchored" data-anchor-id="save-clustered-object"><span class="header-section-number">4.3</span> Save clustered object</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb95"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb95-1"><a href="#cb95-1" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(merged.<span class="fl">18279.</span>skin.singlets,<span class="st">"Skin_scRNA_Part2.rds"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="get-session-info" class="level2" data-number="4.4">
<h2 data-number="4.4" class="anchored" data-anchor-id="get-session-info"><span class="header-section-number">4.4</span> Get session info</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb96"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb96-1"><a href="#cb96-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>R version 4.3.1 (2023-06-16)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Rocky Linux 8.10 (Green Obsidian)

Matrix products: default
BLAS/LAPACK: /usr/lib64/libopenblasp-r0.3.15.so;  LAPACK version 3.9.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

time zone: America/New_York
tzcode source: system (glibc)

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] cellXY_0.99.0               scDblFinder_1.14.0         
 [3] harmony_1.2.0               alevinQC_1.16.1            
 [5] vctrs_0.6.5                 patchwork_1.3.0            
 [7] scater_1.28.0               scuttle_1.10.3             
 [9] speckle_1.0.0               Matrix_1.6-4               
[11] fishpond_2.6.2              readxl_1.4.3               
[13] SingleCellExperiment_1.22.0 SummarizedExperiment_1.30.2
[15] Biobase_2.60.0              GenomicRanges_1.52.1       
[17] GenomeInfoDb_1.36.4         IRanges_2.34.1             
[19] S4Vectors_0.38.2            BiocGenerics_0.46.0        
[21] MatrixGenerics_1.12.3       matrixStats_1.2.0          
[23] flexmix_2.3-19              lattice_0.22-5             
[25] SeuratWrappers_0.3.19       miQC_1.8.0                 
[27] lubridate_1.9.3             forcats_1.0.0              
[29] stringr_1.5.1               dplyr_1.1.4                
[31] purrr_1.0.2                 readr_2.1.5                
[33] tidyr_1.3.1                 tibble_3.2.1               
[35] ggplot2_3.4.4               tidyverse_2.0.0            
[37] Seurat_5.1.0                SeuratObject_5.0.2         
[39] sp_2.1-3                    sctransform_0.4.1          
[41] glmGamPoi_1.12.2            BiocParallel_1.34.2        
[43] presto_1.0.0                Rcpp_1.0.12                
[45] devtools_2.4.5              usethis_2.2.2              
[47] data.table_1.15.0          

loaded via a namespace (and not attached):
  [1] fs_1.6.3                  spatstat.sparse_3.0-3    
  [3] bitops_1.0-7              httr_1.4.7               
  [5] RColorBrewer_1.1-3        profvis_0.3.8            
  [7] tools_4.3.1               utf8_1.2.4               
  [9] R6_2.5.1                  DT_0.31                  
 [11] lazyeval_0.2.2            uwot_0.1.16              
 [13] urlchecker_1.0.1          withr_3.0.0              
 [15] GGally_2.2.1              gridExtra_2.3            
 [17] progressr_0.14.0          cli_3.6.2                
 [19] spatstat.explore_3.2-6    fastDummies_1.7.3        
 [21] labeling_0.4.3            spatstat.data_3.0-4      
 [23] ggridges_0.5.6            pbapply_1.7-2            
 [25] Rsamtools_2.16.0          R.utils_2.12.3           
 [27] parallelly_1.37.0         sessioninfo_1.2.2        
 [29] limma_3.56.2              RSQLite_2.3.5            
 [31] BiocIO_1.10.0             generics_0.1.3           
 [33] gtools_3.9.5              ica_1.0-3                
 [35] spatstat.random_3.2-2     ggbeeswarm_0.7.2         
 [37] fansi_1.0.6               abind_1.4-5              
 [39] R.methodsS3_1.8.2         lifecycle_1.0.4          
 [41] yaml_2.3.8                edgeR_3.42.4             
 [43] Rtsne_0.17                blob_1.2.4               
 [45] grid_4.3.1                dqrng_0.3.2              
 [47] promises_1.2.1            crayon_1.5.2             
 [49] shinydashboard_0.7.2      miniUI_0.1.1.1           
 [51] beachmat_2.16.0           cowplot_1.1.3            
 [53] KEGGREST_1.40.1           metapod_1.8.0            
 [55] pillar_1.9.0              knitr_1.45               
 [57] rjson_0.2.21              xgboost_1.7.7.1          
 [59] future.apply_1.11.1       codetools_0.2-19         
 [61] leiden_0.4.3.1            glue_1.7.0               
 [63] remotes_2.4.2.1           png_0.1-8                
 [65] spam_2.10-0               org.Mm.eg.db_3.18.0      
 [67] cellranger_1.1.0          gtable_0.3.4             
 [69] cachem_1.0.8              xfun_0.42                
 [71] S4Arrays_1.2.0            mime_0.12                
 [73] survival_3.5-8            statmod_1.5.0            
 [75] bluster_1.10.0            ellipsis_0.3.2           
 [77] fitdistrplus_1.1-11       ROCR_1.0-11              
 [79] nlme_3.1-164              bit64_4.0.5              
 [81] RcppAnnoy_0.0.22          irlba_2.3.5.1            
 [83] vipor_0.4.7               KernSmooth_2.23-22       
 [85] DBI_1.2.2                 colorspace_2.1-0         
 [87] nnet_7.3-19               tidyselect_1.2.0         
 [89] bit_4.0.5                 compiler_4.3.1           
 [91] BiocNeighbors_1.18.0      DelayedArray_0.26.7      
 [93] plotly_4.10.4             rtracklayer_1.60.1       
 [95] scales_1.3.0              lmtest_0.9-40            
 [97] digest_0.6.34             goftest_1.2-3            
 [99] spatstat.utils_3.0-4      rmarkdown_2.25           
[101] RhpcBLASctl_0.23-42       XVector_0.40.0           
[103] htmltools_0.5.7           pkgconfig_2.0.3          
[105] sparseMatrixStats_1.12.2  fastmap_1.1.1            
[107] rlang_1.1.3               htmlwidgets_1.6.4        
[109] shiny_1.8.0               DelayedMatrixStats_1.22.6
[111] farver_2.1.1              zoo_1.8-12               
[113] jsonlite_1.8.8            R.oo_1.26.0              
[115] BiocSingular_1.16.0       RCurl_1.98-1.14          
[117] magrittr_2.0.3            modeltools_0.2-23        
[119] GenomeInfoDbData_1.2.10   dotCall64_1.1-1          
[121] munsell_0.5.0             viridis_0.6.5            
[123] reticulate_1.35.0         stringi_1.8.3            
[125] zlibbioc_1.46.0           MASS_7.3-60.0.1          
[127] org.Hs.eg.db_3.18.0       plyr_1.8.9               
[129] pkgbuild_1.4.3            ggstats_0.5.1            
[131] parallel_4.3.1            listenv_0.9.1            
[133] ggrepel_0.9.5             deldir_2.0-2             
[135] Biostrings_2.68.1         splines_4.3.1            
[137] tensor_1.5                hms_1.1.3                
[139] locfit_1.5-9.8            igraph_2.0.2             
[141] spatstat.geom_3.2-8       RcppHNSW_0.6.0           
[143] reshape2_1.4.4            ScaledMatrix_1.8.1       
[145] pkgload_1.3.4             XML_3.99-0.16.1          
[147] evaluate_0.23             scran_1.28.2             
[149] BiocManager_1.30.22       tzdb_0.4.0               
[151] httpuv_1.6.14             RANN_2.6.1               
[153] polyclip_1.10-6           future_1.33.1            
[155] scattermore_1.2           rsvd_1.0.5               
[157] xtable_1.8-4              restfulr_0.0.15          
[159] svMisc_1.2.3              RSpectra_0.16-1          
[161] later_1.3.2               viridisLite_0.4.2        
[163] AnnotationDbi_1.64.1      GenomicAlignments_1.36.0 
[165] memoise_2.0.1             beeswarm_0.4.0           
[167] tximport_1.28.0           cluster_2.1.6            
[169] timechange_0.3.0          globals_0.16.2           </code></pre>
</div>
</div>


</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "î§‹";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="./Skin_scRNA_Part1.html" class="pagination-link" aria-label="Skin: QC &amp; Filtering">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Skin: QC &amp; Filtering</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="./Skin_scRNA_Part3.html" class="pagination-link" aria-label="Skin: Check patient sexes">
        <span class="nav-page-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Skin: Check patient sexes</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->




</body></html>