<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.6.39">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>11&nbsp; Skin: Calculating Post vs Pre 3rd vaccination differences in pathway enrichments with fgsea â€“ Blass &amp; Keskin (2025): Melanoma NeoVax analysis: Skin single-cell RNAseq</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./Skin_scRNA_Part10.html" rel="prev">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-e26003cea8cd680ca0c55a263523d882.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap-41a4eb7d26fa9333a8f88b12b94df59e.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


</head>

<body class="nav-sidebar floating">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./Skin_scRNA_Part11.html"><span class="chapter-number">11</span>&nbsp; <span class="chapter-title">Skin: Calculating Post vs Pre 3rd vaccination differences in pathway enrichments with fgsea</span></a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="./">Blass &amp; Keskin (2025): Melanoma NeoVax analysis: Skin single-cell RNAseq</a> 
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Introduction</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Skin_scRNA_Part1.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Skin: QC &amp; Filtering</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Skin_scRNA_Part2.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Skin: Normalization, doublet discrimination, integration, clustering</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Skin_scRNA_Part3.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Skin: Check patient sexes</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Skin_scRNA_Part4.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Skin: Post-clustering QC and characterization</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Skin_scRNA_Part5.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">Skin: Post-clustering comprehensive marker identification</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Skin_scRNA_Part6.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">Skin: Sub-clustering on T/NK lineage</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Skin_scRNA_Part7.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">Skin: Assisting cell type annotation with enrichR</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Skin_scRNA_Part8.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">8</span>&nbsp; <span class="chapter-title">Skin: Classification of CD4 and CD8 T cell states within T/NK lineage</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Skin_scRNA_Part9.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">9</span>&nbsp; <span class="chapter-title">Skin: Differential expression analysis, Post vs Pre 3rd vaccination</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Skin_scRNA_Part10.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">10</span>&nbsp; <span class="chapter-title">Skin: Deep phenotyping, cluster annotation, and exploration of Post vs Pre 3rd vaccination differences in cell proportions and expression</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Skin_scRNA_Part11.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text"><span class="chapter-number">11</span>&nbsp; <span class="chapter-title">Skin: Calculating Post vs Pre 3rd vaccination differences in pathway enrichments with fgsea</span></span></a>
  </div>
</li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#set-up-seurat-workspace" id="toc-set-up-seurat-workspace" class="nav-link active" data-scroll-target="#set-up-seurat-workspace"><span class="header-section-number">11.1</span> Set up Seurat workspace</a></li>
  <li><a href="#read-in-pseudobulk-de-test-results" id="toc-read-in-pseudobulk-de-test-results" class="nav-link" data-scroll-target="#read-in-pseudobulk-de-test-results"><span class="header-section-number">11.2</span> Read in pseudobulk DE test results</a></li>
  <li><a href="#set-cluster-annotations" id="toc-set-cluster-annotations" class="nav-link" data-scroll-target="#set-cluster-annotations"><span class="header-section-number">11.3</span> Set cluster annotations</a></li>
  <li><a href="#set-cluster-orders-for-plotting" id="toc-set-cluster-orders-for-plotting" class="nav-link" data-scroll-target="#set-cluster-orders-for-plotting"><span class="header-section-number">11.4</span> Set cluster orders for plotting</a></li>
  <li><a href="#split-results-by-cluster-then-extract-ranks-by-log2-fold-change-post3rd-vs-pre3rd" id="toc-split-results-by-cluster-then-extract-ranks-by-log2-fold-change-post3rd-vs-pre3rd" class="nav-link" data-scroll-target="#split-results-by-cluster-then-extract-ranks-by-log2-fold-change-post3rd-vs-pre3rd"><span class="header-section-number">11.5</span> Split results by cluster, then extract ranks by log2 fold-change Post3rd vs Pre3rd</a></li>
  <li><a href="#prepare-gene-sets-from-msigdb" id="toc-prepare-gene-sets-from-msigdb" class="nav-link" data-scroll-target="#prepare-gene-sets-from-msigdb"><span class="header-section-number">11.6</span> Prepare gene sets from MSigDB</a></li>
  <li><a href="#iterate-over-de-results-and-run-fgsea-on-each-clusters-ranked-gene-list" id="toc-iterate-over-de-results-and-run-fgsea-on-each-clusters-ranked-gene-list" class="nav-link" data-scroll-target="#iterate-over-de-results-and-run-fgsea-on-each-clusters-ranked-gene-list"><span class="header-section-number">11.7</span> Iterate over DE results and run fgsea on each clusterâ€™s ranked gene list</a></li>
  <li><a href="#collapse-all-results-into-one-tibble-and-filter-for-significance" id="toc-collapse-all-results-into-one-tibble-and-filter-for-significance" class="nav-link" data-scroll-target="#collapse-all-results-into-one-tibble-and-filter-for-significance"><span class="header-section-number">11.8</span> Collapse all results into one tibble and filter for significance</a></li>
  <li><a href="#print-selection-of-top-most-significant-pathways" id="toc-print-selection-of-top-most-significant-pathways" class="nav-link" data-scroll-target="#print-selection-of-top-most-significant-pathways"><span class="header-section-number">11.9</span> Print selection of top-most significant pathways</a></li>
  <li><a href="#print-selection-of-recurring-top-most-significant-pathways-in-multiple-clusters" id="toc-print-selection-of-recurring-top-most-significant-pathways-in-multiple-clusters" class="nav-link" data-scroll-target="#print-selection-of-recurring-top-most-significant-pathways-in-multiple-clusters"><span class="header-section-number">11.10</span> Print selection of recurring top-most significant pathways in multiple clusters</a></li>
  <li><a href="#filter-to-specific-pathways-related-to-innate-immunity-of-interest" id="toc-filter-to-specific-pathways-related-to-innate-immunity-of-interest" class="nav-link" data-scroll-target="#filter-to-specific-pathways-related-to-innate-immunity-of-interest"><span class="header-section-number">11.11</span> Filter to specific pathways related to innate immunity of interest</a></li>
  <li><a href="#plot-dot-plot-of-nes-and-p-value-for-pathways-of-interest" id="toc-plot-dot-plot-of-nes-and-p-value-for-pathways-of-interest" class="nav-link" data-scroll-target="#plot-dot-plot-of-nes-and-p-value-for-pathways-of-interest"><span class="header-section-number">11.12</span> Plot dot plot of NES and p-value for pathways of interest</a></li>
  <li><a href="#get-session-info" id="toc-get-session-info" class="nav-link" data-scroll-target="#get-session-info"><span class="header-section-number">11.13</span> Get session info</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title"><span id="sec-skin-gsea" class="quarto-section-identifier"><span class="chapter-number">11</span>&nbsp; <span class="chapter-title">Skin: Calculating Post vs Pre 3rd vaccination differences in pathway enrichments with fgsea</span></span></h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<section id="set-up-seurat-workspace" class="level2" data-number="11.1">
<h2 data-number="11.1" class="anchored" data-anchor-id="set-up-seurat-workspace"><span class="header-section-number">11.1</span> Set up Seurat workspace</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load libraries</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Seurat)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(msigdbr)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(fgsea)</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(GSEAtraining)       <span class="co"># This is used because KEGG_MEDICUS isn't yet available in msigdbr</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="read-in-pseudobulk-de-test-results" class="level2" data-number="11.2">
<h2 data-number="11.2" class="anchored" data-anchor-id="read-in-pseudobulk-de-test-results"><span class="header-section-number">11.2</span> Read in pseudobulk DE test results</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>skin_de <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">"Skin_scRNA_PostVsPre3rd_DiffExp_results.tsv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 125307 Columns: 11
â”€â”€ Column specification â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€
Delimiter: "\t"
chr (4): sub.cluster, gene, cluster_id, contrast
dbl (7): baseMean, logFC, lfcSE, stat, p_val, p_adj.loc, p_adj.glb

â„¹ Use `spec()` to retrieve the full column specification for this data.
â„¹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>skin_de</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 125,307 Ã— 11
   sub.cluster gene   cluster_id baseMean   logFC lfcSE   stat   p_val p_adj.loc
   &lt;chr&gt;       &lt;chr&gt;  &lt;chr&gt;         &lt;dbl&gt;   &lt;dbl&gt; &lt;dbl&gt;  &lt;dbl&gt;   &lt;dbl&gt;     &lt;dbl&gt;
 1 0           7SK    0            166.   -2.09   0.470 -4.44  9.12e-6  0.000168
 2 0           A2M    0             84.8  -3.11   1.07  -2.90  3.75e-3  0.0215  
 3 0           AAGAB  0             21.1   0.113  0.282  0.400 6.89e-1  0.817   
 4 0           AAK1   0            105.   -0.111  0.154 -0.724 4.69e-1  0.650   
 5 0           AAMDC  0             22.9   0.303  0.280  1.08  2.79e-1  0.471   
 6 0           AAMP   0             50.9   0.0513 0.219  0.234 8.15e-1  0.897   
 7 0           AAR2   0              9.66  0.0439 0.430  0.102 9.19e-1  0.959   
 8 0           AARS1  0             12.1  -0.160  0.327 -0.489 6.25e-1  0.773   
 9 0           AASDHâ€¦ 0             14.8   0.495  0.315  1.57  1.16e-1  0.262   
10 0           AATF   0             50.2   0.489  0.233  2.10  3.58e-2  0.116   
# â„¹ 125,297 more rows
# â„¹ 2 more variables: p_adj.glb &lt;dbl&gt;, contrast &lt;chr&gt;</code></pre>
</div>
</div>
</section>
<section id="set-cluster-annotations" class="level2" data-number="11.3">
<h2 data-number="11.3" class="anchored" data-anchor-id="set-cluster-annotations"><span class="header-section-number">11.3</span> Set cluster annotations</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>cluster_annot <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>        <span class="st">"0"</span> <span class="ot">=</span> <span class="st">"MonoMac_1"</span>,</span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>        <span class="st">"1_0"</span> <span class="ot">=</span> <span class="st">"CD4_T_Naive/CM"</span>,</span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>        <span class="st">"1_1"</span> <span class="ot">=</span> <span class="st">"CD4_T_CTL"</span>,</span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>        <span class="st">"2_0"</span> <span class="ot">=</span> <span class="st">"CD8_T_EM_1"</span>,</span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a>        <span class="st">"2_1"</span> <span class="ot">=</span> <span class="st">"CD8_T_Naive/CM_1"</span>,</span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a>        <span class="st">"2_2"</span> <span class="ot">=</span> <span class="st">"CD8_T_EM_2"</span>,</span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a>        <span class="st">"3"</span> <span class="ot">=</span> <span class="st">"MonoMac_2"</span>,</span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a>        <span class="st">"4_0"</span> <span class="ot">=</span> <span class="st">"CD8_T_Naive/CM_2"</span>,</span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a>        <span class="st">"4_1"</span> <span class="ot">=</span> <span class="st">"CD8_T_Naive/CM_3"</span>,</span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true" tabindex="-1"></a>        <span class="st">"5_0"</span> <span class="ot">=</span> <span class="st">"NK/T_1"</span>,</span>
<span id="cb6-12"><a href="#cb6-12" aria-hidden="true" tabindex="-1"></a>        <span class="st">"5_1"</span> <span class="ot">=</span> <span class="st">"NK/T_2"</span>,</span>
<span id="cb6-13"><a href="#cb6-13" aria-hidden="true" tabindex="-1"></a>        <span class="st">"5_2"</span> <span class="ot">=</span> <span class="st">"NK_cells"</span>,</span>
<span id="cb6-14"><a href="#cb6-14" aria-hidden="true" tabindex="-1"></a>        <span class="st">"6_0"</span> <span class="ot">=</span> <span class="st">"T_unknown_1"</span>,</span>
<span id="cb6-15"><a href="#cb6-15" aria-hidden="true" tabindex="-1"></a>        <span class="st">"6_1"</span> <span class="ot">=</span> <span class="st">"CD8_T_EM_3"</span>,</span>
<span id="cb6-16"><a href="#cb6-16" aria-hidden="true" tabindex="-1"></a>        <span class="st">"7"</span> <span class="ot">=</span> <span class="st">"cDC2"</span>,</span>
<span id="cb6-17"><a href="#cb6-17" aria-hidden="true" tabindex="-1"></a>        <span class="st">"8"</span> <span class="ot">=</span> <span class="st">"Keratinocytes"</span>,</span>
<span id="cb6-18"><a href="#cb6-18" aria-hidden="true" tabindex="-1"></a>        <span class="st">"9_0"</span> <span class="ot">=</span> <span class="st">"Treg_1"</span>,</span>
<span id="cb6-19"><a href="#cb6-19" aria-hidden="true" tabindex="-1"></a>        <span class="st">"9_1"</span> <span class="ot">=</span> <span class="st">"Treg_2"</span>,</span>
<span id="cb6-20"><a href="#cb6-20" aria-hidden="true" tabindex="-1"></a>        <span class="st">"10"</span> <span class="ot">=</span> <span class="st">"Fibroblasts"</span>,</span>
<span id="cb6-21"><a href="#cb6-21" aria-hidden="true" tabindex="-1"></a>        <span class="st">"11_0"</span> <span class="ot">=</span> <span class="st">"T_unknown_2"</span>,</span>
<span id="cb6-22"><a href="#cb6-22" aria-hidden="true" tabindex="-1"></a>        <span class="st">"11_1"</span> <span class="ot">=</span> <span class="st">"T_unknown_3"</span>,</span>
<span id="cb6-23"><a href="#cb6-23" aria-hidden="true" tabindex="-1"></a>        <span class="st">"12"</span> <span class="ot">=</span> <span class="st">"MonoMac_3"</span>,</span>
<span id="cb6-24"><a href="#cb6-24" aria-hidden="true" tabindex="-1"></a>        <span class="st">"13_0"</span> <span class="ot">=</span> <span class="st">"Proliferating_T_cells_1"</span>,</span>
<span id="cb6-25"><a href="#cb6-25" aria-hidden="true" tabindex="-1"></a>        <span class="st">"13_1"</span> <span class="ot">=</span> <span class="st">"Proliferating_T_cells_2"</span>,</span>
<span id="cb6-26"><a href="#cb6-26" aria-hidden="true" tabindex="-1"></a>        <span class="st">"14"</span> <span class="ot">=</span> <span class="st">"pDC"</span>,</span>
<span id="cb6-27"><a href="#cb6-27" aria-hidden="true" tabindex="-1"></a>        <span class="st">"15"</span> <span class="ot">=</span> <span class="st">"DC_LAMP3"</span>,</span>
<span id="cb6-28"><a href="#cb6-28" aria-hidden="true" tabindex="-1"></a>        <span class="st">"16"</span> <span class="ot">=</span> <span class="st">"Mast"</span>,</span>
<span id="cb6-29"><a href="#cb6-29" aria-hidden="true" tabindex="-1"></a>        <span class="st">"17"</span> <span class="ot">=</span> <span class="st">"Endothelial"</span>,</span>
<span id="cb6-30"><a href="#cb6-30" aria-hidden="true" tabindex="-1"></a>        <span class="st">"18"</span> <span class="ot">=</span> <span class="st">"cDC1"</span>,</span>
<span id="cb6-31"><a href="#cb6-31" aria-hidden="true" tabindex="-1"></a>        <span class="st">"19_0"</span> <span class="ot">=</span> <span class="st">"NK/T_3"</span>,</span>
<span id="cb6-32"><a href="#cb6-32" aria-hidden="true" tabindex="-1"></a>        <span class="st">"20"</span> <span class="ot">=</span> <span class="st">"Pericytes/smooth_muscle"</span>,</span>
<span id="cb6-33"><a href="#cb6-33" aria-hidden="true" tabindex="-1"></a>        <span class="st">"21"</span> <span class="ot">=</span> <span class="st">"Basal_1"</span>,</span>
<span id="cb6-34"><a href="#cb6-34" aria-hidden="true" tabindex="-1"></a>        <span class="st">"22"</span> <span class="ot">=</span> <span class="st">"MonoMac_4"</span>,</span>
<span id="cb6-35"><a href="#cb6-35" aria-hidden="true" tabindex="-1"></a>        <span class="st">"23"</span> <span class="ot">=</span> <span class="st">"B_cells"</span>,</span>
<span id="cb6-36"><a href="#cb6-36" aria-hidden="true" tabindex="-1"></a>        <span class="st">"24"</span> <span class="ot">=</span> <span class="st">"Basal_2"</span>,</span>
<span id="cb6-37"><a href="#cb6-37" aria-hidden="true" tabindex="-1"></a>        <span class="st">"25"</span> <span class="ot">=</span> <span class="st">"Macrophages"</span>,</span>
<span id="cb6-38"><a href="#cb6-38" aria-hidden="true" tabindex="-1"></a>        <span class="st">"26"</span> <span class="ot">=</span> <span class="st">"Proliferating_cDC2"</span></span>
<span id="cb6-39"><a href="#cb6-39" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="set-cluster-orders-for-plotting" class="level2" data-number="11.4">
<h2 data-number="11.4" class="anchored" data-anchor-id="set-cluster-orders-for-plotting"><span class="header-section-number">11.4</span> Set cluster orders for plotting</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>cluster_order <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"0"</span>,<span class="st">"3"</span>,<span class="st">"12"</span>,<span class="st">"22"</span>,<span class="st">"25"</span>,<span class="st">"14"</span>,<span class="st">"18"</span>,<span class="st">"7"</span>,<span class="st">"26"</span>,<span class="st">"15"</span>,<span class="st">"23"</span>,<span class="st">"5_2"</span>,<span class="st">"5_0"</span>,<span class="st">"5_1"</span>,<span class="st">"19_0"</span>,<span class="st">"2_1"</span>,<span class="st">"4_0"</span>,<span class="st">"4_1"</span>,<span class="st">"2_0"</span>,<span class="st">"2_2"</span>,<span class="st">"6_1"</span>,<span class="st">"1_0"</span>,<span class="st">"1_1"</span>,<span class="st">"9_0"</span>,<span class="st">"9_1"</span>,<span class="st">"13_0"</span>,<span class="st">"13_1"</span>,<span class="st">"6_0"</span>,<span class="st">"11_0"</span>,<span class="st">"11_1"</span>,<span class="st">"16"</span>,<span class="st">"8"</span>,<span class="st">"10"</span>,<span class="st">"17"</span>,<span class="st">"20"</span>,<span class="st">"21"</span>,<span class="st">"24"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="split-results-by-cluster-then-extract-ranks-by-log2-fold-change-post3rd-vs-pre3rd" class="level2" data-number="11.5">
<h2 data-number="11.5" class="anchored" data-anchor-id="split-results-by-cluster-then-extract-ranks-by-log2-fold-change-post3rd-vs-pre3rd"><span class="header-section-number">11.5</span> Split results by cluster, then extract ranks by log2 fold-change Post3rd vs Pre3rd</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>skin_de_split <span class="ot">&lt;-</span> skin_de <span class="sc">%&gt;%</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(sub.cluster) <span class="sc">%&gt;%</span></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_split</span>()</span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>skin_de_split_ranks <span class="ot">&lt;-</span> <span class="fu">lapply</span>(skin_de_split, </span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>                                <span class="at">FUN =</span> <span class="cf">function</span>(x) { x <span class="sc">%&gt;%</span></span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a>                                    dplyr<span class="sc">::</span><span class="fu">select</span>(gene,logFC) <span class="sc">%&gt;%</span></span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a>                                    dplyr<span class="sc">::</span><span class="fu">arrange</span>(logFC) <span class="sc">%&gt;%</span></span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">deframe</span>()</span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a>                                    }</span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true" tabindex="-1"></a>                        )</span>
<span id="cb8-12"><a href="#cb8-12" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(skin_de_split_ranks) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"Cluster"</span>, </span>
<span id="cb8-13"><a href="#cb8-13" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">sapply</span>(skin_de_split, </span>
<span id="cb8-14"><a href="#cb8-14" aria-hidden="true" tabindex="-1"></a>                                <span class="at">FUN =</span> <span class="cf">function</span>(x) { x <span class="sc">%&gt;%</span> </span>
<span id="cb8-15"><a href="#cb8-15" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">pull</span>(sub.cluster) <span class="sc">%&gt;%</span></span>
<span id="cb8-16"><a href="#cb8-16" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">unique</span>()</span>
<span id="cb8-17"><a href="#cb8-17" aria-hidden="true" tabindex="-1"></a>                                    }</span>
<span id="cb8-18"><a href="#cb8-18" aria-hidden="true" tabindex="-1"></a>                                )</span>
<span id="cb8-19"><a href="#cb8-19" aria-hidden="true" tabindex="-1"></a>                            )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="prepare-gene-sets-from-msigdb" class="level2" data-number="11.6">
<h2 data-number="11.6" class="anchored" data-anchor-id="prepare-gene-sets-from-msigdb"><span class="header-section-number">11.6</span> Prepare gene sets from MSigDB</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>c2 <span class="ot">&lt;-</span> msigdbr<span class="sc">::</span><span class="fu">msigdbr</span>(<span class="at">species=</span><span class="st">"Homo sapiens"</span>,</span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>        <span class="at">category =</span> <span class="st">"C2"</span>)</span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>immunesig <span class="ot">&lt;-</span> msigdbr<span class="sc">::</span><span class="fu">msigdbr</span>(<span class="at">species=</span><span class="st">"Homo sapiens"</span>,</span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a>        <span class="at">category =</span> <span class="st">"C7"</span>, </span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a>        <span class="at">subcategory =</span> <span class="st">"IMMUNESIGDB"</span>)</span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a>hallmark <span class="ot">&lt;-</span> msigdbr<span class="sc">::</span><span class="fu">msigdbr</span>(<span class="at">species=</span><span class="st">"Homo sapiens"</span>,</span>
<span id="cb9-9"><a href="#cb9-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">category =</span> <span class="st">"H"</span>)                         </span>
<span id="cb9-10"><a href="#cb9-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-11"><a href="#cb9-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Get updated KEGG MEDICUS collection from GSEAtraining package </span></span>
<span id="cb9-12"><a href="#cb9-12" aria-hidden="true" tabindex="-1"></a>km <span class="ot">&lt;-</span> <span class="fu">get_msigdb</span>(<span class="at">version =</span> <span class="st">"2024.1.Hs"</span>, </span>
<span id="cb9-13"><a href="#cb9-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">collection =</span> <span class="st">"c2.cp.kegg_medicus"</span>,</span>
<span id="cb9-14"><a href="#cb9-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">gene_id_type =</span> <span class="st">"symbols"</span>)</span>
<span id="cb9-15"><a href="#cb9-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-16"><a href="#cb9-16" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine into named list</span></span>
<span id="cb9-17"><a href="#cb9-17" aria-hidden="true" tabindex="-1"></a>pathwaysToTest <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(c2, immunesig, hallmark)</span>
<span id="cb9-18"><a href="#cb9-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-19"><a href="#cb9-19" aria-hidden="true" tabindex="-1"></a>pathwaysToTest_split <span class="ot">&lt;-</span> pathwaysToTest <span class="sc">%&gt;%</span></span>
<span id="cb9-20"><a href="#cb9-20" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(gs_name,human_gene_symbol) <span class="sc">%&gt;%</span></span>
<span id="cb9-21"><a href="#cb9-21" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(gs_name) <span class="sc">%&gt;%</span></span>
<span id="cb9-22"><a href="#cb9-22" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_split</span>()</span>
<span id="cb9-23"><a href="#cb9-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-24"><a href="#cb9-24" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(pathwaysToTest_split) <span class="ot">&lt;-</span> <span class="fu">sapply</span>(pathwaysToTest_split, </span>
<span id="cb9-25"><a href="#cb9-25" aria-hidden="true" tabindex="-1"></a>                                        <span class="at">FUN =</span> <span class="cf">function</span>(x) { x <span class="sc">%&gt;%</span> </span>
<span id="cb9-26"><a href="#cb9-26" aria-hidden="true" tabindex="-1"></a>                                            <span class="fu">pull</span>(gs_name) <span class="sc">%&gt;%</span></span>
<span id="cb9-27"><a href="#cb9-27" aria-hidden="true" tabindex="-1"></a>                                            <span class="fu">unique</span>()</span>
<span id="cb9-28"><a href="#cb9-28" aria-hidden="true" tabindex="-1"></a>                                            }</span>
<span id="cb9-29"><a href="#cb9-29" aria-hidden="true" tabindex="-1"></a>                                        )</span>
<span id="cb9-30"><a href="#cb9-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-31"><a href="#cb9-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-32"><a href="#cb9-32" aria-hidden="true" tabindex="-1"></a>pathwaysToTest_split_list <span class="ot">&lt;-</span> <span class="fu">sapply</span>(pathwaysToTest_split, </span>
<span id="cb9-33"><a href="#cb9-33" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">simplify =</span> <span class="cn">FALSE</span>, </span>
<span id="cb9-34"><a href="#cb9-34" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">USE.NAMES =</span> <span class="cn">TRUE</span>, </span>
<span id="cb9-35"><a href="#cb9-35" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">FUN =</span> <span class="cf">function</span>(x) { x <span class="sc">%&gt;%</span></span>
<span id="cb9-36"><a href="#cb9-36" aria-hidden="true" tabindex="-1"></a>                                        <span class="fu">pull</span>(human_gene_symbol) <span class="sc">%&gt;%</span></span>
<span id="cb9-37"><a href="#cb9-37" aria-hidden="true" tabindex="-1"></a>                                        <span class="fu">unique</span>()</span>
<span id="cb9-38"><a href="#cb9-38" aria-hidden="true" tabindex="-1"></a>                                    }</span>
<span id="cb9-39"><a href="#cb9-39" aria-hidden="true" tabindex="-1"></a>                                )</span>
<span id="cb9-40"><a href="#cb9-40" aria-hidden="true" tabindex="-1"></a><span class="co"># Add KEGG MEDICUS to the list</span></span>
<span id="cb9-41"><a href="#cb9-41" aria-hidden="true" tabindex="-1"></a>pathwaysToTest_split_list <span class="ot">&lt;-</span> <span class="fu">c</span>(pathwaysToTest_split_list, km)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="iterate-over-de-results-and-run-fgsea-on-each-clusters-ranked-gene-list" class="level2" data-number="11.7">
<h2 data-number="11.7" class="anchored" data-anchor-id="iterate-over-de-results-and-run-fgsea-on-each-clusters-ranked-gene-list"><span class="header-section-number">11.7</span> Iterate over DE results and run fgsea on each clusterâ€™s ranked gene list</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">123</span>)</span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>fgseaRes <span class="ot">&lt;-</span> <span class="fu">sapply</span>(skin_de_split_ranks, </span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">simplify =</span> <span class="cn">FALSE</span>, </span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">USE.NAMES =</span> <span class="cn">TRUE</span>, </span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">FUN =</span> <span class="cf">function</span>(x) {</span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a>        <span class="fu">fgsea</span>(<span class="at">pathways =</span> pathwaysToTest_split_list, </span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a>              <span class="at">stats =</span> x,</span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a>              <span class="at">minSize =</span> <span class="dv">5</span>, <span class="co"># Note this is the intersection of genes in the ranked list and genes in the pathway</span></span>
<span id="cb10-9"><a href="#cb10-9" aria-hidden="true" tabindex="-1"></a>              <span class="at">maxSize =</span> <span class="dv">1000</span>,</span>
<span id="cb10-10"><a href="#cb10-10" aria-hidden="true" tabindex="-1"></a>              <span class="at">nproc =</span> <span class="dv">1</span></span>
<span id="cb10-11"><a href="#cb10-11" aria-hidden="true" tabindex="-1"></a>              )</span>
<span id="cb10-12"><a href="#cb10-12" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb10-13"><a href="#cb10-13" aria-hidden="true" tabindex="-1"></a>    )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="collapse-all-results-into-one-tibble-and-filter-for-significance" class="level2" data-number="11.8">
<h2 data-number="11.8" class="anchored" data-anchor-id="collapse-all-results-into-one-tibble-and-filter-for-significance"><span class="header-section-number">11.8</span> Collapse all results into one tibble and filter for significance</h2>
<p>Save the filtered output</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="fu">bind_rows</span>(fgseaRes,<span class="at">.id =</span> <span class="st">"sub.cluster"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">leadingEdgeGenes =</span> <span class="fu">map_chr</span>(leadingEdge, toString)) <span class="sc">%&gt;%</span></span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>leadingEdge) <span class="sc">%&gt;%</span></span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(padj <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span></span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">arrange</span>(sub.cluster,padj,NES) <span class="sc">%&gt;%</span></span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">write_tsv</span>(<span class="st">"Skin_scRNA_PostVsPre3rd_DiffExp_fgseaRes_sig.tsv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="print-selection-of-top-most-significant-pathways" class="level2" data-number="11.9">
<h2 data-number="11.9" class="anchored" data-anchor-id="print-selection-of-top-most-significant-pathways"><span class="header-section-number">11.9</span> Print selection of top-most significant pathways</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="fu">bind_rows</span>(fgseaRes,<span class="at">.id =</span> <span class="st">"sub.cluster"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">leadingEdgeGenes =</span> <span class="fu">map_chr</span>(leadingEdge, toString)) <span class="sc">%&gt;%</span></span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>leadingEdge) <span class="sc">%&gt;%</span></span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(padj <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span></span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">arrange</span>(sub.cluster,padj,<span class="sc">-</span>NES) <span class="sc">%&gt;%</span></span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(sub.cluster) <span class="sc">%&gt;%</span></span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">slice_head</span>(<span class="at">n =</span> <span class="dv">10</span>) <span class="sc">%&gt;%</span></span>
<span id="cb12-9"><a href="#cb12-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pull</span>(pathway) <span class="sc">%&gt;%</span></span>
<span id="cb12-10"><a href="#cb12-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">unique</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "GSE42724_NAIVE_BCELL_VS_PLASMABLAST_UP"                                 
 [2] "GSE13485_DAY3_VS_DAY7_YF17D_VACCINE_PBMC_DN"                            
 [3] "GSE13484_UNSTIM_VS_YF17D_VACCINE_STIM_PBMC_DN"                          
 [4] "GSE13485_CTRL_VS_DAY7_YF17D_VACCINE_PBMC_DN"                            
 [5] "GSE18791_UNSTIM_VS_NEWCATSLE_VIRUS_DC_6H_DN"                            
 [6] "GSE18791_CTRL_VS_NEWCASTLE_VIRUS_DC_8H_DN"                              
 [7] "GSE37533_PPARG1_FOXP3_VS_FOXP3_TRANSDUCED_CD4_TCELL_DN"                 
 [8] "GSE42021_TREG_VS_TCONV_PLN_UP"                                          
 [9] "BLANCO_MELO_BRONCHIAL_EPITHELIAL_CELLS_INFLUENZA_A_DEL_NS1_INFECTION_UP"
[10] "GSE2770_TGFB_AND_IL4_ACT_VS_ACT_CD4_TCELL_2H_DN"                        
[11] "HALLMARK_INTERFERON_GAMMA_RESPONSE"                                     
[12] "GSE37533_PPARG2_FOXP3_VS_FOXP3_TRANSDUCED_CD4_TCELL_DN"                 
[13] "GSE21546_WT_VS_SAP1A_KO_DP_THYMOCYTES_UP"                               
[14] "GSE42021_CD24HI_VS_CD24INT_TREG_THYMUS_DN"                              
[15] "GSE42021_TREG_PLN_VS_TREG_PRECURSORS_THYMUS_DN"                         
[16] "GSE13485_CTRL_VS_DAY3_YF17D_VACCINE_PBMC_DN"                            
[17] "GSE42021_TREG_PLN_VS_CD24LO_TREG_THYMUS_DN"                             
[18] "GSE13485_PRE_VS_POST_YF17D_VACCINATION_PBMC_DN"                         
[19] "TAKEDA_TARGETS_OF_NUP98_HOXA9_FUSION_3D_UP"                             
[20] "GSE13485_DAY1_VS_DAY7_YF17D_VACCINE_PBMC_DN"                            
[21] "GSE21927_SPLEEN_C57BL6_VS_4T1_TUMOR_BALBC_MONOCYTES_DN"                 
[22] "HECKER_IFNB1_TARGETS"                                                   
[23] "REACTOME_EUKARYOTIC_TRANSLATION_ELONGATION"                             
[24] "KEGG_MEDICUS_REFERENCE_TRANSLATION_INITIATION"                          
[25] "BILANGES_SERUM_AND_RAPAMYCIN_SENSITIVE_GENES"                           
[26] "REACTOME_EUKARYOTIC_TRANSLATION_INITIATION"                             
[27] "KEGG_RIBOSOME"                                                          
[28] "GSE42021_TREG_PLN_VS_CD24INT_TREG_THYMUS_DN"                            
[29] "REACTOME_INTERFERON_SIGNALING"                                          
[30] "GSE21546_ELK1_KO_VS_SAP1A_KO_AND_ELK1_KO_DP_THYMOCYTES_UP"              
[31] "WP_CYTOPLASMIC_RIBOSOMAL_PROTEINS"                                      
[32] "REACTOME_SELENOAMINO_ACID_METABOLISM"                                   
[33] "GSE6269_FLU_VS_STREP_PNEUMO_INF_PBMC_UP"                                
[34] "REACTOME_TRANSLATION"                                                   
[35] "BLANCO_MELO_HUMAN_PARAINFLUENZA_VIRUS_3_INFECTION_A594_CELLS_UP"        
[36] "REACTOME_SRP_DEPENDENT_COTRANSLATIONAL_PROTEIN_TARGETING_TO_MEMBRANE"   
[37] "REACTOME_RRNA_PROCESSING"                                               
[38] "REACTOME_NONSENSE_MEDIATED_DECAY_NMD"                                   
[39] "DACOSTA_UV_RESPONSE_VIA_ERCC3_DN"                                       </code></pre>
</div>
</div>
</section>
<section id="print-selection-of-recurring-top-most-significant-pathways-in-multiple-clusters" class="level2" data-number="11.10">
<h2 data-number="11.10" class="anchored" data-anchor-id="print-selection-of-recurring-top-most-significant-pathways-in-multiple-clusters"><span class="header-section-number">11.10</span> Print selection of recurring top-most significant pathways in multiple clusters</h2>
<p>Restrict to HALLMARK, KEGG, WP, or REACTOME pathways seen in the top 100 most significant results in at least 2 clusters</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>recurring_topSigPathways <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(fgseaRes,<span class="at">.id =</span> <span class="st">"sub.cluster"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">leadingEdgeGenes =</span> <span class="fu">map_chr</span>(leadingEdge, toString)) <span class="sc">%&gt;%</span></span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>leadingEdge) <span class="sc">%&gt;%</span></span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(padj <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> NES <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span></span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">arrange</span>(sub.cluster,padj) <span class="sc">%&gt;%</span></span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(sub.cluster) <span class="sc">%&gt;%</span></span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">slice_head</span>(<span class="at">n =</span> <span class="dv">100</span>) <span class="sc">%&gt;%</span></span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(sub.cluster,pathway,padj,NES) <span class="sc">%&gt;%</span></span>
<span id="cb14-10"><a href="#cb14-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(pathway) <span class="sc">%&gt;%</span></span>
<span id="cb14-11"><a href="#cb14-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarize</span>(<span class="at">n =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb14-12"><a href="#cb14-12" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(n<span class="sc">&gt;=</span><span class="dv">2</span>) <span class="sc">%&gt;%</span></span>
<span id="cb14-13"><a href="#cb14-13" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="fu">str_detect</span>(pathway,<span class="st">"HALLMARK|WP|KEGG|REACTOME"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb14-14"><a href="#cb14-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pull</span>(pathway) <span class="sc">%&gt;%</span></span>
<span id="cb14-15"><a href="#cb14-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">unique</span>()</span>
<span id="cb14-16"><a href="#cb14-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-17"><a href="#cb14-17" aria-hidden="true" tabindex="-1"></a>recurring_topSigPathways</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "HALLMARK_INTERFERON_ALPHA_RESPONSE"          
[2] "HALLMARK_INTERFERON_GAMMA_RESPONSE"          
[3] "REACTOME_ADAPTIVE_IMMUNE_SYSTEM"             
[4] "REACTOME_CYTOKINE_SIGNALING_IN_IMMUNE_SYSTEM"
[5] "REACTOME_INTERFERON_ALPHA_BETA_SIGNALING"    
[6] "REACTOME_INTERFERON_SIGNALING"               
[7] "WP_PROTEASOME_DEGRADATION"                   </code></pre>
</div>
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(recurring_topSigPathways) <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="st">"Other recurring significant"</span>,<span class="fu">length</span>(recurring_topSigPathways))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="filter-to-specific-pathways-related-to-innate-immunity-of-interest" class="level2" data-number="11.11">
<h2 data-number="11.11" class="anchored" data-anchor-id="filter-to-specific-pathways-related-to-innate-immunity-of-interest"><span class="header-section-number">11.11</span> Filter to specific pathways related to innate immunity of interest</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>topsigPathways <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Innate stimulation"</span> <span class="ot">=</span> <span class="st">"KEGG_TOLL_LIKE_RECEPTOR_SIGNALING_PATHWAY"</span>,</span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>                            <span class="st">"Interferon signaling"</span> <span class="ot">=</span> <span class="st">"REACTOME_INTERFERON_SIGNALING"</span>,</span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a>                            <span class="st">"Interferon signaling"</span> <span class="ot">=</span> <span class="st">"REACTOME_INTERFERON_ALPHA_BETA_SIGNALING"</span>,</span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a>                            <span class="st">"Interferon signaling"</span> <span class="ot">=</span> <span class="st">"HALLMARK_INTERFERON_ALPHA_RESPONSE"</span>,</span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a>                            <span class="st">"Interferon signaling"</span> <span class="ot">=</span> <span class="st">"HALLMARK_INTERFERON_GAMMA_RESPONSE"</span>,</span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a>                            <span class="st">"Cytokine signaling"</span> <span class="ot">=</span> <span class="st">"REACTOME_CYTOKINE_SIGNALING_IN_IMMUNE_SYSTEM"</span>,</span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a>                            <span class="st">"Cytokine signaling"</span> <span class="ot">=</span> <span class="st">"REACTOME_SIGNALING_BY_INTERLEUKINS"</span>,</span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a>                            <span class="st">"Cytokine signaling"</span> <span class="ot">=</span> <span class="st">"REACTOME_INTERLEUKIN_1_FAMILY_SIGNALING"</span>,</span>
<span id="cb17-9"><a href="#cb17-9" aria-hidden="true" tabindex="-1"></a>                            <span class="st">"Cytokine signaling"</span> <span class="ot">=</span> <span class="st">"REACTOME_INTERLEUKIN_2_FAMILY_SIGNALING"</span>,</span>
<span id="cb17-10"><a href="#cb17-10" aria-hidden="true" tabindex="-1"></a>                            <span class="st">"Cytokine signaling"</span> <span class="ot">=</span> <span class="st">"HALLMARK_IL2_STAT5_SIGNALING"</span>,</span>
<span id="cb17-11"><a href="#cb17-11" aria-hidden="true" tabindex="-1"></a>                            <span class="st">"Cytokine signaling"</span> <span class="ot">=</span> <span class="st">"HALLMARK_IL6_JAK_STAT3_SIGNALING"</span>,</span>
<span id="cb17-12"><a href="#cb17-12" aria-hidden="true" tabindex="-1"></a>                            <span class="st">"Cytokine signaling"</span> <span class="ot">=</span> <span class="st">"HALLMARK_TNFA_SIGNALING_VIA_NFKB"</span>,</span>
<span id="cb17-13"><a href="#cb17-13" aria-hidden="true" tabindex="-1"></a>                            <span class="st">"Antigen processing"</span> <span class="ot">=</span> <span class="st">"REACTOME_ANTIGEN_PROCESSING_CROSS_PRESENTATION"</span>,</span>
<span id="cb17-14"><a href="#cb17-14" aria-hidden="true" tabindex="-1"></a>                            <span class="st">"Antigen processing"</span> <span class="ot">=</span> <span class="st">"REACTOME_ANTIGEN_PROCESSING_UBIQUITINATION_PROTEASOME_DEGRADATION"</span></span>
<span id="cb17-15"><a href="#cb17-15" aria-hidden="true" tabindex="-1"></a>                )</span>
<span id="cb17-16"><a href="#cb17-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-17"><a href="#cb17-17" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine with top significant and remove redundant pathways</span></span>
<span id="cb17-18"><a href="#cb17-18" aria-hidden="true" tabindex="-1"></a>topsigPathways <span class="ot">&lt;-</span> <span class="fu">c</span>(topsigPathways,recurring_topSigPathways[<span class="sc">!</span>recurring_topSigPathways <span class="sc">%in%</span> topsigPathways])</span>
<span id="cb17-19"><a href="#cb17-19" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb17-20"><a href="#cb17-20" aria-hidden="true" tabindex="-1"></a>gs_pathways <span class="ot">&lt;-</span> pathwaysToTest_split_list[topsigPathways]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="plot-dot-plot-of-nes-and-p-value-for-pathways-of-interest" class="level2" data-number="11.12">
<h2 data-number="11.12" class="anchored" data-anchor-id="plot-dot-plot-of-nes-and-p-value-for-pathways-of-interest"><span class="header-section-number">11.12</span> Plot dot plot of NES and p-value for pathways of interest</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="fu">bind_rows</span>(fgseaRes,<span class="at">.id =</span> <span class="st">"sub.cluster"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">leadingEdgeGenes =</span> <span class="fu">map_chr</span>(leadingEdge, toString)) <span class="sc">%&gt;%</span></span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>leadingEdge) <span class="sc">%&gt;%</span></span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(pathway <span class="sc">%in%</span> topsigPathways) <span class="sc">%&gt;%</span></span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(<span class="fu">enframe</span>(topsigPathways,<span class="at">name =</span> <span class="st">"PathwayGroup"</span>,<span class="at">value =</span> <span class="st">"pathway"</span>),<span class="at">by=</span><span class="st">"pathway"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb18-7"><a href="#cb18-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(<span class="fu">enframe</span>(cluster_annot,<span class="at">name =</span> <span class="st">"sub.cluster"</span>, <span class="at">value =</span> <span class="st">"Cluster"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb18-8"><a href="#cb18-8" aria-hidden="true" tabindex="-1"></a>                <span class="fu">mutate</span>(<span class="at">sub.cluster =</span> <span class="fu">paste0</span>(<span class="st">"Cluster"</span>,sub.cluster)), </span>
<span id="cb18-9"><a href="#cb18-9" aria-hidden="true" tabindex="-1"></a>              <span class="at">by =</span> <span class="st">"sub.cluster"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb18-10"><a href="#cb18-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">logP =</span> <span class="fu">ifelse</span>(padj <span class="sc">&lt;</span> <span class="fl">0.05</span>, <span class="sc">-</span><span class="fu">log10</span>(padj), <span class="dv">0</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb18-11"><a href="#cb18-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">fct_relevel</span>(Cluster,cluster_annot[cluster_order][cluster_annot[cluster_order] <span class="sc">%in%</span> Cluster]), <span class="at">y =</span> pathway)) <span class="sc">+</span></span>
<span id="cb18-12"><a href="#cb18-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">color =</span> NES, <span class="at">size =</span> logP, <span class="at">alpha =</span> <span class="fu">I</span>(<span class="fu">ifelse</span>(logP <span class="sc">&lt;</span> <span class="sc">-</span><span class="fu">log10</span>(<span class="fl">0.05</span>), <span class="dv">0</span>, <span class="dv">1</span>)))) <span class="sc">+</span></span>
<span id="cb18-13"><a href="#cb18-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_size_continuous</span>(<span class="st">"-log10 adj p"</span>, <span class="at">range =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">10</span>), <span class="at">breaks =</span> <span class="fu">c</span>(<span class="fl">1.3</span>, <span class="dv">5</span>, <span class="dv">10</span>, <span class="dv">20</span>, <span class="dv">30</span>)) <span class="sc">+</span></span>
<span id="cb18-14"><a href="#cb18-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_color_gradient2</span>(<span class="at">low =</span> <span class="st">"blue"</span>, <span class="at">mid =</span> <span class="st">"white"</span>, <span class="at">high =</span> <span class="st">"red"</span>, <span class="at">limits=</span><span class="fu">c</span>(<span class="sc">-</span><span class="dv">3</span>,<span class="dv">3</span>), <span class="at">midpoint =</span> <span class="dv">0</span>, <span class="at">oob =</span> scales<span class="sc">::</span>squish) <span class="sc">+</span></span>
<span id="cb18-15"><a href="#cb18-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb18-16"><a href="#cb18-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">facet_grid</span>(<span class="fu">fct_relevel</span>(PathwayGroup,</span>
<span id="cb18-17"><a href="#cb18-17" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">c</span>(<span class="st">"Innate stimulation"</span>,</span>
<span id="cb18-18"><a href="#cb18-18" aria-hidden="true" tabindex="-1"></a>                              <span class="st">"Interferon signaling"</span>,</span>
<span id="cb18-19"><a href="#cb18-19" aria-hidden="true" tabindex="-1"></a>                              <span class="st">"Cytokine signaling"</span>,</span>
<span id="cb18-20"><a href="#cb18-20" aria-hidden="true" tabindex="-1"></a>                              <span class="st">"Antigen processing"</span>,</span>
<span id="cb18-21"><a href="#cb18-21" aria-hidden="true" tabindex="-1"></a>                              <span class="st">"Other recurring significant"</span>)</span>
<span id="cb18-22"><a href="#cb18-22" aria-hidden="true" tabindex="-1"></a>                           )<span class="sc">~</span>.,</span>
<span id="cb18-23"><a href="#cb18-23" aria-hidden="true" tabindex="-1"></a>               <span class="at">scales=</span><span class="st">"free"</span>,</span>
<span id="cb18-24"><a href="#cb18-24" aria-hidden="true" tabindex="-1"></a>               <span class="at">switch=</span><span class="st">"y"</span>,</span>
<span id="cb18-25"><a href="#cb18-25" aria-hidden="true" tabindex="-1"></a>               <span class="at">space=</span><span class="st">"free"</span>) <span class="sc">+</span></span>
<span id="cb18-26"><a href="#cb18-26" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_y_discrete</span>(<span class="at">position =</span> <span class="st">"right"</span>) <span class="sc">+</span></span>
<span id="cb18-27"><a href="#cb18-27" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ylab</span>(<span class="st">""</span>) <span class="sc">+</span></span>
<span id="cb18-28"><a href="#cb18-28" aria-hidden="true" tabindex="-1"></a>    <span class="fu">xlab</span>(<span class="st">""</span>) <span class="sc">+</span></span>
<span id="cb18-29"><a href="#cb18-29" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">90</span>, <span class="at">vjust =</span> <span class="fl">0.5</span>, <span class="at">hjust=</span><span class="dv">1</span>),</span>
<span id="cb18-30"><a href="#cb18-30" aria-hidden="true" tabindex="-1"></a>          <span class="at">strip.text.y.left =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">0</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Skin_scRNA_Part11_files/figure-html/unnamed-chunk-12-1.png" class="img-fluid figure-img" width="1440"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="get-session-info" class="level2" data-number="11.13">
<h2 data-number="11.13" class="anchored" data-anchor-id="get-session-info"><span class="header-section-number">11.13</span> Get session info</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>R version 4.3.1 (2023-06-16)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Rocky Linux 8.10 (Green Obsidian)

Matrix products: default
BLAS/LAPACK: /usr/lib64/libopenblasp-r0.3.15.so;  LAPACK version 3.9.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

time zone: America/New_York
tzcode source: system (glibc)

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] GSEAtraining_0.99.1 fgsea_1.26.0        msigdbr_7.5.1      
 [4] lubridate_1.9.3     forcats_1.0.0       stringr_1.5.1      
 [7] dplyr_1.1.4         purrr_1.0.2         readr_2.1.5        
[10] tidyr_1.3.1         tibble_3.2.1        tidyverse_2.0.0    
[13] Seurat_5.1.0        SeuratObject_5.0.2  sp_2.1-3           
[16] ggplot2_3.5.1      

loaded via a namespace (and not attached):
  [1] RColorBrewer_1.1-3      jsonlite_1.8.8          magrittr_2.0.3         
  [4] spatstat.utils_3.0-4    farver_2.1.1            rmarkdown_2.25         
  [7] zlibbioc_1.46.0         vctrs_0.6.5             ROCR_1.0-11            
 [10] memoise_2.0.1           spatstat.explore_3.2-6  RCurl_1.98-1.14        
 [13] htmltools_0.5.7         sctransform_0.4.1       parallelly_1.37.0      
 [16] KernSmooth_2.23-22      htmlwidgets_1.6.4       ica_1.0-3              
 [19] plyr_1.8.9              cachem_1.0.8            plotly_4.10.4          
 [22] zoo_1.8-12              igraph_2.0.2            mime_0.12              
 [25] lifecycle_1.0.4         pkgconfig_2.0.3         Matrix_1.6-4           
 [28] R6_2.5.1                fastmap_1.1.1           GenomeInfoDbData_1.2.10
 [31] fitdistrplus_1.1-11     future_1.33.1           shiny_1.8.0            
 [34] digest_0.6.34           colorspace_2.1-0        S4Vectors_0.38.2       
 [37] AnnotationDbi_1.64.1    patchwork_1.3.0         tensor_1.5             
 [40] RSpectra_0.16-1         irlba_2.3.5.1           RSQLite_2.3.5          
 [43] labeling_0.4.3          progressr_0.14.0        fansi_1.0.6            
 [46] spatstat.sparse_3.0-3   timechange_0.3.0        httr_1.4.7             
 [49] polyclip_1.10-6         abind_1.4-5             compiler_4.3.1         
 [52] bit64_4.0.5             withr_3.0.0             BiocParallel_1.34.2    
 [55] DBI_1.2.2               fastDummies_1.7.3       MASS_7.3-60.0.1        
 [58] tools_4.3.1             lmtest_0.9-40           httpuv_1.6.14          
 [61] future.apply_1.11.1     goftest_1.2-3           glue_1.7.0             
 [64] nlme_3.1-164            promises_1.2.1          grid_4.3.1             
 [67] Rtsne_0.17              cluster_2.1.6           reshape2_1.4.4         
 [70] generics_0.1.3          gtable_0.3.4            spatstat.data_3.0-4    
 [73] tzdb_0.4.0              data.table_1.15.0       hms_1.1.3              
 [76] XVector_0.40.0          utf8_1.2.4              BiocGenerics_0.46.0    
 [79] spatstat.geom_3.2-8     RcppAnnoy_0.0.22        ggrepel_0.9.5          
 [82] RANN_2.6.1              pillar_1.9.0            vroom_1.6.5            
 [85] spam_2.10-0             babelgene_22.9          RcppHNSW_0.6.0         
 [88] later_1.3.2             splines_4.3.1           lattice_0.22-5         
 [91] bit_4.0.5               survival_3.5-8          deldir_2.0-2           
 [94] tidyselect_1.2.0        Biostrings_2.68.1       miniUI_0.1.1.1         
 [97] pbapply_1.7-2           knitr_1.45              gridExtra_2.3          
[100] IRanges_2.34.1          scattermore_1.2         stats4_4.3.1           
[103] xfun_0.42               Biobase_2.60.0          matrixStats_1.2.0      
[106] DT_0.31                 stringi_1.8.3           lazyeval_0.2.2         
[109] yaml_2.3.8              evaluate_0.23           codetools_0.2-19       
[112] cli_3.6.2               uwot_0.1.16             xtable_1.8-4           
[115] reticulate_1.35.0       munsell_0.5.0           GenomeInfoDb_1.36.4    
[118] Rcpp_1.0.12             globals_0.16.2          spatstat.random_3.2-2  
[121] png_0.1-8               parallel_4.3.1          ellipsis_0.3.2         
[124] blob_1.2.4              dotCall64_1.1-1         bitops_1.0-7           
[127] listenv_0.9.1           viridisLite_0.4.2       scales_1.3.0           
[130] ggridges_0.5.6          crayon_1.5.2            leiden_0.4.3.1         
[133] rlang_1.1.3             KEGGREST_1.40.1         fastmatch_1.1-4        
[136] cowplot_1.1.3          </code></pre>
</div>
</div>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "î§‹";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="./Skin_scRNA_Part10.html" class="pagination-link" aria-label="Skin: Deep phenotyping, cluster annotation, and exploration of Post vs Pre 3rd vaccination differences in cell proportions and expression">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">10</span>&nbsp; <span class="chapter-title">Skin: Deep phenotyping, cluster annotation, and exploration of Post vs Pre 3rd vaccination differences in cell proportions and expression</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
  </div>
</nav>
</div> <!-- /content -->




</body></html>