<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.6.39">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>6&nbsp; Skin: Sub-clustering on T/NK lineage â€“ Blass &amp; Keskin (2025): Melanoma NeoVax analysis: Skin single-cell RNAseq</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./Skin_scRNA_Part7.html" rel="next">
<link href="./Skin_scRNA_Part5.html" rel="prev">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-e26003cea8cd680ca0c55a263523d882.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap-41a4eb7d26fa9333a8f88b12b94df59e.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


</head>

<body class="nav-sidebar floating">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./Skin_scRNA_Part6.html"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">Skin: Sub-clustering on T/NK lineage</span></a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="./">Blass &amp; Keskin (2025): Melanoma NeoVax analysis: Skin single-cell RNAseq</a> 
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Introduction</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Skin_scRNA_Part1.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Skin: QC &amp; Filtering</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Skin_scRNA_Part2.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Skin: Normalization, doublet discrimination, integration, clustering</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Skin_scRNA_Part3.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Skin: Check patient sexes</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Skin_scRNA_Part4.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Skin: Post-clustering QC and characterization</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Skin_scRNA_Part5.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">Skin: Post-clustering comprehensive marker identification</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Skin_scRNA_Part6.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">Skin: Sub-clustering on T/NK lineage</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Skin_scRNA_Part7.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">Skin: Assisting cell type annotation with enrichR</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Skin_scRNA_Part8.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">8</span>&nbsp; <span class="chapter-title">Skin: Classification of CD4 and CD8 T cell states within T/NK lineage</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Skin_scRNA_Part9.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">9</span>&nbsp; <span class="chapter-title">Skin: Differential expression analysis, Post vs Pre 3rd vaccination</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Skin_scRNA_Part10.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">10</span>&nbsp; <span class="chapter-title">Skin: Deep phenotyping, cluster annotation, and exploration of Post vs Pre 3rd vaccination differences in cell proportions and expression</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Skin_scRNA_Part11.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">11</span>&nbsp; <span class="chapter-title">Skin: Calculating Post vs Pre 3rd vaccination differences in pathway enrichments with fgsea</span></span></a>
  </div>
</li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#set-up-seurat-workspace" id="toc-set-up-seurat-workspace" class="nav-link active" data-scroll-target="#set-up-seurat-workspace"><span class="header-section-number">6.1</span> Set up Seurat workspace</a></li>
  <li><a href="#load-previously-saved-object" id="toc-load-previously-saved-object" class="nav-link" data-scroll-target="#load-previously-saved-object"><span class="header-section-number">6.2</span> Load previously saved object</a></li>
  <li><a href="#set-idents-to-preferred-initial-clustering-resolution" id="toc-set-idents-to-preferred-initial-clustering-resolution" class="nav-link" data-scroll-target="#set-idents-to-preferred-initial-clustering-resolution"><span class="header-section-number">6.3</span> Set idents to preferred initial clustering resolution</a></li>
  <li><a href="#run-sub-clustering-on-tnk-lineage" id="toc-run-sub-clustering-on-tnk-lineage" class="nav-link" data-scroll-target="#run-sub-clustering-on-tnk-lineage"><span class="header-section-number">6.4</span> Run sub-clustering on T/NK lineage</a>
  <ul class="collapse">
  <li><a href="#cluster-1" id="toc-cluster-1" class="nav-link" data-scroll-target="#cluster-1"><span class="header-section-number">6.4.1</span> Cluster 1</a></li>
  <li><a href="#cluster-2" id="toc-cluster-2" class="nav-link" data-scroll-target="#cluster-2"><span class="header-section-number">6.4.2</span> Cluster 2</a></li>
  <li><a href="#cluster-4" id="toc-cluster-4" class="nav-link" data-scroll-target="#cluster-4"><span class="header-section-number">6.4.3</span> Cluster 4</a></li>
  <li><a href="#cluster-5" id="toc-cluster-5" class="nav-link" data-scroll-target="#cluster-5"><span class="header-section-number">6.4.4</span> Cluster 5</a></li>
  <li><a href="#cluster-6" id="toc-cluster-6" class="nav-link" data-scroll-target="#cluster-6"><span class="header-section-number">6.4.5</span> Cluster 6</a></li>
  <li><a href="#cluster-9" id="toc-cluster-9" class="nav-link" data-scroll-target="#cluster-9"><span class="header-section-number">6.4.6</span> Cluster 9</a></li>
  <li><a href="#cluster-11" id="toc-cluster-11" class="nav-link" data-scroll-target="#cluster-11"><span class="header-section-number">6.4.7</span> Cluster 11</a></li>
  <li><a href="#cluster-13" id="toc-cluster-13" class="nav-link" data-scroll-target="#cluster-13"><span class="header-section-number">6.4.8</span> Cluster 13</a></li>
  <li><a href="#cluster-19" id="toc-cluster-19" class="nav-link" data-scroll-target="#cluster-19"><span class="header-section-number">6.4.9</span> Cluster 19</a></li>
  </ul></li>
  <li><a href="#plot-updated-qc-metrics-per-clustersub-cluster" id="toc-plot-updated-qc-metrics-per-clustersub-cluster" class="nav-link" data-scroll-target="#plot-updated-qc-metrics-per-clustersub-cluster"><span class="header-section-number">6.5</span> Plot updated QC metrics per cluster/sub-cluster</a></li>
  <li><a href="#identify-updated-cursory-marker-genes-per-clustersub-cluster" id="toc-identify-updated-cursory-marker-genes-per-clustersub-cluster" class="nav-link" data-scroll-target="#identify-updated-cursory-marker-genes-per-clustersub-cluster"><span class="header-section-number">6.6</span> Identify updated cursory marker genes per cluster/sub-cluster</a></li>
  <li><a href="#plot-top-markers-identified-and-canonical-genes-as-a-dotplot" id="toc-plot-top-markers-identified-and-canonical-genes-as-a-dotplot" class="nav-link" data-scroll-target="#plot-top-markers-identified-and-canonical-genes-as-a-dotplot"><span class="header-section-number">6.7</span> Plot top markers identified and canonical genes as a dotplot</a></li>
  <li><a href="#plot-updated-skin-sample-proportions-pre-vs-post-3rd-vaccine-per-cluster" id="toc-plot-updated-skin-sample-proportions-pre-vs-post-3rd-vaccine-per-cluster" class="nav-link" data-scroll-target="#plot-updated-skin-sample-proportions-pre-vs-post-3rd-vaccine-per-cluster"><span class="header-section-number">6.8</span> Plot updated skin sample proportions, pre vs post 3rd vaccine, per cluster</a></li>
  <li><a href="#save-updated-object" id="toc-save-updated-object" class="nav-link" data-scroll-target="#save-updated-object"><span class="header-section-number">6.9</span> Save updated object</a></li>
  <li><a href="#get-session-info" id="toc-get-session-info" class="nav-link" data-scroll-target="#get-session-info"><span class="header-section-number">6.10</span> Get session info</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title"><span id="sec-skin-tnk-subclustering" class="quarto-section-identifier"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">Skin: Sub-clustering on T/NK lineage</span></span></h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<section id="set-up-seurat-workspace" class="level2" data-number="6.1">
<h2 data-number="6.1" class="anchored" data-anchor-id="set-up-seurat-workspace"><span class="header-section-number">6.1</span> Set up Seurat workspace</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load libraries</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(data.table)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(devtools)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(presto)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(glmGamPoi)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(sctransform) </span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Seurat)</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(miQC)  </span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(SeuratWrappers)</span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(flexmix)</span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(SingleCellExperiment)</span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(SummarizedExperiment)</span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readxl)</span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(fishpond)</span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Matrix)</span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(speckle)</span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(scater)</span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(vctrs)</span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(alevinQC)</span>
<span id="cb1-22"><a href="#cb1-22" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(harmony)</span>
<span id="cb1-23"><a href="#cb1-23" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(scDblFinder)</span>
<span id="cb1-24"><a href="#cb1-24" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(cellXY)</span>
<span id="cb1-25"><a href="#cb1-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-26"><a href="#cb1-26" aria-hidden="true" tabindex="-1"></a><span class="co"># Set global options for Seurat v5 objects</span></span>
<span id="cb1-27"><a href="#cb1-27" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">Seurat.object.assay.version =</span> <span class="st">'v5'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="load-previously-saved-object" class="level2" data-number="6.2">
<h2 data-number="6.2" class="anchored" data-anchor-id="load-previously-saved-object"><span class="header-section-number">6.2</span> Load previously saved object</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>merged.<span class="fl">18279.</span>skin.singlets <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"Skin_scRNA_Part3.rds"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="set-idents-to-preferred-initial-clustering-resolution" class="level2" data-number="6.3">
<h2 data-number="6.3" class="anchored" data-anchor-id="set-idents-to-preferred-initial-clustering-resolution"><span class="header-section-number">6.3</span> Set idents to preferred initial clustering resolution</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">Idents</span>(merged.<span class="fl">18279.</span>skin.singlets) <span class="ot">&lt;-</span> merged.<span class="fl">18279.</span>skin.singlets<span class="sc">$</span>RNA_snn_res.<span class="fl">0.8</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>merged.<span class="fl">18279.</span>skin.singlets<span class="sc">$</span>seurat_clusters <span class="ot">&lt;-</span> merged.<span class="fl">18279.</span>skin.singlets<span class="sc">$</span>RNA_snn_res.<span class="fl">0.8</span></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(merged.<span class="fl">18279.</span>skin.singlets, <span class="at">reduction=</span><span class="st">"umap.harmony"</span>, <span class="at">label=</span>T)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Skin_scRNA_Part6_files/figure-html/unnamed-chunk-3-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="run-sub-clustering-on-tnk-lineage" class="level2" data-number="6.4">
<h2 data-number="6.4" class="anchored" data-anchor-id="run-sub-clustering-on-tnk-lineage"><span class="header-section-number">6.4</span> Run sub-clustering on T/NK lineage</h2>
<p>T/NK cells are in clusters 1, 2, 4, 5, 6, 9, 11, 13, 19</p>
<section id="cluster-1" class="level3" data-number="6.4.1">
<h3 data-number="6.4.1" class="anchored" data-anchor-id="cluster-1"><span class="header-section-number">6.4.1</span> Cluster 1</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>merged.<span class="fl">18279.</span>skin.singlets <span class="ot">&lt;-</span> <span class="fu">FindSubCluster</span>(<span class="at">object =</span> merged.<span class="fl">18279.</span>skin.singlets, </span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>                                        <span class="at">cluster =</span> <span class="st">"1"</span>,</span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>                                        <span class="at">resolution =</span> <span class="fl">0.2</span>,</span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>                                        <span class="at">graph.name =</span> <span class="st">"RNA_snn"</span>,  </span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>                                        <span class="at">algorithm =</span> <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 3688
Number of edges: 114356

Running Louvain algorithm with multilevel refinement...
Maximum modularity in 10 random starts: 0.8361
Number of communities: 2
Elapsed time: 0 seconds</code></pre>
</div>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(merged.<span class="fl">18279.</span>skin.singlets,  </span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>        <span class="at">reduction =</span> <span class="st">"umap.harmony"</span>,  </span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>        <span class="at">group.by =</span> <span class="st">"sub.cluster"</span>, </span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>        <span class="at">label =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Skin_scRNA_Part6_files/figure-html/unnamed-chunk-4-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(merged.<span class="fl">18279.</span>skin.singlets<span class="sc">$</span>sub.cluster)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
   0  1_0  1_1   10   11   12   13   14   15   16   17   18   19    2   20   21 
5768 2588 1100 1123 1086  907  569  557  472  407  360  308  280 2851  262  238 
  22   23   24   25   26    3    4    5    6    7    8    9 
 228  177  175   46   33 2418 2363 2200 2186 1746 1483 1466 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="fu">Idents</span>(merged.<span class="fl">18279.</span>skin.singlets) <span class="ot">&lt;-</span> <span class="st">"sub.cluster"</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a><span class="fu">levels</span>(merged.<span class="fl">18279.</span>skin.singlets)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "10"  "7"   "0"   "25"  "8"   "18"  "21"  "3"   "1_1" "13"  "14"  "17" 
[13] "23"  "1_0" "12"  "4"   "11"  "9"   "5"   "6"   "2"   "16"  "19"  "22" 
[25] "24"  "15"  "20"  "26" </code></pre>
</div>
</div>
</section>
<section id="cluster-2" class="level3" data-number="6.4.2">
<h3 data-number="6.4.2" class="anchored" data-anchor-id="cluster-2"><span class="header-section-number">6.4.2</span> Cluster 2</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>merged.<span class="fl">18279.</span>skin.singlets <span class="ot">&lt;-</span> <span class="fu">FindSubCluster</span>(<span class="at">object =</span> merged.<span class="fl">18279.</span>skin.singlets, </span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>                                        <span class="at">cluster =</span> <span class="st">"2"</span>,</span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>                                        <span class="at">resolution =</span> <span class="fl">0.2</span>,</span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>                                        <span class="at">graph.name =</span> <span class="st">"RNA_snn"</span>,  </span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a>                                        <span class="at">algorithm =</span> <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 2851
Number of edges: 89069

Running Louvain algorithm with multilevel refinement...
Maximum modularity in 10 random starts: 0.8349
Number of communities: 3
Elapsed time: 0 seconds</code></pre>
</div>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(merged.<span class="fl">18279.</span>skin.singlets,  </span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>        <span class="at">reduction =</span> <span class="st">"umap.harmony"</span>,  </span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>        <span class="at">group.by =</span> <span class="st">"sub.cluster"</span>, </span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a>        <span class="at">label =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Skin_scRNA_Part6_files/figure-html/unnamed-chunk-5-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(merged.<span class="fl">18279.</span>skin.singlets<span class="sc">$</span>sub.cluster)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
   0  1_0  1_1   10   11   12   13   14   15   16   17   18   19  2_0  2_1  2_2 
5768 2588 1100 1123 1086  907  569  557  472  407  360  308  280 1464  980  407 
  20   21   22   23   24   25   26    3    4    5    6    7    8    9 
 262  238  228  177  175   46   33 2418 2363 2200 2186 1746 1483 1466 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="fu">Idents</span>(merged.<span class="fl">18279.</span>skin.singlets) <span class="ot">&lt;-</span> <span class="st">"sub.cluster"</span></span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a><span class="fu">levels</span>(merged.<span class="fl">18279.</span>skin.singlets)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "10"  "7"   "0"   "25"  "8"   "18"  "21"  "3"   "1_1" "13"  "14"  "17" 
[13] "23"  "1_0" "12"  "4"   "11"  "9"   "5"   "6"   "2_1" "2_0" "16"  "19" 
[25] "2_2" "22"  "24"  "15"  "20"  "26" </code></pre>
</div>
</div>
</section>
<section id="cluster-4" class="level3" data-number="6.4.3">
<h3 data-number="6.4.3" class="anchored" data-anchor-id="cluster-4"><span class="header-section-number">6.4.3</span> Cluster 4</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a>merged.<span class="fl">18279.</span>skin.singlets <span class="ot">&lt;-</span> <span class="fu">FindSubCluster</span>(<span class="at">object =</span> merged.<span class="fl">18279.</span>skin.singlets, </span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>                                        <span class="at">cluster =</span> <span class="st">"4"</span>,</span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a>                                        <span class="at">resolution =</span> <span class="fl">0.2</span>,</span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a>                                        <span class="at">graph.name =</span> <span class="st">"RNA_snn"</span>,  </span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a>                                        <span class="at">algorithm =</span> <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 2363
Number of edges: 65111

Running Louvain algorithm with multilevel refinement...
Maximum modularity in 10 random starts: 0.8414
Number of communities: 2
Elapsed time: 0 seconds</code></pre>
</div>
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(merged.<span class="fl">18279.</span>skin.singlets,  </span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a>        <span class="at">reduction =</span> <span class="st">"umap.harmony"</span>,  </span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a>        <span class="at">group.by =</span> <span class="st">"sub.cluster"</span>, </span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a>        <span class="at">label =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Skin_scRNA_Part6_files/figure-html/unnamed-chunk-6-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(merged.<span class="fl">18279.</span>skin.singlets<span class="sc">$</span>sub.cluster)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
   0  1_0  1_1   10   11   12   13   14   15   16   17   18   19  2_0  2_1  2_2 
5768 2588 1100 1123 1086  907  569  557  472  407  360  308  280 1464  980  407 
  20   21   22   23   24   25   26    3  4_0  4_1    5    6    7    8    9 
 262  238  228  177  175   46   33 2418 1826  537 2200 2186 1746 1483 1466 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="fu">Idents</span>(merged.<span class="fl">18279.</span>skin.singlets) <span class="ot">&lt;-</span> <span class="st">"sub.cluster"</span></span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a><span class="fu">levels</span>(merged.<span class="fl">18279.</span>skin.singlets)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "10"  "7"   "0"   "25"  "8"   "18"  "21"  "3"   "1_1" "13"  "14"  "17" 
[13] "23"  "1_0" "12"  "4_0" "11"  "9"   "5"   "6"   "2_1" "2_0" "16"  "19" 
[25] "2_2" "22"  "24"  "15"  "20"  "4_1" "26" </code></pre>
</div>
</div>
</section>
<section id="cluster-5" class="level3" data-number="6.4.4">
<h3 data-number="6.4.4" class="anchored" data-anchor-id="cluster-5"><span class="header-section-number">6.4.4</span> Cluster 5</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a>merged.<span class="fl">18279.</span>skin.singlets <span class="ot">&lt;-</span> <span class="fu">FindSubCluster</span>(<span class="at">object =</span> merged.<span class="fl">18279.</span>skin.singlets, </span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a>                                        <span class="at">cluster =</span> <span class="st">"5"</span>,</span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a>                                        <span class="at">resolution =</span> <span class="fl">0.2</span>,</span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a>                                        <span class="at">graph.name =</span> <span class="st">"RNA_snn"</span>,  </span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a>                                        <span class="at">algorithm =</span> <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 2200
Number of edges: 82669

Running Louvain algorithm with multilevel refinement...
Maximum modularity in 10 random starts: 0.8513
Number of communities: 3
Elapsed time: 0 seconds</code></pre>
</div>
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(merged.<span class="fl">18279.</span>skin.singlets,  </span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a>        <span class="at">reduction =</span> <span class="st">"umap.harmony"</span>,  </span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a>        <span class="at">group.by =</span> <span class="st">"sub.cluster"</span>, </span>
<span id="cb27-4"><a href="#cb27-4" aria-hidden="true" tabindex="-1"></a>        <span class="at">label =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Skin_scRNA_Part6_files/figure-html/unnamed-chunk-7-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(merged.<span class="fl">18279.</span>skin.singlets<span class="sc">$</span>sub.cluster)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
   0  1_0  1_1   10   11   12   13   14   15   16   17   18   19  2_0  2_1  2_2 
5768 2588 1100 1123 1086  907  569  557  472  407  360  308  280 1464  980  407 
  20   21   22   23   24   25   26    3  4_0  4_1  5_0  5_1  5_2    6    7    8 
 262  238  228  177  175   46   33 2418 1826  537 1465  564  171 2186 1746 1483 
   9 
1466 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="fu">Idents</span>(merged.<span class="fl">18279.</span>skin.singlets) <span class="ot">&lt;-</span> <span class="st">"sub.cluster"</span></span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a><span class="fu">levels</span>(merged.<span class="fl">18279.</span>skin.singlets)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "10"  "7"   "0"   "25"  "8"   "18"  "21"  "3"   "1_1" "13"  "14"  "17" 
[13] "23"  "1_0" "12"  "4_0" "11"  "9"   "5_0" "6"   "2_1" "2_0" "16"  "5_1"
[25] "19"  "2_2" "5_2" "22"  "24"  "15"  "20"  "4_1" "26" </code></pre>
</div>
</div>
</section>
<section id="cluster-6" class="level3" data-number="6.4.5">
<h3 data-number="6.4.5" class="anchored" data-anchor-id="cluster-6"><span class="header-section-number">6.4.5</span> Cluster 6</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a>merged.<span class="fl">18279.</span>skin.singlets <span class="ot">&lt;-</span> <span class="fu">FindSubCluster</span>(<span class="at">object =</span> merged.<span class="fl">18279.</span>skin.singlets, </span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a>                                        <span class="at">cluster =</span> <span class="st">"6"</span>,</span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true" tabindex="-1"></a>                                        <span class="at">resolution =</span> <span class="fl">0.2</span>,</span>
<span id="cb32-4"><a href="#cb32-4" aria-hidden="true" tabindex="-1"></a>                                        <span class="at">graph.name =</span> <span class="st">"RNA_snn"</span>,  </span>
<span id="cb32-5"><a href="#cb32-5" aria-hidden="true" tabindex="-1"></a>                                        <span class="at">algorithm =</span> <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 2186
Number of edges: 58961

Running Louvain algorithm with multilevel refinement...
Maximum modularity in 10 random starts: 0.8570
Number of communities: 2
Elapsed time: 0 seconds</code></pre>
</div>
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(merged.<span class="fl">18279.</span>skin.singlets,  </span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a>        <span class="at">reduction =</span> <span class="st">"umap.harmony"</span>,  </span>
<span id="cb34-3"><a href="#cb34-3" aria-hidden="true" tabindex="-1"></a>        <span class="at">group.by =</span> <span class="st">"sub.cluster"</span>, </span>
<span id="cb34-4"><a href="#cb34-4" aria-hidden="true" tabindex="-1"></a>        <span class="at">label =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Skin_scRNA_Part6_files/figure-html/unnamed-chunk-8-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(merged.<span class="fl">18279.</span>skin.singlets<span class="sc">$</span>sub.cluster)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
   0  1_0  1_1   10   11   12   13   14   15   16   17   18   19  2_0  2_1  2_2 
5768 2588 1100 1123 1086  907  569  557  472  407  360  308  280 1464  980  407 
  20   21   22   23   24   25   26    3  4_0  4_1  5_0  5_1  5_2  6_0  6_1    7 
 262  238  228  177  175   46   33 2418 1826  537 1465  564  171 1263  923 1746 
   8    9 
1483 1466 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a><span class="fu">Idents</span>(merged.<span class="fl">18279.</span>skin.singlets) <span class="ot">&lt;-</span> <span class="st">"sub.cluster"</span></span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a><span class="fu">levels</span>(merged.<span class="fl">18279.</span>skin.singlets)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "10"  "7"   "0"   "25"  "8"   "18"  "21"  "3"   "1_1" "13"  "14"  "17" 
[13] "23"  "1_0" "12"  "4_0" "11"  "9"   "5_0" "6_0" "6_1" "2_1" "2_0" "16" 
[25] "5_1" "19"  "2_2" "5_2" "22"  "24"  "15"  "20"  "4_1" "26" </code></pre>
</div>
</div>
</section>
<section id="cluster-9" class="level3" data-number="6.4.6">
<h3 data-number="6.4.6" class="anchored" data-anchor-id="cluster-9"><span class="header-section-number">6.4.6</span> Cluster 9</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb39"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a>merged.<span class="fl">18279.</span>skin.singlets <span class="ot">&lt;-</span> <span class="fu">FindSubCluster</span>(<span class="at">object =</span> merged.<span class="fl">18279.</span>skin.singlets, </span>
<span id="cb39-2"><a href="#cb39-2" aria-hidden="true" tabindex="-1"></a>                                        <span class="at">cluster =</span> <span class="st">"9"</span>,</span>
<span id="cb39-3"><a href="#cb39-3" aria-hidden="true" tabindex="-1"></a>                                        <span class="at">resolution =</span> <span class="fl">0.2</span>,</span>
<span id="cb39-4"><a href="#cb39-4" aria-hidden="true" tabindex="-1"></a>                                        <span class="at">graph.name =</span> <span class="st">"RNA_snn"</span>,  </span>
<span id="cb39-5"><a href="#cb39-5" aria-hidden="true" tabindex="-1"></a>                                        <span class="at">algorithm =</span> <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 1466
Number of edges: 53905

Running Louvain algorithm with multilevel refinement...
Maximum modularity in 10 random starts: 0.8001
Number of communities: 2
Elapsed time: 0 seconds</code></pre>
</div>
<div class="sourceCode cell-code" id="cb41"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(merged.<span class="fl">18279.</span>skin.singlets,  </span>
<span id="cb41-2"><a href="#cb41-2" aria-hidden="true" tabindex="-1"></a>        <span class="at">reduction =</span> <span class="st">"umap.harmony"</span>,  </span>
<span id="cb41-3"><a href="#cb41-3" aria-hidden="true" tabindex="-1"></a>        <span class="at">group.by =</span> <span class="st">"sub.cluster"</span>, </span>
<span id="cb41-4"><a href="#cb41-4" aria-hidden="true" tabindex="-1"></a>        <span class="at">label =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Skin_scRNA_Part6_files/figure-html/unnamed-chunk-9-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb42"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(merged.<span class="fl">18279.</span>skin.singlets<span class="sc">$</span>sub.cluster)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
   0  1_0  1_1   10   11   12   13   14   15   16   17   18   19  2_0  2_1  2_2 
5768 2588 1100 1123 1086  907  569  557  472  407  360  308  280 1464  980  407 
  20   21   22   23   24   25   26    3  4_0  4_1  5_0  5_1  5_2  6_0  6_1    7 
 262  238  228  177  175   46   33 2418 1826  537 1465  564  171 1263  923 1746 
   8  9_0  9_1 
1483 1310  156 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb44"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a><span class="fu">Idents</span>(merged.<span class="fl">18279.</span>skin.singlets) <span class="ot">&lt;-</span> <span class="st">"sub.cluster"</span></span>
<span id="cb44-2"><a href="#cb44-2" aria-hidden="true" tabindex="-1"></a><span class="fu">levels</span>(merged.<span class="fl">18279.</span>skin.singlets)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "10"  "7"   "0"   "25"  "8"   "18"  "21"  "3"   "1_1" "13"  "14"  "17" 
[13] "23"  "1_0" "12"  "4_0" "11"  "9_0" "5_0" "6_0" "6_1" "2_1" "2_0" "16" 
[25] "5_1" "19"  "2_2" "5_2" "22"  "24"  "15"  "20"  "4_1" "9_1" "26" </code></pre>
</div>
</div>
</section>
<section id="cluster-11" class="level3" data-number="6.4.7">
<h3 data-number="6.4.7" class="anchored" data-anchor-id="cluster-11"><span class="header-section-number">6.4.7</span> Cluster 11</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb46"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a>merged.<span class="fl">18279.</span>skin.singlets <span class="ot">&lt;-</span> <span class="fu">FindSubCluster</span>(<span class="at">object =</span> merged.<span class="fl">18279.</span>skin.singlets,</span>
<span id="cb46-2"><a href="#cb46-2" aria-hidden="true" tabindex="-1"></a>                                                                                <span class="at">cluster =</span> <span class="st">"11"</span>,</span>
<span id="cb46-3"><a href="#cb46-3" aria-hidden="true" tabindex="-1"></a>                                                                                <span class="at">resolution =</span> <span class="fl">0.2</span>,</span>
<span id="cb46-4"><a href="#cb46-4" aria-hidden="true" tabindex="-1"></a>                                                                                <span class="at">graph.name =</span> <span class="st">"RNA_snn"</span>,</span>
<span id="cb46-5"><a href="#cb46-5" aria-hidden="true" tabindex="-1"></a>                                                                                <span class="at">algorithm =</span> <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 1086
Number of edges: 32215

Running Louvain algorithm with multilevel refinement...
Maximum modularity in 10 random starts: 0.8363
Number of communities: 2
Elapsed time: 0 seconds</code></pre>
</div>
<div class="sourceCode cell-code" id="cb48"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(merged.<span class="fl">18279.</span>skin.singlets,</span>
<span id="cb48-2"><a href="#cb48-2" aria-hidden="true" tabindex="-1"></a>        <span class="at">reduction =</span> <span class="st">"umap.harmony"</span>,</span>
<span id="cb48-3"><a href="#cb48-3" aria-hidden="true" tabindex="-1"></a>        <span class="at">group.by =</span> <span class="st">"sub.cluster"</span>,</span>
<span id="cb48-4"><a href="#cb48-4" aria-hidden="true" tabindex="-1"></a>        <span class="at">label =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Skin_scRNA_Part6_files/figure-html/unnamed-chunk-10-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb49"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb49-1"><a href="#cb49-1" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(merged.<span class="fl">18279.</span>skin.singlets<span class="sc">$</span>sub.cluster)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
   0  1_0  1_1   10 11_0 11_1   12   13   14   15   16   17   18   19  2_0  2_1 
5768 2588 1100 1123  929  157  907  569  557  472  407  360  308  280 1464  980 
 2_2   20   21   22   23   24   25   26    3  4_0  4_1  5_0  5_1  5_2  6_0  6_1 
 407  262  238  228  177  175   46   33 2418 1826  537 1465  564  171 1263  923 
   7    8  9_0  9_1 
1746 1483 1310  156 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb51"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1" aria-hidden="true" tabindex="-1"></a><span class="fu">Idents</span>(merged.<span class="fl">18279.</span>skin.singlets) <span class="ot">&lt;-</span> <span class="st">"sub.cluster"</span></span>
<span id="cb51-2"><a href="#cb51-2" aria-hidden="true" tabindex="-1"></a><span class="fu">levels</span>(merged.<span class="fl">18279.</span>skin.singlets)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "10"   "7"    "0"    "25"   "8"    "18"   "21"   "3"    "1_1"  "13"  
[11] "14"   "17"   "23"   "1_0"  "12"   "4_0"  "11_0" "9_0"  "11_1" "5_0" 
[21] "6_0"  "6_1"  "2_1"  "2_0"  "16"   "5_1"  "19"   "2_2"  "5_2"  "22"  
[31] "24"   "15"   "20"   "4_1"  "9_1"  "26"  </code></pre>
</div>
</div>
</section>
<section id="cluster-13" class="level3" data-number="6.4.8">
<h3 data-number="6.4.8" class="anchored" data-anchor-id="cluster-13"><span class="header-section-number">6.4.8</span> Cluster 13</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb53"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb53-1"><a href="#cb53-1" aria-hidden="true" tabindex="-1"></a>merged.<span class="fl">18279.</span>skin.singlets <span class="ot">&lt;-</span> <span class="fu">FindSubCluster</span>(<span class="at">object =</span> merged.<span class="fl">18279.</span>skin.singlets,</span>
<span id="cb53-2"><a href="#cb53-2" aria-hidden="true" tabindex="-1"></a>                                                                                <span class="at">cluster =</span> <span class="st">"13"</span>,</span>
<span id="cb53-3"><a href="#cb53-3" aria-hidden="true" tabindex="-1"></a>                                                                                <span class="at">resolution =</span> <span class="fl">0.2</span>,</span>
<span id="cb53-4"><a href="#cb53-4" aria-hidden="true" tabindex="-1"></a>                                                                                <span class="at">graph.name =</span> <span class="st">"RNA_snn"</span>,</span>
<span id="cb53-5"><a href="#cb53-5" aria-hidden="true" tabindex="-1"></a>                                                                                <span class="at">algorithm =</span> <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 569
Number of edges: 18408

Running Louvain algorithm with multilevel refinement...
Maximum modularity in 10 random starts: 0.8667
Number of communities: 2
Elapsed time: 0 seconds</code></pre>
</div>
<div class="sourceCode cell-code" id="cb55"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb55-1"><a href="#cb55-1" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(merged.<span class="fl">18279.</span>skin.singlets,</span>
<span id="cb55-2"><a href="#cb55-2" aria-hidden="true" tabindex="-1"></a>        <span class="at">reduction =</span> <span class="st">"umap.harmony"</span>,</span>
<span id="cb55-3"><a href="#cb55-3" aria-hidden="true" tabindex="-1"></a>        <span class="at">group.by =</span> <span class="st">"sub.cluster"</span>,</span>
<span id="cb55-4"><a href="#cb55-4" aria-hidden="true" tabindex="-1"></a>        <span class="at">label =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Skin_scRNA_Part6_files/figure-html/unnamed-chunk-11-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb56"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb56-1"><a href="#cb56-1" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(merged.<span class="fl">18279.</span>skin.singlets<span class="sc">$</span>sub.cluster)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
   0  1_0  1_1   10 11_0 11_1   12 13_0 13_1   14   15   16   17   18   19  2_0 
5768 2588 1100 1123  929  157  907  314  255  557  472  407  360  308  280 1464 
 2_1  2_2   20   21   22   23   24   25   26    3  4_0  4_1  5_0  5_1  5_2  6_0 
 980  407  262  238  228  177  175   46   33 2418 1826  537 1465  564  171 1263 
 6_1    7    8  9_0  9_1 
 923 1746 1483 1310  156 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb58"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb58-1"><a href="#cb58-1" aria-hidden="true" tabindex="-1"></a><span class="fu">Idents</span>(merged.<span class="fl">18279.</span>skin.singlets) <span class="ot">&lt;-</span> <span class="st">"sub.cluster"</span></span>
<span id="cb58-2"><a href="#cb58-2" aria-hidden="true" tabindex="-1"></a><span class="fu">levels</span>(merged.<span class="fl">18279.</span>skin.singlets)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "10"   "7"    "0"    "25"   "8"    "18"   "21"   "3"    "1_1"  "13_1"
[11] "14"   "17"   "13_0" "23"   "1_0"  "12"   "4_0"  "11_0" "9_0"  "11_1"
[21] "5_0"  "6_0"  "6_1"  "2_1"  "2_0"  "16"   "5_1"  "19"   "2_2"  "5_2" 
[31] "22"   "24"   "15"   "20"   "4_1"  "9_1"  "26"  </code></pre>
</div>
</div>
</section>
<section id="cluster-19" class="level3" data-number="6.4.9">
<h3 data-number="6.4.9" class="anchored" data-anchor-id="cluster-19"><span class="header-section-number">6.4.9</span> Cluster 19</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb60"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb60-1"><a href="#cb60-1" aria-hidden="true" tabindex="-1"></a>merged.<span class="fl">18279.</span>skin.singlets <span class="ot">&lt;-</span> <span class="fu">FindSubCluster</span>(<span class="at">object =</span> merged.<span class="fl">18279.</span>skin.singlets,</span>
<span id="cb60-2"><a href="#cb60-2" aria-hidden="true" tabindex="-1"></a>                                                                                <span class="at">cluster =</span> <span class="st">"19"</span>,</span>
<span id="cb60-3"><a href="#cb60-3" aria-hidden="true" tabindex="-1"></a>                                                                                <span class="at">resolution =</span> <span class="fl">0.2</span>,</span>
<span id="cb60-4"><a href="#cb60-4" aria-hidden="true" tabindex="-1"></a>                                                                                <span class="at">graph.name =</span> <span class="st">"RNA_snn"</span>,</span>
<span id="cb60-5"><a href="#cb60-5" aria-hidden="true" tabindex="-1"></a>                                                                                <span class="at">algorithm =</span> <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 280
Number of edges: 10030

Running Louvain algorithm with multilevel refinement...
Maximum modularity in 10 random starts: 0.8000
Number of communities: 1
Elapsed time: 0 seconds</code></pre>
</div>
<div class="sourceCode cell-code" id="cb62"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb62-1"><a href="#cb62-1" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(merged.<span class="fl">18279.</span>skin.singlets,</span>
<span id="cb62-2"><a href="#cb62-2" aria-hidden="true" tabindex="-1"></a>        <span class="at">reduction =</span> <span class="st">"umap.harmony"</span>,</span>
<span id="cb62-3"><a href="#cb62-3" aria-hidden="true" tabindex="-1"></a>        <span class="at">group.by =</span> <span class="st">"sub.cluster"</span>,</span>
<span id="cb62-4"><a href="#cb62-4" aria-hidden="true" tabindex="-1"></a>        <span class="at">label =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Skin_scRNA_Part6_files/figure-html/unnamed-chunk-12-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb63"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb63-1"><a href="#cb63-1" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(merged.<span class="fl">18279.</span>skin.singlets<span class="sc">$</span>sub.cluster)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
   0  1_0  1_1   10 11_0 11_1   12 13_0 13_1   14   15   16   17   18 19_0  2_0 
5768 2588 1100 1123  929  157  907  314  255  557  472  407  360  308  280 1464 
 2_1  2_2   20   21   22   23   24   25   26    3  4_0  4_1  5_0  5_1  5_2  6_0 
 980  407  262  238  228  177  175   46   33 2418 1826  537 1465  564  171 1263 
 6_1    7    8  9_0  9_1 
 923 1746 1483 1310  156 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb65"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb65-1"><a href="#cb65-1" aria-hidden="true" tabindex="-1"></a><span class="fu">Idents</span>(merged.<span class="fl">18279.</span>skin.singlets) <span class="ot">&lt;-</span> <span class="st">"sub.cluster"</span></span>
<span id="cb65-2"><a href="#cb65-2" aria-hidden="true" tabindex="-1"></a><span class="fu">levels</span>(merged.<span class="fl">18279.</span>skin.singlets)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "10"   "7"    "0"    "25"   "8"    "18"   "21"   "3"    "1_1"  "13_1"
[11] "14"   "17"   "13_0" "23"   "1_0"  "12"   "4_0"  "11_0" "9_0"  "11_1"
[21] "5_0"  "6_0"  "6_1"  "2_1"  "2_0"  "16"   "5_1"  "19_0" "2_2"  "5_2" 
[31] "22"   "24"   "15"   "20"   "4_1"  "9_1"  "26"  </code></pre>
</div>
</div>
</section>
</section>
<section id="plot-updated-qc-metrics-per-clustersub-cluster" class="level2" data-number="6.5">
<h2 data-number="6.5" class="anchored" data-anchor-id="plot-updated-qc-metrics-per-clustersub-cluster"><span class="header-section-number">6.5</span> Plot updated QC metrics per cluster/sub-cluster</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb67"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb67-1"><a href="#cb67-1" aria-hidden="true" tabindex="-1"></a><span class="fu">VlnPlot</span>(merged.<span class="fl">18279.</span>skin.singlets, <span class="at">features =</span> <span class="st">"nCount_RNA"</span>, <span class="at">group.by=</span><span class="st">"sub.cluster"</span>, <span class="at">pt.size=</span><span class="dv">0</span>) <span class="sc">+</span> <span class="fu">NoLegend</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Skin_scRNA_Part6_files/figure-html/unnamed-chunk-13-1.png" class="img-fluid figure-img" width="1440"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb68"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb68-1"><a href="#cb68-1" aria-hidden="true" tabindex="-1"></a><span class="fu">VlnPlot</span>(merged.<span class="fl">18279.</span>skin.singlets, <span class="at">features =</span> <span class="st">"nFeature_RNA"</span>, <span class="at">group.by=</span><span class="st">"sub.cluster"</span>, <span class="at">pt.size=</span><span class="dv">0</span>) <span class="sc">+</span> <span class="fu">NoLegend</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Skin_scRNA_Part6_files/figure-html/unnamed-chunk-13-2.png" class="img-fluid figure-img" width="1440"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb69"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb69-1"><a href="#cb69-1" aria-hidden="true" tabindex="-1"></a><span class="fu">VlnPlot</span>(merged.<span class="fl">18279.</span>skin.singlets, <span class="at">features =</span> <span class="st">"percent.mt"</span>, <span class="at">group.by=</span><span class="st">"sub.cluster"</span>, <span class="at">pt.size=</span><span class="dv">0</span>) <span class="sc">+</span> <span class="fu">NoLegend</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Skin_scRNA_Part6_files/figure-html/unnamed-chunk-13-3.png" class="img-fluid figure-img" width="1440"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="identify-updated-cursory-marker-genes-per-clustersub-cluster" class="level2" data-number="6.6">
<h2 data-number="6.6" class="anchored" data-anchor-id="identify-updated-cursory-marker-genes-per-clustersub-cluster"><span class="header-section-number">6.6</span> Identify updated cursory marker genes per cluster/sub-cluster</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb70"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb70-1"><a href="#cb70-1" aria-hidden="true" tabindex="-1"></a><span class="fu">DefaultAssay</span>(merged.<span class="fl">18279.</span>skin.singlets) <span class="ot">&lt;-</span> <span class="st">"RNA"</span></span>
<span id="cb70-2"><a href="#cb70-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb70-3"><a href="#cb70-3" aria-hidden="true" tabindex="-1"></a>vargenes <span class="ot">&lt;-</span> presto<span class="sc">::</span><span class="fu">wilcoxauc</span>(merged.<span class="fl">18279.</span>skin.singlets, <span class="st">'sub.cluster'</span>, <span class="at">seurat_assay =</span> <span class="st">'RNA'</span>)</span>
<span id="cb70-4"><a href="#cb70-4" aria-hidden="true" tabindex="-1"></a>top_vargenes <span class="ot">&lt;-</span> <span class="fu">top_markers</span>(vargenes, <span class="at">n =</span> <span class="dv">100</span>, <span class="at">auc_min =</span> <span class="fl">0.5</span>, <span class="at">pct_in_min =</span> <span class="dv">50</span>, <span class="at">pct_out_max =</span> <span class="dv">50</span>)</span>
<span id="cb70-5"><a href="#cb70-5" aria-hidden="true" tabindex="-1"></a>top_vargenes</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 100 Ã— 38
    rank `0`     `1_0` `1_1` `10`  `11_0` `11_1` `12`  `13_0` `13_1` `14`  `15` 
   &lt;int&gt; &lt;chr&gt;   &lt;chr&gt; &lt;chr&gt; &lt;chr&gt; &lt;chr&gt;  &lt;chr&gt;  &lt;chr&gt; &lt;chr&gt;  &lt;chr&gt;  &lt;chr&gt; &lt;chr&gt;
 1     1 TYROBP  IL7R  ICOS  COL6â€¦ CNOT6L LDLRAâ€¦ DOCK4 STMN1  STMN1  ENSGâ€¦ LAMP3
 2     2 BCL2A1  LTB   CD2   DCN   PARP8  SKAP1  RBM47 HMGB2  TYMS   IL3RA FSCN1
 3     3 IL1RN   SPOCâ€¦ CD3E  COL6â€¦ SYNE2  ZEB1   FNDCâ€¦ TYMS   HMGB2  IRF4  DAPP1
 4     4 FCER1G  CD3E  CD3D  COL1â€¦ BCL11B SYNE2  MYO9B RRM2   GZMB   P2RY6 MARCâ€¦
 5     5 IFI30   KLF2  SPOCâ€¦ NNMT  SLC38â€¦ STAM   KYNU  DUT    CTSW   ENSGâ€¦ NUB1 
 6     6 MARCKS  LDHB  CLECâ€¦ CALD1 PRKCH  CASK   CYRIA MKI67  NKG7   PLD4  CERS6
 7     7 PLAUR   ICOS  TENTâ€¦ C1R   PPP1Râ€¦ IKZF2  ABCA1 PPP1CA GZMA   SLC1â€¦ CD83 
 8     8 S100A9  CD6   CD6   SELEâ€¦ CAMK4  CD247  DMXL2 CKS1B  GNLY   JCHAâ€¦ CCR7 
 9     9 ENSG00â€¦ CD3D  PDCD1 CXCLâ€¦ PITPNâ€¦ PRKCH  ATP1â€¦ PCLAF  RRM2   POLB  BASP1
10    10 PLEK    BCL1â€¦ BCL1â€¦ COL1â€¦ KLF12  RORA   FNIP2 H4C3   PRF1   FCHSâ€¦ CSF2â€¦
# â„¹ 90 more rows
# â„¹ 26 more variables: `16` &lt;chr&gt;, `17` &lt;chr&gt;, `18` &lt;chr&gt;, `19_0` &lt;chr&gt;,
#   `2_0` &lt;chr&gt;, `2_1` &lt;chr&gt;, `2_2` &lt;chr&gt;, `20` &lt;chr&gt;, `21` &lt;chr&gt;, `22` &lt;chr&gt;,
#   `23` &lt;chr&gt;, `24` &lt;chr&gt;, `25` &lt;chr&gt;, `26` &lt;chr&gt;, `3` &lt;chr&gt;, `4_0` &lt;chr&gt;,
#   `4_1` &lt;chr&gt;, `5_0` &lt;chr&gt;, `5_1` &lt;chr&gt;, `5_2` &lt;chr&gt;, `6_0` &lt;chr&gt;,
#   `6_1` &lt;chr&gt;, `7` &lt;chr&gt;, `8` &lt;chr&gt;, `9_0` &lt;chr&gt;, `9_1` &lt;chr&gt;</code></pre>
</div>
<div class="sourceCode cell-code" id="cb72"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb72-1"><a href="#cb72-1" aria-hidden="true" tabindex="-1"></a><span class="fu">write_tsv</span>(top_vargenes,<span class="st">"Skin_scRNA_prestoMarkers.tsv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="plot-top-markers-identified-and-canonical-genes-as-a-dotplot" class="level2" data-number="6.7">
<h2 data-number="6.7" class="anchored" data-anchor-id="plot-top-markers-identified-and-canonical-genes-as-a-dotplot"><span class="header-section-number">6.7</span> Plot top markers identified and canonical genes as a dotplot</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb73"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb73-1"><a href="#cb73-1" aria-hidden="true" tabindex="-1"></a>tumor <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"DCT"</span>,<span class="st">"MLANA"</span>,<span class="st">"MITF"</span>,<span class="st">"PMEL"</span>,<span class="st">"S100A1"</span>,<span class="st">"TYR"</span>,<span class="st">"APOC1"</span>)</span>
<span id="cb73-2"><a href="#cb73-2" aria-hidden="true" tabindex="-1"></a>endothelial <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"PECAM1"</span>,<span class="st">"VWF"</span>)</span>
<span id="cb73-3"><a href="#cb73-3" aria-hidden="true" tabindex="-1"></a>fibroblast <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"COL3A1"</span>,<span class="st">"COL1A1"</span>,<span class="st">"COL1A2"</span>,<span class="st">"LUM"</span>)</span>
<span id="cb73-4"><a href="#cb73-4" aria-hidden="true" tabindex="-1"></a>tcell <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"FGFBP2"</span>,<span class="st">"KLRD1"</span>,<span class="st">"CD3E"</span>,<span class="st">"CD3D"</span>,<span class="st">"GZMB"</span>,<span class="st">"XCL2"</span>,<span class="st">"GZMH"</span>,<span class="st">"CST7"</span>,<span class="st">"GZMK"</span>,<span class="st">"GZMA"</span>,<span class="st">"IFNG"</span>,<span class="st">"GNLY"</span>,<span class="st">"CCL4"</span>,<span class="st">"NKG7"</span>,<span class="st">"CCL5"</span>,<span class="st">"CD8A"</span>,<span class="st">"CD8B"</span>,<span class="st">"CTLA4"</span>,<span class="st">"TNFRSF4"</span>,<span class="st">"BATF"</span>,<span class="st">"ITM2A"</span>)</span>
<span id="cb73-5"><a href="#cb73-5" aria-hidden="true" tabindex="-1"></a>mono <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"LYZ"</span>,<span class="st">"CD74"</span>,<span class="st">"CD68"</span>)</span>
<span id="cb73-6"><a href="#cb73-6" aria-hidden="true" tabindex="-1"></a>bcell <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"MS4A1"</span>,<span class="st">"CD79A"</span>)</span>
<span id="cb73-7"><a href="#cb73-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb73-8"><a href="#cb73-8" aria-hidden="true" tabindex="-1"></a>top_vargenes <span class="ot">&lt;-</span> <span class="fu">top_markers</span>(vargenes, <span class="at">n =</span> <span class="dv">5</span>, <span class="at">auc_min =</span> <span class="fl">0.5</span>, <span class="at">pct_in_min =</span> <span class="dv">50</span>, <span class="at">pct_out_max =</span> <span class="dv">50</span>)</span>
<span id="cb73-9"><a href="#cb73-9" aria-hidden="true" tabindex="-1"></a>top_markers <span class="ot">&lt;-</span> top_vargenes <span class="sc">%&gt;%</span></span>
<span id="cb73-10"><a href="#cb73-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">select</span>(<span class="sc">-</span>rank) <span class="sc">%&gt;%</span> </span>
<span id="cb73-11"><a href="#cb73-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">unclass</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb73-12"><a href="#cb73-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">stack</span>() <span class="sc">%&gt;%</span></span>
<span id="cb73-13"><a href="#cb73-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pull</span>(values) <span class="sc">%&gt;%</span></span>
<span id="cb73-14"><a href="#cb73-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">unique</span>() <span class="sc">%&gt;%</span></span>
<span id="cb73-15"><a href="#cb73-15" aria-hidden="true" tabindex="-1"></a>    .[<span class="sc">!</span><span class="fu">is.na</span>(.)]</span>
<span id="cb73-16"><a href="#cb73-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb73-17"><a href="#cb73-17" aria-hidden="true" tabindex="-1"></a>dotplotmarkers <span class="ot">&lt;-</span> <span class="fu">unique</span>(<span class="fu">c</span>(top_markers,tumor,endothelial,fibroblast,tcell,mono,bcell))</span>
<span id="cb73-18"><a href="#cb73-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb73-19"><a href="#cb73-19" aria-hidden="true" tabindex="-1"></a><span class="co"># Compute aggregated expression values of these genes and cluster them to order the figure</span></span>
<span id="cb73-20"><a href="#cb73-20" aria-hidden="true" tabindex="-1"></a>rna <span class="ot">&lt;-</span> <span class="fu">AverageExpression</span>(merged.<span class="fl">18279.</span>skin.singlets,<span class="at">assay=</span><span class="st">"RNA"</span>,<span class="at">slot=</span><span class="st">"data"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>As of Seurat v5, we recommend using AggregateExpression to perform pseudo-bulk analysis.
Names of identity class contain underscores ('_'), replacing with dashes ('-')
First group.by variable `ident` starts with a number, appending `g` to ensure valid variable names
This message is displayed once per session.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb75"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb75-1"><a href="#cb75-1" aria-hidden="true" tabindex="-1"></a>rna.sub <span class="ot">&lt;-</span> rna<span class="sc">$</span>RNA[dotplotmarkers,]</span>
<span id="cb75-2"><a href="#cb75-2" aria-hidden="true" tabindex="-1"></a>cors.genes <span class="ot">&lt;-</span> <span class="fu">as.dist</span>(<span class="dv">1</span><span class="sc">-</span><span class="fu">cor</span>(<span class="fu">as.matrix</span>(<span class="fu">t</span>(rna.sub)),<span class="at">method=</span><span class="st">"pearson"</span>))</span>
<span id="cb75-3"><a href="#cb75-3" aria-hidden="true" tabindex="-1"></a>hc.genes <span class="ot">&lt;-</span> <span class="fu">hclust</span>(cors.genes)</span>
<span id="cb75-4"><a href="#cb75-4" aria-hidden="true" tabindex="-1"></a>dotplotmarkers.sorted <span class="ot">&lt;-</span> <span class="fu">rownames</span>(rna.sub)[hc.genes<span class="sc">$</span>order]</span>
<span id="cb75-5"><a href="#cb75-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb75-6"><a href="#cb75-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Also re-order the sub-clusters</span></span>
<span id="cb75-7"><a href="#cb75-7" aria-hidden="true" tabindex="-1"></a>cors.subs <span class="ot">&lt;-</span> <span class="fu">as.dist</span>(<span class="dv">1</span><span class="sc">-</span><span class="fu">cor</span>(<span class="fu">as.matrix</span>(rna.sub),<span class="at">method=</span><span class="st">"pearson"</span>))</span>
<span id="cb75-8"><a href="#cb75-8" aria-hidden="true" tabindex="-1"></a>hc.subs <span class="ot">&lt;-</span> <span class="fu">hclust</span>(cors.subs)</span>
<span id="cb75-9"><a href="#cb75-9" aria-hidden="true" tabindex="-1"></a>merged.<span class="fl">18279.</span>skin.singlets<span class="sc">@</span>active.ident <span class="ot">&lt;-</span> <span class="fu">factor</span>(merged.<span class="fl">18279.</span>skin.singlets<span class="sc">@</span>active.ident, </span>
<span id="cb75-10"><a href="#cb75-10" aria-hidden="true" tabindex="-1"></a>                            <span class="at">levels=</span><span class="fu">str_replace_all</span>(<span class="fu">str_replace_all</span>(<span class="fu">colnames</span>(rna.sub)[hc.subs<span class="sc">$</span>order],<span class="st">"g"</span>,<span class="st">""</span>),<span class="st">"-"</span>,<span class="st">"_"</span>))</span>
<span id="cb75-11"><a href="#cb75-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb75-12"><a href="#cb75-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb75-13"><a href="#cb75-13" aria-hidden="true" tabindex="-1"></a><span class="fu">DotPlot</span>(merged.<span class="fl">18279.</span>skin.singlets,</span>
<span id="cb75-14"><a href="#cb75-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">features =</span> dotplotmarkers.sorted,</span>
<span id="cb75-15"><a href="#cb75-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">assay =</span> <span class="st">"RNA"</span>,</span>
<span id="cb75-16"><a href="#cb75-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">cols =</span> <span class="fu">c</span>(<span class="st">"blue"</span>,<span class="st">"red"</span>)</span>
<span id="cb75-17"><a href="#cb75-17" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb75-18"><a href="#cb75-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">RotatedAxis</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Skin_scRNA_Part6_files/figure-html/unnamed-chunk-15-1.png" class="img-fluid figure-img" width="3360"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="plot-updated-skin-sample-proportions-pre-vs-post-3rd-vaccine-per-cluster" class="level2" data-number="6.8">
<h2 data-number="6.8" class="anchored" data-anchor-id="plot-updated-skin-sample-proportions-pre-vs-post-3rd-vaccine-per-cluster"><span class="header-section-number">6.8</span> Plot updated skin sample proportions, pre vs post 3rd vaccine, per cluster</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb76"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb76-1"><a href="#cb76-1" aria-hidden="true" tabindex="-1"></a><span class="fu">as.data.frame</span>(<span class="fu">table</span>(merged.<span class="fl">18279.</span>skin.singlets<span class="sc">$</span>sub.cluster, merged.<span class="fl">18279.</span>skin.singlets<span class="sc">$</span>Sample)) <span class="sc">%&gt;%</span> </span>
<span id="cb76-2"><a href="#cb76-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb76-3"><a href="#cb76-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"Sample"</span> <span class="ot">=</span> Var2,<span class="st">"Cluster"</span> <span class="ot">=</span> Var1) <span class="sc">%&gt;%</span></span>
<span id="cb76-4"><a href="#cb76-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(Sample, <span class="at">into=</span><span class="fu">c</span>(<span class="st">"Patient"</span>,<span class="st">"Site"</span>,<span class="st">"Timepoint"</span>,<span class="st">"IpiCohort"</span>,<span class="st">"Assay"</span>),<span class="at">sep=</span><span class="st">"_"</span>,<span class="at">remove=</span>F) <span class="sc">%&gt;%</span></span>
<span id="cb76-5"><a href="#cb76-5" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(Timepoint<span class="sc">==</span><span class="st">"Pre3rd"</span> <span class="sc">|</span> Timepoint<span class="sc">==</span><span class="st">"Post3rd"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb76-6"><a href="#cb76-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Sample) <span class="sc">%&gt;%</span></span>
<span id="cb76-7"><a href="#cb76-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Proportion =</span> Freq <span class="sc">/</span> <span class="fu">sum</span>(Freq)) <span class="sc">%&gt;%</span></span>
<span id="cb76-8"><a href="#cb76-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">fill =</span> Timepoint, <span class="at">x =</span> <span class="fu">fct_relevel</span>(Timepoint,<span class="fu">c</span>(<span class="st">"Pre3rd"</span>,<span class="st">"Post3rd"</span>)), <span class="at">y=</span>Proportion)) <span class="sc">+</span></span>
<span id="cb76-9"><a href="#cb76-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">outlier.shape=</span><span class="cn">NA</span>) <span class="sc">+</span></span>
<span id="cb76-10"><a href="#cb76-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">pch =</span> <span class="dv">21</span>, <span class="at">position =</span> <span class="fu">position_jitterdodge</span>(<span class="at">jitter.width=</span><span class="fl">0.2</span>)) <span class="sc">+</span></span>
<span id="cb76-11"><a href="#cb76-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>Cluster,<span class="at">scales=</span><span class="st">"free"</span>) <span class="sc">+</span></span>
<span id="cb76-12"><a href="#cb76-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">"Timepoint"</span>) <span class="sc">+</span></span>
<span id="cb76-13"><a href="#cb76-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_discrete</span>(<span class="at">breaks=</span><span class="fu">c</span>(<span class="st">"Pre3rd"</span>,<span class="st">"Post3rd"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Skin_scRNA_Part6_files/figure-html/unnamed-chunk-16-1.png" class="img-fluid figure-img" width="960"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="save-updated-object" class="level2" data-number="6.9">
<h2 data-number="6.9" class="anchored" data-anchor-id="save-updated-object"><span class="header-section-number">6.9</span> Save updated object</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb77"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb77-1"><a href="#cb77-1" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(merged.<span class="fl">18279.</span>skin.singlets, <span class="st">"Skin_scRNA_Part6.rds"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="get-session-info" class="level2" data-number="6.10">
<h2 data-number="6.10" class="anchored" data-anchor-id="get-session-info"><span class="header-section-number">6.10</span> Get session info</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb78"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb78-1"><a href="#cb78-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>R version 4.3.1 (2023-06-16)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Rocky Linux 8.10 (Green Obsidian)

Matrix products: default
BLAS/LAPACK: /usr/lib64/libopenblasp-r0.3.15.so;  LAPACK version 3.9.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

time zone: America/New_York
tzcode source: system (glibc)

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] cellXY_0.99.0               scDblFinder_1.14.0         
 [3] harmony_1.2.0               alevinQC_1.16.1            
 [5] vctrs_0.6.5                 patchwork_1.3.0            
 [7] scater_1.28.0               scuttle_1.10.3             
 [9] speckle_1.0.0               Matrix_1.6-4               
[11] fishpond_2.6.2              readxl_1.4.3               
[13] SingleCellExperiment_1.22.0 SummarizedExperiment_1.30.2
[15] Biobase_2.60.0              GenomicRanges_1.52.1       
[17] GenomeInfoDb_1.36.4         IRanges_2.34.1             
[19] S4Vectors_0.38.2            BiocGenerics_0.46.0        
[21] MatrixGenerics_1.12.3       matrixStats_1.2.0          
[23] flexmix_2.3-19              lattice_0.22-5             
[25] SeuratWrappers_0.3.19       miQC_1.8.0                 
[27] lubridate_1.9.3             forcats_1.0.0              
[29] stringr_1.5.1               dplyr_1.1.4                
[31] purrr_1.0.2                 readr_2.1.5                
[33] tidyr_1.3.1                 tibble_3.2.1               
[35] ggplot2_3.4.4               tidyverse_2.0.0            
[37] Seurat_5.1.0                SeuratObject_5.0.2         
[39] sp_2.1-3                    sctransform_0.4.1          
[41] glmGamPoi_1.12.2            presto_1.0.0               
[43] Rcpp_1.0.12                 devtools_2.4.5             
[45] usethis_2.2.2               data.table_1.15.0          

loaded via a namespace (and not attached):
  [1] fs_1.6.3                  spatstat.sparse_3.0-3    
  [3] bitops_1.0-7              httr_1.4.7               
  [5] RColorBrewer_1.1-3        profvis_0.3.8            
  [7] tools_4.3.1               utf8_1.2.4               
  [9] R6_2.5.1                  DT_0.31                  
 [11] lazyeval_0.2.2            uwot_0.1.16              
 [13] urlchecker_1.0.1          withr_3.0.0              
 [15] GGally_2.2.1              gridExtra_2.3            
 [17] progressr_0.14.0          cli_3.6.2                
 [19] spatstat.explore_3.2-6    fastDummies_1.7.3        
 [21] labeling_0.4.3            spatstat.data_3.0-4      
 [23] ggridges_0.5.6            pbapply_1.7-2            
 [25] Rsamtools_2.16.0          R.utils_2.12.3           
 [27] parallelly_1.37.0         sessioninfo_1.2.2        
 [29] limma_3.56.2              RSQLite_2.3.5            
 [31] BiocIO_1.10.0             generics_0.1.3           
 [33] vroom_1.6.5               gtools_3.9.5             
 [35] ica_1.0-3                 spatstat.random_3.2-2    
 [37] ggbeeswarm_0.7.2          fansi_1.0.6              
 [39] abind_1.4-5               R.methodsS3_1.8.2        
 [41] lifecycle_1.0.4           yaml_2.3.8               
 [43] edgeR_3.42.4              Rtsne_0.17               
 [45] blob_1.2.4                grid_4.3.1               
 [47] dqrng_0.3.2               promises_1.2.1           
 [49] crayon_1.5.2              shinydashboard_0.7.2     
 [51] miniUI_0.1.1.1            beachmat_2.16.0          
 [53] cowplot_1.1.3             KEGGREST_1.40.1          
 [55] metapod_1.8.0             pillar_1.9.0             
 [57] knitr_1.45                rjson_0.2.21             
 [59] xgboost_1.7.7.1           future.apply_1.11.1      
 [61] codetools_0.2-19          leiden_0.4.3.1           
 [63] glue_1.7.0                remotes_2.4.2.1          
 [65] png_0.1-8                 spam_2.10-0              
 [67] org.Mm.eg.db_3.18.0       cellranger_1.1.0         
 [69] gtable_0.3.4              cachem_1.0.8             
 [71] xfun_0.42                 S4Arrays_1.2.0           
 [73] mime_0.12                 survival_3.5-8           
 [75] statmod_1.5.0             bluster_1.10.0           
 [77] ellipsis_0.3.2            fitdistrplus_1.1-11      
 [79] ROCR_1.0-11               nlme_3.1-164             
 [81] bit64_4.0.5               RcppAnnoy_0.0.22         
 [83] irlba_2.3.5.1             vipor_0.4.7              
 [85] KernSmooth_2.23-22        DBI_1.2.2                
 [87] colorspace_2.1-0          nnet_7.3-19              
 [89] ggrastr_1.0.2             tidyselect_1.2.0         
 [91] bit_4.0.5                 compiler_4.3.1           
 [93] BiocNeighbors_1.18.0      DelayedArray_0.26.7      
 [95] plotly_4.10.4             rtracklayer_1.60.1       
 [97] scales_1.3.0              lmtest_0.9-40            
 [99] digest_0.6.34             goftest_1.2-3            
[101] spatstat.utils_3.0-4      rmarkdown_2.25           
[103] XVector_0.40.0            htmltools_0.5.7          
[105] pkgconfig_2.0.3           sparseMatrixStats_1.12.2 
[107] fastmap_1.1.1             rlang_1.1.3              
[109] htmlwidgets_1.6.4         shiny_1.8.0              
[111] DelayedMatrixStats_1.22.6 farver_2.1.1             
[113] zoo_1.8-12                jsonlite_1.8.8           
[115] BiocParallel_1.34.2       R.oo_1.26.0              
[117] BiocSingular_1.16.0       RCurl_1.98-1.14          
[119] magrittr_2.0.3            modeltools_0.2-23        
[121] GenomeInfoDbData_1.2.10   dotCall64_1.1-1          
[123] munsell_0.5.0             viridis_0.6.5            
[125] reticulate_1.35.0         stringi_1.8.3            
[127] zlibbioc_1.46.0           MASS_7.3-60.0.1          
[129] org.Hs.eg.db_3.18.0       plyr_1.8.9               
[131] pkgbuild_1.4.3            ggstats_0.5.1            
[133] parallel_4.3.1            listenv_0.9.1            
[135] ggrepel_0.9.5             deldir_2.0-2             
[137] Biostrings_2.68.1         splines_4.3.1            
[139] tensor_1.5                hms_1.1.3                
[141] locfit_1.5-9.8            igraph_2.0.2             
[143] spatstat.geom_3.2-8       RcppHNSW_0.6.0           
[145] reshape2_1.4.4            ScaledMatrix_1.8.1       
[147] pkgload_1.3.4             XML_3.99-0.16.1          
[149] evaluate_0.23             scran_1.28.2             
[151] BiocManager_1.30.22       tzdb_0.4.0               
[153] httpuv_1.6.14             RANN_2.6.1               
[155] polyclip_1.10-6           future_1.33.1            
[157] scattermore_1.2           rsvd_1.0.5               
[159] xtable_1.8-4              restfulr_0.0.15          
[161] svMisc_1.2.3              RSpectra_0.16-1          
[163] later_1.3.2               viridisLite_0.4.2        
[165] AnnotationDbi_1.64.1      GenomicAlignments_1.36.0 
[167] memoise_2.0.1             beeswarm_0.4.0           
[169] tximport_1.28.0           cluster_2.1.6            
[171] timechange_0.3.0          globals_0.16.2           </code></pre>
</div>
</div>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "î§‹";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="./Skin_scRNA_Part5.html" class="pagination-link" aria-label="Skin: Post-clustering comprehensive marker identification">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">Skin: Post-clustering comprehensive marker identification</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="./Skin_scRNA_Part7.html" class="pagination-link" aria-label="Skin: Assisting cell type annotation with enrichR">
        <span class="nav-page-text"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">Skin: Assisting cell type annotation with enrichR</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->




</body></html>