<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.6.39">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>8&nbsp; Skin: Classification of CD4 and CD8 T cell states within T/NK lineage â€“ Blass &amp; Keskin (2025): Melanoma NeoVax analysis: Skin single-cell RNAseq</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./Skin_scRNA_Part9.html" rel="next">
<link href="./Skin_scRNA_Part7.html" rel="prev">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-e26003cea8cd680ca0c55a263523d882.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap-41a4eb7d26fa9333a8f88b12b94df59e.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


</head>

<body class="nav-sidebar floating">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./Skin_scRNA_Part8.html"><span class="chapter-number">8</span>&nbsp; <span class="chapter-title">Skin: Classification of CD4 and CD8 T cell states within T/NK lineage</span></a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="./">Blass &amp; Keskin (2025): Melanoma NeoVax analysis: Skin single-cell RNAseq</a> 
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Introduction</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Skin_scRNA_Part1.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Skin: QC &amp; Filtering</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Skin_scRNA_Part2.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Skin: Normalization, doublet discrimination, integration, clustering</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Skin_scRNA_Part3.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Skin: Check patient sexes</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Skin_scRNA_Part4.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Skin: Post-clustering QC and characterization</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Skin_scRNA_Part5.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">Skin: Post-clustering comprehensive marker identification</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Skin_scRNA_Part6.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">Skin: Sub-clustering on T/NK lineage</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Skin_scRNA_Part7.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">Skin: Assisting cell type annotation with enrichR</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Skin_scRNA_Part8.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text"><span class="chapter-number">8</span>&nbsp; <span class="chapter-title">Skin: Classification of CD4 and CD8 T cell states within T/NK lineage</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Skin_scRNA_Part9.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">9</span>&nbsp; <span class="chapter-title">Skin: Differential expression analysis, Post vs Pre 3rd vaccination</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Skin_scRNA_Part10.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">10</span>&nbsp; <span class="chapter-title">Skin: Deep phenotyping, cluster annotation, and exploration of Post vs Pre 3rd vaccination differences in cell proportions and expression</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Skin_scRNA_Part11.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">11</span>&nbsp; <span class="chapter-title">Skin: Calculating Post vs Pre 3rd vaccination differences in pathway enrichments with fgsea</span></span></a>
  </div>
</li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#set-up-seurat-workspace" id="toc-set-up-seurat-workspace" class="nav-link active" data-scroll-target="#set-up-seurat-workspace"><span class="header-section-number">8.1</span> Set up Seurat workspace</a></li>
  <li><a href="#load-previously-saved-sub-clustered-object" id="toc-load-previously-saved-sub-clustered-object" class="nav-link" data-scroll-target="#load-previously-saved-sub-clustered-object"><span class="header-section-number">8.2</span> Load previously saved sub-clustered object</a></li>
  <li><a href="#run-cd8-and-cd4-projection-with-projectils-as-in-this-vignette" id="toc-run-cd8-and-cd4-projection-with-projectils-as-in-this-vignette" class="nav-link" data-scroll-target="#run-cd8-and-cd4-projection-with-projectils-as-in-this-vignette"><span class="header-section-number">8.3</span> Run CD8 and CD4 projection with <code>projecTILs</code> as in this vignette</a>
  <ul class="collapse">
  <li><a href="#retrieve-human-cd8-and-cd4-reference-maps" id="toc-retrieve-human-cd8-and-cd4-reference-maps" class="nav-link" data-scroll-target="#retrieve-human-cd8-and-cd4-reference-maps"><span class="header-section-number">8.3.1</span> Retrieve human CD8 and CD4 reference maps</a></li>
  <li><a href="#load-reference-maps" id="toc-load-reference-maps" class="nav-link" data-scroll-target="#load-reference-maps"><span class="header-section-number">8.3.2</span> Load reference maps</a></li>
  <li><a href="#plot-reference-atlases" id="toc-plot-reference-atlases" class="nav-link" data-scroll-target="#plot-reference-atlases"><span class="header-section-number">8.3.3</span> Plot reference atlases</a></li>
  <li><a href="#classify-cd8-t-subtypes" id="toc-classify-cd8-t-subtypes" class="nav-link" data-scroll-target="#classify-cd8-t-subtypes"><span class="header-section-number">8.3.4</span> Classify CD8 T subtypes</a></li>
  <li><a href="#classify-cd4-t-subtypes" id="toc-classify-cd4-t-subtypes" class="nav-link" data-scroll-target="#classify-cd4-t-subtypes"><span class="header-section-number">8.3.5</span> Classify CD4 T subtypes</a></li>
  </ul></li>
  <li><a href="#tabulate-and-visualize-projections" id="toc-tabulate-and-visualize-projections" class="nav-link" data-scroll-target="#tabulate-and-visualize-projections"><span class="header-section-number">8.4</span> Tabulate and visualize projections</a>
  <ul class="collapse">
  <li><a href="#count-total-cells-by-t-cell-state" id="toc-count-total-cells-by-t-cell-state" class="nav-link" data-scroll-target="#count-total-cells-by-t-cell-state"><span class="header-section-number">8.4.1</span> Count total cells by T cell state</a></li>
  <li><a href="#tabulate-which-t-cell-state-was-most-abundant-in-each-subcluster" id="toc-tabulate-which-t-cell-state-was-most-abundant-in-each-subcluster" class="nav-link" data-scroll-target="#tabulate-which-t-cell-state-was-most-abundant-in-each-subcluster"><span class="header-section-number">8.4.2</span> Tabulate which T cell state was most abundant in each subcluster</a></li>
  <li><a href="#plot-umap-of-all-cells-labeled-by-projected-t-cell-state" id="toc-plot-umap-of-all-cells-labeled-by-projected-t-cell-state" class="nav-link" data-scroll-target="#plot-umap-of-all-cells-labeled-by-projected-t-cell-state"><span class="header-section-number">8.4.3</span> Plot UMAP of all cells labeled by projected T cell state</a></li>
  <li><a href="#plot-just-tnk-cell-lineage" id="toc-plot-just-tnk-cell-lineage" class="nav-link" data-scroll-target="#plot-just-tnk-cell-lineage"><span class="header-section-number">8.4.4</span> Plot just T/NK cell lineage</a></li>
  </ul></li>
  <li><a href="#save-modified-object" id="toc-save-modified-object" class="nav-link" data-scroll-target="#save-modified-object"><span class="header-section-number">8.5</span> Save modified object</a></li>
  <li><a href="#get-session-info" id="toc-get-session-info" class="nav-link" data-scroll-target="#get-session-info"><span class="header-section-number">8.6</span> Get session info</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title"><span id="sec-skin-projectils" class="quarto-section-identifier"><span class="chapter-number">8</span>&nbsp; <span class="chapter-title">Skin: Classification of CD4 and CD8 T cell states within T/NK lineage</span></span></h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<section id="set-up-seurat-workspace" class="level2" data-number="8.1">
<h2 data-number="8.1" class="anchored" data-anchor-id="set-up-seurat-workspace"><span class="header-section-number">8.1</span> Set up Seurat workspace</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load libraries</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(data.table)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(devtools)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(presto) </span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(glmGamPoi)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(sctransform) </span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Seurat)</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(miQC) </span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(SeuratWrappers)</span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(flexmix)</span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(SingleCellExperiment)</span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(SummarizedExperiment)</span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readxl)</span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(fishpond)</span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Matrix)</span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(speckle)</span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(scater)</span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(vctrs)</span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(alevinQC)</span>
<span id="cb1-22"><a href="#cb1-22" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(harmony)</span>
<span id="cb1-23"><a href="#cb1-23" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(scDblFinder)</span>
<span id="cb1-24"><a href="#cb1-24" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(cellXY)</span>
<span id="cb1-25"><a href="#cb1-25" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(STACAS)</span>
<span id="cb1-26"><a href="#cb1-26" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ProjecTILs)</span>
<span id="cb1-27"><a href="#cb1-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-28"><a href="#cb1-28" aria-hidden="true" tabindex="-1"></a><span class="co"># Set global options for Seurat v5 objects</span></span>
<span id="cb1-29"><a href="#cb1-29" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">Seurat.object.assay.version =</span> <span class="st">'v5'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="load-previously-saved-sub-clustered-object" class="level2" data-number="8.2">
<h2 data-number="8.2" class="anchored" data-anchor-id="load-previously-saved-sub-clustered-object"><span class="header-section-number">8.2</span> Load previously saved sub-clustered object</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>merged.<span class="fl">18279.</span>skin.singlets <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"Skin_scRNA_Part6.rds"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="run-cd8-and-cd4-projection-with-projectils-as-in-this-vignette" class="level2" data-number="8.3">
<h2 data-number="8.3" class="anchored" data-anchor-id="run-cd8-and-cd4-projection-with-projectils-as-in-this-vignette"><span class="header-section-number">8.3</span> Run CD8 and CD4 projection with <code>projecTILs</code> as in <a href="https://carmonalab.github.io/ProjecTILs_CaseStudies/Bassez_BC.html">this vignette</a></h2>
<p>Note these classifications are limited only to CD4/CD8 and do not include NK cells</p>
<section id="retrieve-human-cd8-and-cd4-reference-maps" class="level3" data-number="8.3.1">
<h3 data-number="8.3.1" class="anchored" data-anchor-id="retrieve-human-cd8-and-cd4-reference-maps"><span class="header-section-number">8.3.1</span> Retrieve human CD8 and CD4 reference maps</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">timeout =</span> <span class="fu">max</span>(<span class="dv">900</span>, <span class="fu">getOption</span>(<span class="st">"timeout"</span>)))</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="fu">download.file</span>(<span class="st">"https://figshare.com/ndownloader/files/41414556"</span>, <span class="at">destfile =</span> <span class="st">"CD8T_human_ref_v1.rds"</span>)</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a><span class="fu">download.file</span>(<span class="st">"https://figshare.com/ndownloader/files/43794750"</span>, <span class="at">destfile =</span> <span class="st">"CD4T_human_ref_v2.rds"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="load-reference-maps" class="level3" data-number="8.3.2">
<h3 data-number="8.3.2" class="anchored" data-anchor-id="load-reference-maps"><span class="header-section-number">8.3.2</span> Load reference maps</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>ref.cd8 <span class="ot">&lt;-</span> <span class="fu">load.reference.map</span>(<span class="st">"CD8T_human_ref_v1.rds"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Loading Custom Reference Atlas..."
[1] "Loaded Custom Reference map Human CD8 TILs"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>ref.cd4 <span class="ot">&lt;-</span> <span class="fu">load.reference.map</span>(<span class="st">"CD4T_human_ref_v2.rds"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Loading Custom Reference Atlas..."
[1] "Loaded Custom Reference map custom_reference"</code></pre>
</div>
</div>
</section>
<section id="plot-reference-atlases" class="level3" data-number="8.3.3">
<h3 data-number="8.3.3" class="anchored" data-anchor-id="plot-reference-atlases"><span class="header-section-number">8.3.3</span> Plot reference atlases</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>a <span class="ot">&lt;-</span> <span class="fu">DimPlot</span>(ref.cd8, <span class="at">cols =</span> ref.cd8<span class="sc">@</span>misc<span class="sc">$</span>atlas.palette, <span class="at">label =</span> T) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">aspect.ratio =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggtitle</span>(<span class="st">"CD8 T reference"</span>) <span class="sc">+</span> <span class="fu">NoLegend</span>()</span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>b <span class="ot">&lt;-</span> <span class="fu">DimPlot</span>(ref.cd4, <span class="at">cols =</span> ref.cd4<span class="sc">@</span>misc<span class="sc">$</span>atlas.palette, <span class="at">label =</span> T) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">aspect.ratio =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggtitle</span>(<span class="st">"CD4 T reference"</span>) <span class="sc">+</span> <span class="fu">NoLegend</span>()</span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a>a <span class="sc">|</span> b</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Skin_scRNA_Part8_files/figure-html/unnamed-chunk-5-1.png" class="img-fluid figure-img" width="960"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="classify-cd8-t-subtypes" class="level3" data-number="8.3.4">
<h3 data-number="8.3.4" class="anchored" data-anchor-id="classify-cd8-t-subtypes"><span class="header-section-number">8.3.4</span> Classify CD8 T subtypes</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>merged.<span class="fl">18279.</span>skin.singlets <span class="ot">&lt;-</span> <span class="fu">ProjecTILs.classifier</span>(merged.<span class="fl">18279.</span>skin.singlets, ref.cd8, <span class="at">ncores =</span> <span class="dv">8</span>, <span class="at">split.by =</span> <span class="st">"Sample"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>
  |                                                                            
  |                                                                      |   0%[1] "Using assay RNA for query"
[1] "383 out of 437 ( 88% ) non-pure cells removed. Use filter.cells=FALSE to avoid pre-filtering"
Performing log-normalization
0%   10   20   30   40   50   60   70   80   90   100%
[----|----|----|----|----|----|----|----|----|----|
**************************************************|
[1] "Aligning query to reference map for batch-correction..."

Projecting corrected query onto Reference PCA space

Projecting corrected query onto Reference UMAP space

  |                                                                            
  |======================================================================| 100%


  |                                                                            
  |                                                                      |   0%[1] "Using assay RNA for query"
[1] "1255 out of 1342 ( 94% ) non-pure cells removed. Use filter.cells=FALSE to avoid pre-filtering"
Performing log-normalization
0%   10   20   30   40   50   60   70   80   90   100%
[----|----|----|----|----|----|----|----|----|----|
**************************************************|
[1] "Aligning query to reference map for batch-correction..."

Projecting corrected query onto Reference PCA space

Projecting corrected query onto Reference UMAP space

  |                                                                            
  |======================================================================| 100%</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
  |                                                                            
  |                                                                      |   0%[1] "Using assay RNA for query"
[1] "339 out of 426 ( 80% ) non-pure cells removed. Use filter.cells=FALSE to avoid pre-filtering"
Performing log-normalization
0%   10   20   30   40   50   60   70   80   90   100%
[----|----|----|----|----|----|----|----|----|----|
**************************************************|
[1] "Aligning query to reference map for batch-correction..."

Projecting corrected query onto Reference PCA space

Projecting corrected query onto Reference UMAP space

  |                                                                            
  |======================================================================| 100%


  |                                                                            
  |                                                                      |   0%[1] "Using assay RNA for query"
[1] "1501 out of 1718 ( 87% ) non-pure cells removed. Use filter.cells=FALSE to avoid pre-filtering"
Performing log-normalization
0%   10   20   30   40   50   60   70   80   90   100%
[----|----|----|----|----|----|----|----|----|----|
**************************************************|
[1] "Aligning query to reference map for batch-correction..."

Projecting corrected query onto Reference PCA space

Projecting corrected query onto Reference UMAP space

  |                                                                            
  |======================================================================| 100%


  |                                                                            
  |                                                                      |   0%[1] "Using assay RNA for query"
[1] "743 out of 818 ( 91% ) non-pure cells removed. Use filter.cells=FALSE to avoid pre-filtering"
Performing log-normalization
0%   10   20   30   40   50   60   70   80   90   100%
[----|----|----|----|----|----|----|----|----|----|
**************************************************|
[1] "Aligning query to reference map for batch-correction..."

Projecting corrected query onto Reference PCA space

Projecting corrected query onto Reference UMAP space

  |                                                                            
  |======================================================================| 100%</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
  |                                                                            
  |                                                                      |   0%[1] "Using assay RNA for query"
[1] "1221 out of 1375 ( 89% ) non-pure cells removed. Use filter.cells=FALSE to avoid pre-filtering"
Performing log-normalization
0%   10   20   30   40   50   60   70   80   90   100%
[----|----|----|----|----|----|----|----|----|----|
**************************************************|
[1] "Aligning query to reference map for batch-correction..."

Projecting corrected query onto Reference PCA space

Projecting corrected query onto Reference UMAP space

  |                                                                            
  |======================================================================| 100%


  |                                                                            
  |                                                                      |   0%[1] "Using assay RNA for query"
[1] "974 out of 1191 ( 82% ) non-pure cells removed. Use filter.cells=FALSE to avoid pre-filtering"
Performing log-normalization
0%   10   20   30   40   50   60   70   80   90   100%
[----|----|----|----|----|----|----|----|----|----|
**************************************************|
[1] "Aligning query to reference map for batch-correction..."

Projecting corrected query onto Reference PCA space

Projecting corrected query onto Reference UMAP space

  |                                                                            
  |======================================================================| 100%


  |                                                                            
  |                                                                      |   0%[1] "Using assay RNA for query"
[1] "1284 out of 1481 ( 87% ) non-pure cells removed. Use filter.cells=FALSE to avoid pre-filtering"
Performing log-normalization
0%   10   20   30   40   50   60   70   80   90   100%
[----|----|----|----|----|----|----|----|----|----|
**************************************************|
[1] "Aligning query to reference map for batch-correction..."

Projecting corrected query onto Reference PCA space

Projecting corrected query onto Reference UMAP space

  |                                                                            
  |======================================================================| 100%</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
  |                                                                            
  |                                                                      |   0%[1] "Using assay RNA for query"
[1] "1516 out of 1655 ( 92% ) non-pure cells removed. Use filter.cells=FALSE to avoid pre-filtering"
Performing log-normalization
0%   10   20   30   40   50   60   70   80   90   100%
[----|----|----|----|----|----|----|----|----|----|
**************************************************|
[1] "Aligning query to reference map for batch-correction..."

Projecting corrected query onto Reference PCA space

Projecting corrected query onto Reference UMAP space

  |                                                                            
  |======================================================================| 100%


  |                                                                            
  |                                                                      |   0%[1] "Using assay RNA for query"
[1] "881 out of 937 ( 94% ) non-pure cells removed. Use filter.cells=FALSE to avoid pre-filtering"
Performing log-normalization
0%   10   20   30   40   50   60   70   80   90   100%
[----|----|----|----|----|----|----|----|----|----|
**************************************************|
[1] "Aligning query to reference map for batch-correction..."

Projecting corrected query onto Reference PCA space

Projecting corrected query onto Reference UMAP space

  |                                                                            
  |======================================================================| 100%


  |                                                                            
  |                                                                      |   0%[1] "Using assay RNA for query"
[1] "2364 out of 2689 ( 88% ) non-pure cells removed. Use filter.cells=FALSE to avoid pre-filtering"
Performing log-normalization
0%   10   20   30   40   50   60   70   80   90   100%
[----|----|----|----|----|----|----|----|----|----|
**************************************************|
[1] "Aligning query to reference map for batch-correction..."

Projecting corrected query onto Reference PCA space

Projecting corrected query onto Reference UMAP space

  |                                                                            
  |======================================================================| 100%</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
  |                                                                            
  |                                                                      |   0%[1] "Using assay RNA for query"
[1] "2658 out of 3429 ( 78% ) non-pure cells removed. Use filter.cells=FALSE to avoid pre-filtering"
Performing log-normalization
0%   10   20   30   40   50   60   70   80   90   100%
[----|----|----|----|----|----|----|----|----|----|
**************************************************|
[1] "Aligning query to reference map for batch-correction..."

Projecting corrected query onto Reference PCA space

Projecting corrected query onto Reference UMAP space

  |                                                                            
  |======================================================================| 100%


  |                                                                            
  |                                                                      |   0%[1] "Using assay RNA for query"
[1] "2212 out of 2526 ( 88% ) non-pure cells removed. Use filter.cells=FALSE to avoid pre-filtering"
Performing log-normalization
0%   10   20   30   40   50   60   70   80   90   100%
[----|----|----|----|----|----|----|----|----|----|
**************************************************|
[1] "Aligning query to reference map for batch-correction..."

Projecting corrected query onto Reference PCA space

Projecting corrected query onto Reference UMAP space

  |                                                                            
  |======================================================================| 100%


  |                                                                            
  |                                                                      |   0%[1] "Using assay RNA for query"
[1] "1008 out of 1121 ( 90% ) non-pure cells removed. Use filter.cells=FALSE to avoid pre-filtering"
Performing log-normalization
0%   10   20   30   40   50   60   70   80   90   100%
[----|----|----|----|----|----|----|----|----|----|
**************************************************|
[1] "Aligning query to reference map for batch-correction..."

Projecting corrected query onto Reference PCA space

Projecting corrected query onto Reference UMAP space

  |                                                                            
  |======================================================================| 100%</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
  |                                                                            
  |                                                                      |   0%[1] "Using assay RNA for query"
[1] "3610 out of 3726 ( 97% ) non-pure cells removed. Use filter.cells=FALSE to avoid pre-filtering"
Performing log-normalization
0%   10   20   30   40   50   60   70   80   90   100%
[----|----|----|----|----|----|----|----|----|----|
**************************************************|
[1] "Aligning query to reference map for batch-correction..."

Projecting corrected query onto Reference PCA space

Projecting corrected query onto Reference UMAP space

  |                                                                            
  |======================================================================| 100%


  |                                                                            
  |                                                                      |   0%[1] "Using assay RNA for query"
[1] "4680 out of 5434 ( 86% ) non-pure cells removed. Use filter.cells=FALSE to avoid pre-filtering"
Performing log-normalization
0%   10   20   30   40   50   60   70   80   90   100%
[----|----|----|----|----|----|----|----|----|----|
**************************************************|
[1] "Aligning query to reference map for batch-correction..."

Projecting corrected query onto Reference PCA space

Projecting corrected query onto Reference UMAP space

  |                                                                            
  |======================================================================| 100%


  |                                                                            
  |                                                                      |   0%[1] "Using assay RNA for query"
[1] "2661 out of 3092 ( 86% ) non-pure cells removed. Use filter.cells=FALSE to avoid pre-filtering"
Performing log-normalization
0%   10   20   30   40   50   60   70   80   90   100%
[----|----|----|----|----|----|----|----|----|----|
**************************************************|
[1] "Aligning query to reference map for batch-correction..."

Projecting corrected query onto Reference PCA space

Projecting corrected query onto Reference UMAP space

  |                                                                            
  |======================================================================| 100%</code></pre>
</div>
</div>
</section>
<section id="classify-cd4-t-subtypes" class="level3" data-number="8.3.5">
<h3 data-number="8.3.5" class="anchored" data-anchor-id="classify-cd4-t-subtypes"><span class="header-section-number">8.3.5</span> Classify CD4 T subtypes</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>merged.<span class="fl">18279.</span>skin.singlets <span class="ot">&lt;-</span> <span class="fu">ProjecTILs.classifier</span>(merged.<span class="fl">18279.</span>skin.singlets, ref.cd4, <span class="at">ncores =</span> <span class="dv">8</span>, <span class="at">split.by =</span> <span class="st">"Sample"</span>, <span class="at">overwrite =</span> F)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>
  |                                                                            
  |                                                                      |   0%[1] "Using assay RNA for query"
[1] "361 out of 437 ( 83% ) non-pure cells removed. Use filter.cells=FALSE to avoid pre-filtering"
Performing log-normalization
0%   10   20   30   40   50   60   70   80   90   100%
[----|----|----|----|----|----|----|----|----|----|
**************************************************|
[1] "Aligning query to reference map for batch-correction..."

Projecting corrected query onto Reference PCA space

Projecting corrected query onto Reference UMAP space

  |                                                                            
  |======================================================================| 100%


  |                                                                            
  |                                                                      |   0%[1] "Using assay RNA for query"
[1] "1139 out of 1342 ( 85% ) non-pure cells removed. Use filter.cells=FALSE to avoid pre-filtering"
Performing log-normalization
0%   10   20   30   40   50   60   70   80   90   100%
[----|----|----|----|----|----|----|----|----|----|
**************************************************|
[1] "Aligning query to reference map for batch-correction..."

Projecting corrected query onto Reference PCA space

Projecting corrected query onto Reference UMAP space

  |                                                                            
  |======================================================================| 100%</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
  |                                                                            
  |                                                                      |   0%[1] "Using assay RNA for query"
[1] "325 out of 426 ( 76% ) non-pure cells removed. Use filter.cells=FALSE to avoid pre-filtering"
Performing log-normalization
0%   10   20   30   40   50   60   70   80   90   100%
[----|----|----|----|----|----|----|----|----|----|
**************************************************|
[1] "Aligning query to reference map for batch-correction..."

Projecting corrected query onto Reference PCA space

Projecting corrected query onto Reference UMAP space

  |                                                                            
  |======================================================================| 100%


  |                                                                            
  |                                                                      |   0%[1] "Using assay RNA for query"
[1] "1519 out of 1718 ( 88% ) non-pure cells removed. Use filter.cells=FALSE to avoid pre-filtering"
Performing log-normalization
0%   10   20   30   40   50   60   70   80   90   100%
[----|----|----|----|----|----|----|----|----|----|
**************************************************|
[1] "Aligning query to reference map for batch-correction..."

Projecting corrected query onto Reference PCA space

Projecting corrected query onto Reference UMAP space

  |                                                                            
  |======================================================================| 100%


  |                                                                            
  |                                                                      |   0%[1] "Using assay RNA for query"
[1] "483 out of 818 ( 59% ) non-pure cells removed. Use filter.cells=FALSE to avoid pre-filtering"
Performing log-normalization
0%   10   20   30   40   50   60   70   80   90   100%
[----|----|----|----|----|----|----|----|----|----|
**************************************************|
[1] "Aligning query to reference map for batch-correction..."

Projecting corrected query onto Reference PCA space

Projecting corrected query onto Reference UMAP space

  |                                                                            
  |======================================================================| 100%</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
  |                                                                            
  |                                                                      |   0%[1] "Using assay RNA for query"
[1] "1102 out of 1375 ( 80% ) non-pure cells removed. Use filter.cells=FALSE to avoid pre-filtering"
Performing log-normalization
0%   10   20   30   40   50   60   70   80   90   100%
[----|----|----|----|----|----|----|----|----|----|
**************************************************|
[1] "Aligning query to reference map for batch-correction..."

Projecting corrected query onto Reference PCA space

Projecting corrected query onto Reference UMAP space

  |                                                                            
  |======================================================================| 100%


  |                                                                            
  |                                                                      |   0%[1] "Using assay RNA for query"
[1] "849 out of 1191 ( 71% ) non-pure cells removed. Use filter.cells=FALSE to avoid pre-filtering"
Performing log-normalization
0%   10   20   30   40   50   60   70   80   90   100%
[----|----|----|----|----|----|----|----|----|----|
**************************************************|
[1] "Aligning query to reference map for batch-correction..."

Projecting corrected query onto Reference PCA space

Projecting corrected query onto Reference UMAP space

  |                                                                            
  |======================================================================| 100%


  |                                                                            
  |                                                                      |   0%[1] "Using assay RNA for query"
[1] "1101 out of 1481 ( 74% ) non-pure cells removed. Use filter.cells=FALSE to avoid pre-filtering"
Performing log-normalization
0%   10   20   30   40   50   60   70   80   90   100%
[----|----|----|----|----|----|----|----|----|----|
**************************************************|
[1] "Aligning query to reference map for batch-correction..."

Projecting corrected query onto Reference PCA space

Projecting corrected query onto Reference UMAP space

  |                                                                            
  |======================================================================| 100%</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
  |                                                                            
  |                                                                      |   0%[1] "Using assay RNA for query"
[1] "1363 out of 1655 ( 82% ) non-pure cells removed. Use filter.cells=FALSE to avoid pre-filtering"
Performing log-normalization
0%   10   20   30   40   50   60   70   80   90   100%
[----|----|----|----|----|----|----|----|----|----|
**************************************************|
[1] "Aligning query to reference map for batch-correction..."

Projecting corrected query onto Reference PCA space

Projecting corrected query onto Reference UMAP space

  |                                                                            
  |======================================================================| 100%


  |                                                                            
  |                                                                      |   0%[1] "Using assay RNA for query"
[1] "731 out of 937 ( 78% ) non-pure cells removed. Use filter.cells=FALSE to avoid pre-filtering"
Performing log-normalization
0%   10   20   30   40   50   60   70   80   90   100%
[----|----|----|----|----|----|----|----|----|----|
**************************************************|
[1] "Aligning query to reference map for batch-correction..."

Projecting corrected query onto Reference PCA space

Projecting corrected query onto Reference UMAP space

  |                                                                            
  |======================================================================| 100%


  |                                                                            
  |                                                                      |   0%[1] "Using assay RNA for query"
[1] "1854 out of 2689 ( 69% ) non-pure cells removed. Use filter.cells=FALSE to avoid pre-filtering"
Performing log-normalization
0%   10   20   30   40   50   60   70   80   90   100%
[----|----|----|----|----|----|----|----|----|----|
**************************************************|
[1] "Aligning query to reference map for batch-correction..."

Projecting corrected query onto Reference PCA space

Projecting corrected query onto Reference UMAP space

  |                                                                            
  |======================================================================| 100%</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
  |                                                                            
  |                                                                      |   0%[1] "Using assay RNA for query"
[1] "2485 out of 3429 ( 72% ) non-pure cells removed. Use filter.cells=FALSE to avoid pre-filtering"
Performing log-normalization
0%   10   20   30   40   50   60   70   80   90   100%
[----|----|----|----|----|----|----|----|----|----|
**************************************************|
[1] "Aligning query to reference map for batch-correction..."

Projecting corrected query onto Reference PCA space

Projecting corrected query onto Reference UMAP space

  |                                                                            
  |======================================================================| 100%


  |                                                                            
  |                                                                      |   0%[1] "Using assay RNA for query"
[1] "2158 out of 2526 ( 85% ) non-pure cells removed. Use filter.cells=FALSE to avoid pre-filtering"
Performing log-normalization
0%   10   20   30   40   50   60   70   80   90   100%
[----|----|----|----|----|----|----|----|----|----|
**************************************************|
[1] "Aligning query to reference map for batch-correction..."

Projecting corrected query onto Reference PCA space

Projecting corrected query onto Reference UMAP space

  |                                                                            
  |======================================================================| 100%


  |                                                                            
  |                                                                      |   0%[1] "Using assay RNA for query"
[1] "819 out of 1121 ( 73% ) non-pure cells removed. Use filter.cells=FALSE to avoid pre-filtering"
Performing log-normalization
0%   10   20   30   40   50   60   70   80   90   100%
[----|----|----|----|----|----|----|----|----|----|
**************************************************|
[1] "Aligning query to reference map for batch-correction..."

Projecting corrected query onto Reference PCA space

Projecting corrected query onto Reference UMAP space

  |                                                                            
  |======================================================================| 100%</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
  |                                                                            
  |                                                                      |   0%[1] "Using assay RNA for query"
[1] "3262 out of 3726 ( 88% ) non-pure cells removed. Use filter.cells=FALSE to avoid pre-filtering"
Performing log-normalization
0%   10   20   30   40   50   60   70   80   90   100%
[----|----|----|----|----|----|----|----|----|----|
**************************************************|
[1] "Aligning query to reference map for batch-correction..."

Projecting corrected query onto Reference PCA space

Projecting corrected query onto Reference UMAP space

  |                                                                            
  |======================================================================| 100%


  |                                                                            
  |                                                                      |   0%[1] "Using assay RNA for query"
[1] "4574 out of 5434 ( 84% ) non-pure cells removed. Use filter.cells=FALSE to avoid pre-filtering"
Performing log-normalization
0%   10   20   30   40   50   60   70   80   90   100%
[----|----|----|----|----|----|----|----|----|----|
**************************************************|
[1] "Aligning query to reference map for batch-correction..."

Projecting corrected query onto Reference PCA space

Projecting corrected query onto Reference UMAP space

  |                                                                            
  |======================================================================| 100%


  |                                                                            
  |                                                                      |   0%[1] "Using assay RNA for query"
[1] "2233 out of 3092 ( 72% ) non-pure cells removed. Use filter.cells=FALSE to avoid pre-filtering"
Performing log-normalization
0%   10   20   30   40   50   60   70   80   90   100%
[----|----|----|----|----|----|----|----|----|----|
**************************************************|
[1] "Aligning query to reference map for batch-correction..."

Projecting corrected query onto Reference PCA space

Projecting corrected query onto Reference UMAP space

  |                                                                            
  |======================================================================| 100%</code></pre>
</div>
</div>
</section>
</section>
<section id="tabulate-and-visualize-projections" class="level2" data-number="8.4">
<h2 data-number="8.4" class="anchored" data-anchor-id="tabulate-and-visualize-projections"><span class="header-section-number">8.4</span> Tabulate and visualize projections</h2>
<p>In addition to guiding annotation, these projected cell states can be quantified downstream comparing pre vs post-vaccine, tumor samples over time, or to characterize the phenotypes of T-cells that have expanded with vaccination</p>
<section id="count-total-cells-by-t-cell-state" class="level3" data-number="8.4.1">
<h3 data-number="8.4.1" class="anchored" data-anchor-id="count-total-cells-by-t-cell-state"><span class="header-section-number">8.4.1</span> Count total cells by T cell state</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(merged.<span class="fl">18279.</span>skin.singlets<span class="sc">$</span>functional.cluster, <span class="at">useNA =</span> <span class="st">"ifany"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
CD4.CTL_EOMES   CD4.CTL_Exh  CD4.CTL_GNLY    CD4.Memory CD4.NaiveLike 
         1110           166           534          1742          1486 
      CD4.Tfh      CD4.Th17      CD4.Treg        CD8.CM        CD8.EM 
          575            84          1329          2312          1092 
     CD8.MAIT CD8.NaiveLike     CD8.TEMRA       CD8.TEX      CD8.TPEX 
           47           239            77           266            61 
         &lt;NA&gt; 
        22277 </code></pre>
</div>
</div>
</section>
<section id="tabulate-which-t-cell-state-was-most-abundant-in-each-subcluster" class="level3" data-number="8.4.2">
<h3 data-number="8.4.2" class="anchored" data-anchor-id="tabulate-which-t-cell-state-was-most-abundant-in-each-subcluster"><span class="header-section-number">8.4.2</span> Tabulate which T cell state was most abundant in each subcluster</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="fu">as.data.frame</span>(<span class="fu">table</span>(merged.<span class="fl">18279.</span>skin.singlets<span class="sc">$</span>sub.cluster, merged.<span class="fl">18279.</span>skin.singlets<span class="sc">$</span>functional.cluster, <span class="at">useNA =</span> <span class="st">"ifany"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"Cluster"</span> <span class="ot">=</span> Var1, <span class="st">"CellState"</span> <span class="ot">=</span> Var2) <span class="sc">%&gt;%</span></span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(Cluster) <span class="sc">%&gt;%</span></span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">slice_max</span>(Freq,<span class="at">n=</span><span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb25-6"><a href="#cb25-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">inner_join</span>(<span class="fu">enframe</span>(<span class="fu">table</span>(merged.<span class="fl">18279.</span>skin.singlets<span class="sc">$</span>sub.cluster),<span class="at">name=</span><span class="st">"Cluster"</span>,<span class="at">value=</span><span class="st">"TotalCount"</span>),<span class="at">by=</span><span class="st">"Cluster"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb25-7"><a href="#cb25-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as.data.frame</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>   Cluster     CellState Freq TotalCount
1        0          &lt;NA&gt; 5768       5768
2      1_0 CD4.NaiveLike 1093       2588
3      1_1 CD4.CTL_EOMES  454       1100
4       10          &lt;NA&gt; 1123       1123
5     11_0          &lt;NA&gt;  660        929
6     11_1          &lt;NA&gt;  106        157
7       12          &lt;NA&gt;  907        907
8     13_0       CD8.TEX   67        314
9     13_1          &lt;NA&gt;  247        255
10      14          &lt;NA&gt;  557        557
11      15          &lt;NA&gt;  472        472
12      16          &lt;NA&gt;  407        407
13      17          &lt;NA&gt;  360        360
14      18          &lt;NA&gt;  308        308
15    19_0          &lt;NA&gt;  274        280
16     2_0        CD8.EM  766       1464
17     2_1        CD8.CM  696        980
18     2_2          &lt;NA&gt;  142        407
19      20          &lt;NA&gt;  262        262
20      21          &lt;NA&gt;  238        238
21      22          &lt;NA&gt;  228        228
22      23          &lt;NA&gt;  175        177
23      24          &lt;NA&gt;  175        175
24      25          &lt;NA&gt;   46         46
25      26          &lt;NA&gt;   33         33
26       3          &lt;NA&gt; 2418       2418
27     4_0        CD8.CM  664       1826
28     4_1          &lt;NA&gt;  343        537
29     5_0          &lt;NA&gt; 1421       1465
30     5_1          &lt;NA&gt;  562        564
31     5_2          &lt;NA&gt;  169        171
32     6_0    CD4.Memory  446       1263
33     6_1    CD4.Memory  301        923
34       7          &lt;NA&gt; 1746       1746
35       8          &lt;NA&gt; 1483       1483
36     9_0      CD4.Treg 1092       1310
37     9_1      CD4.Treg   82        156</code></pre>
</div>
</div>
</section>
<section id="plot-umap-of-all-cells-labeled-by-projected-t-cell-state" class="level3" data-number="8.4.3">
<h3 data-number="8.4.3" class="anchored" data-anchor-id="plot-umap-of-all-cells-labeled-by-projected-t-cell-state"><span class="header-section-number">8.4.3</span> Plot UMAP of all cells labeled by projected T cell state</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(merged.<span class="fl">18279.</span>skin.singlets, <span class="at">group.by =</span> <span class="st">"functional.cluster"</span>, <span class="at">reduction=</span><span class="st">"umap.harmony"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Skin_scRNA_Part8_files/figure-html/unnamed-chunk-10-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="plot-just-tnk-cell-lineage" class="level3" data-number="8.4.4">
<h3 data-number="8.4.4" class="anchored" data-anchor-id="plot-just-tnk-cell-lineage"><span class="header-section-number">8.4.4</span> Plot just T/NK cell lineage</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a>tnk_clusters <span class="ot">&lt;-</span> <span class="fu">sort</span>(<span class="fu">str_subset</span>(<span class="fu">unique</span>(merged.<span class="fl">18279.</span>skin.singlets<span class="sc">$</span>sub.cluster),<span class="st">"_"</span>))</span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a>tnk_clusters</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "1_0"  "1_1"  "11_0" "11_1" "13_0" "13_1" "19_0" "2_0"  "2_1"  "2_2" 
[11] "4_0"  "4_1"  "5_0"  "5_1"  "5_2"  "6_0"  "6_1"  "9_0"  "9_1" </code></pre>
</div>
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a>tnk_cells <span class="ot">&lt;-</span> <span class="fu">WhichCells</span>(merged.<span class="fl">18279.</span>skin.singlets,<span class="at">idents =</span> tnk_clusters)</span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(merged.<span class="fl">18279.</span>skin.singlets, </span>
<span id="cb30-3"><a href="#cb30-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">group.by =</span> <span class="st">"functional.cluster"</span>, </span>
<span id="cb30-4"><a href="#cb30-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">reduction=</span><span class="st">"umap.harmony"</span>,</span>
<span id="cb30-5"><a href="#cb30-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">cells =</span> tnk_cells)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Skin_scRNA_Part8_files/figure-html/unnamed-chunk-11-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="save-modified-object" class="level2" data-number="8.5">
<h2 data-number="8.5" class="anchored" data-anchor-id="save-modified-object"><span class="header-section-number">8.5</span> Save modified object</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(merged.<span class="fl">18279.</span>skin.singlets, <span class="st">"Skin_scRNA_Part8.rds"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="get-session-info" class="level2" data-number="8.6">
<h2 data-number="8.6" class="anchored" data-anchor-id="get-session-info"><span class="header-section-number">8.6</span> Get session info</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>R version 4.3.1 (2023-06-16)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Rocky Linux 8.10 (Green Obsidian)

Matrix products: default
BLAS/LAPACK: /usr/lib64/libopenblasp-r0.3.15.so;  LAPACK version 3.9.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

time zone: America/New_York
tzcode source: system (glibc)

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] ProjecTILs_3.3.1            STACAS_2.2.2               
 [3] cellXY_0.99.0               scDblFinder_1.14.0         
 [5] harmony_1.2.0               alevinQC_1.16.1            
 [7] vctrs_0.6.5                 patchwork_1.3.0            
 [9] scater_1.28.0               scuttle_1.10.3             
[11] speckle_1.0.0               Matrix_1.6-4               
[13] fishpond_2.6.2              readxl_1.4.3               
[15] SingleCellExperiment_1.22.0 SummarizedExperiment_1.30.2
[17] Biobase_2.60.0              GenomicRanges_1.52.1       
[19] GenomeInfoDb_1.36.4         IRanges_2.34.1             
[21] S4Vectors_0.38.2            BiocGenerics_0.46.0        
[23] MatrixGenerics_1.12.3       matrixStats_1.2.0          
[25] flexmix_2.3-19              lattice_0.22-5             
[27] SeuratWrappers_0.3.19       miQC_1.8.0                 
[29] lubridate_1.9.3             forcats_1.0.0              
[31] stringr_1.5.1               dplyr_1.1.4                
[33] purrr_1.0.2                 readr_2.1.5                
[35] tidyr_1.3.1                 tibble_3.2.1               
[37] ggplot2_3.4.4               tidyverse_2.0.0            
[39] Seurat_5.1.0                SeuratObject_5.0.2         
[41] sp_2.1-3                    sctransform_0.4.1          
[43] glmGamPoi_1.12.2            presto_1.0.0               
[45] Rcpp_1.0.12                 devtools_2.4.5             
[47] usethis_2.2.2               data.table_1.15.0          

loaded via a namespace (and not attached):
  [1] fs_1.6.3                  spatstat.sparse_3.0-3    
  [3] bitops_1.0-7              httr_1.4.7               
  [5] RColorBrewer_1.1-3        profvis_0.3.8            
  [7] tools_4.3.1               utf8_1.2.4               
  [9] R6_2.5.1                  DT_0.31                  
 [11] lazyeval_0.2.2            uwot_0.1.16              
 [13] urlchecker_1.0.1          withr_3.0.0              
 [15] GGally_2.2.1              gridExtra_2.3            
 [17] progressr_0.14.0          cli_3.6.2                
 [19] spatstat.explore_3.2-6    fastDummies_1.7.3        
 [21] labeling_0.4.3            spatstat.data_3.0-4      
 [23] askpass_1.2.0             ggridges_0.5.6           
 [25] pbapply_1.7-2             Rsamtools_2.16.0         
 [27] R.utils_2.12.3            parallelly_1.37.0        
 [29] sessioninfo_1.2.2         limma_3.56.2             
 [31] RSQLite_2.3.5             BiocIO_1.10.0            
 [33] generics_0.1.3            gtools_3.9.5             
 [35] ica_1.0-3                 spatstat.random_3.2-2    
 [37] ggbeeswarm_0.7.2          fansi_1.0.6              
 [39] abind_1.4-5               R.methodsS3_1.8.2        
 [41] lifecycle_1.0.4           yaml_2.3.8               
 [43] edgeR_3.42.4              Rtsne_0.17               
 [45] blob_1.2.4                grid_4.3.1               
 [47] dqrng_0.3.2               promises_1.2.1           
 [49] crayon_1.5.2              shinydashboard_0.7.2     
 [51] miniUI_0.1.1.1            beachmat_2.16.0          
 [53] cowplot_1.1.3             KEGGREST_1.40.1          
 [55] scGate_1.6.0              metapod_1.8.0            
 [57] pillar_1.9.0              knitr_1.45               
 [59] rjson_0.2.21              xgboost_1.7.7.1          
 [61] future.apply_1.11.1       codetools_0.2-19         
 [63] leiden_0.4.3.1            glue_1.7.0               
 [65] remotes_2.4.2.1           png_0.1-8                
 [67] spam_2.10-0               org.Mm.eg.db_3.18.0      
 [69] cellranger_1.1.0          gtable_0.3.4             
 [71] cachem_1.0.8              xfun_0.42                
 [73] S4Arrays_1.2.0            mime_0.12                
 [75] pracma_2.4.4              survival_3.5-8           
 [77] pheatmap_1.0.12           statmod_1.5.0            
 [79] bluster_1.10.0            ellipsis_0.3.2           
 [81] fitdistrplus_1.1-11       ROCR_1.0-11              
 [83] nlme_3.1-164              bit64_4.0.5              
 [85] RcppAnnoy_0.0.22          irlba_2.3.5.1            
 [87] vipor_0.4.7               KernSmooth_2.23-22       
 [89] DBI_1.2.2                 colorspace_2.1-0         
 [91] nnet_7.3-19               UCell_2.6.2              
 [93] tidyselect_1.2.0          bit_4.0.5                
 [95] compiler_4.3.1            BiocNeighbors_1.18.0     
 [97] DelayedArray_0.26.7       plotly_4.10.4            
 [99] rtracklayer_1.60.1        scales_1.3.0             
[101] lmtest_0.9-40             digest_0.6.34            
[103] goftest_1.2-3             spatstat.utils_3.0-4     
[105] rmarkdown_2.25            XVector_0.40.0           
[107] htmltools_0.5.7           pkgconfig_2.0.3          
[109] umap_0.2.10.0             sparseMatrixStats_1.12.2 
[111] fastmap_1.1.1             rlang_1.1.3              
[113] htmlwidgets_1.6.4         shiny_1.8.0              
[115] DelayedMatrixStats_1.22.6 farver_2.1.1             
[117] zoo_1.8-12                jsonlite_1.8.8           
[119] BiocParallel_1.34.2       R.oo_1.26.0              
[121] BiocSingular_1.16.0       RCurl_1.98-1.14          
[123] magrittr_2.0.3            modeltools_0.2-23        
[125] GenomeInfoDbData_1.2.10   dotCall64_1.1-1          
[127] munsell_0.5.0             viridis_0.6.5            
[129] reticulate_1.35.0         stringi_1.8.3            
[131] zlibbioc_1.46.0           MASS_7.3-60.0.1          
[133] org.Hs.eg.db_3.18.0       plyr_1.8.9               
[135] pkgbuild_1.4.3            ggstats_0.5.1            
[137] parallel_4.3.1            listenv_0.9.1            
[139] ggrepel_0.9.5             deldir_2.0-2             
[141] Biostrings_2.68.1         splines_4.3.1            
[143] tensor_1.5                hms_1.1.3                
[145] locfit_1.5-9.8            igraph_2.0.2             
[147] spatstat.geom_3.2-8       RcppHNSW_0.6.0           
[149] reshape2_1.4.4            ScaledMatrix_1.8.1       
[151] pkgload_1.3.4             XML_3.99-0.16.1          
[153] evaluate_0.23             scran_1.28.2             
[155] BiocManager_1.30.22       tzdb_0.4.0               
[157] httpuv_1.6.14             openssl_2.1.1            
[159] RANN_2.6.1                polyclip_1.10-6          
[161] future_1.33.1             scattermore_1.2          
[163] rsvd_1.0.5                xtable_1.8-4             
[165] restfulr_0.0.15           svMisc_1.2.3             
[167] RSpectra_0.16-1           later_1.3.2              
[169] viridisLite_0.4.2         AnnotationDbi_1.64.1     
[171] GenomicAlignments_1.36.0  memoise_2.0.1            
[173] beeswarm_0.4.0            tximport_1.28.0          
[175] cluster_2.1.6             timechange_0.3.0         
[177] globals_0.16.2           </code></pre>
</div>
</div>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "î§‹";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="./Skin_scRNA_Part7.html" class="pagination-link" aria-label="Skin: Assisting cell type annotation with enrichR">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">Skin: Assisting cell type annotation with enrichR</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="./Skin_scRNA_Part9.html" class="pagination-link" aria-label="Skin: Differential expression analysis, Post vs Pre 3rd vaccination">
        <span class="nav-page-text"><span class="chapter-number">9</span>&nbsp; <span class="chapter-title">Skin: Differential expression analysis, Post vs Pre 3rd vaccination</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->




</body></html>