<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.6.39">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>9&nbsp; Skin: Differential expression analysis, Post vs Pre 3rd vaccination â€“ Blass &amp; Keskin (2025): Melanoma NeoVax analysis: Skin single-cell RNAseq</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./Skin_scRNA_Part10.html" rel="next">
<link href="./Skin_scRNA_Part8.html" rel="prev">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-e26003cea8cd680ca0c55a263523d882.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap-41a4eb7d26fa9333a8f88b12b94df59e.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


</head>

<body class="nav-sidebar floating">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./Skin_scRNA_Part9.html"><span class="chapter-number">9</span>&nbsp; <span class="chapter-title">Skin: Differential expression analysis, Post vs Pre 3rd vaccination</span></a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="./">Blass &amp; Keskin (2025): Melanoma NeoVax analysis: Skin single-cell RNAseq</a> 
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Introduction</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Skin_scRNA_Part1.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Skin: QC &amp; Filtering</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Skin_scRNA_Part2.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Skin: Normalization, doublet discrimination, integration, clustering</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Skin_scRNA_Part3.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Skin: Check patient sexes</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Skin_scRNA_Part4.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Skin: Post-clustering QC and characterization</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Skin_scRNA_Part5.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">Skin: Post-clustering comprehensive marker identification</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Skin_scRNA_Part6.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">Skin: Sub-clustering on T/NK lineage</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Skin_scRNA_Part7.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">Skin: Assisting cell type annotation with enrichR</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Skin_scRNA_Part8.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">8</span>&nbsp; <span class="chapter-title">Skin: Classification of CD4 and CD8 T cell states within T/NK lineage</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Skin_scRNA_Part9.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text"><span class="chapter-number">9</span>&nbsp; <span class="chapter-title">Skin: Differential expression analysis, Post vs Pre 3rd vaccination</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Skin_scRNA_Part10.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">10</span>&nbsp; <span class="chapter-title">Skin: Deep phenotyping, cluster annotation, and exploration of Post vs Pre 3rd vaccination differences in cell proportions and expression</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Skin_scRNA_Part11.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">11</span>&nbsp; <span class="chapter-title">Skin: Calculating Post vs Pre 3rd vaccination differences in pathway enrichments with fgsea</span></span></a>
  </div>
</li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#set-up-seurat-workspace" id="toc-set-up-seurat-workspace" class="nav-link active" data-scroll-target="#set-up-seurat-workspace"><span class="header-section-number">9.1</span> Set up Seurat workspace</a></li>
  <li><a href="#load-previous-seurat-object" id="toc-load-previous-seurat-object" class="nav-link" data-scroll-target="#load-previous-seurat-object"><span class="header-section-number">9.2</span> Load previous seurat object</a></li>
  <li><a href="#run-pseudobulk-differential-expression-test-with-muscat" id="toc-run-pseudobulk-differential-expression-test-with-muscat" class="nav-link" data-scroll-target="#run-pseudobulk-differential-expression-test-with-muscat"><span class="header-section-number">9.3</span> Run pseudobulk differential expression test with <code>muscat</code></a>
  <ul class="collapse">
  <li><a href="#convert-seurat-object-to-singlecellexperiment" id="toc-convert-seurat-object-to-singlecellexperiment" class="nav-link" data-scroll-target="#convert-seurat-object-to-singlecellexperiment"><span class="header-section-number">9.3.1</span> Convert Seurat object to SingleCellExperiment</a></li>
  <li><a href="#prep-coldata-and-create-pseudobulk-of-raw-counts-by-sub-cluster-x-sample" id="toc-prep-coldata-and-create-pseudobulk-of-raw-counts-by-sub-cluster-x-sample" class="nav-link" data-scroll-target="#prep-coldata-and-create-pseudobulk-of-raw-counts-by-sub-cluster-x-sample"><span class="header-section-number">9.3.2</span> Prep coldata and create pseudobulk of raw counts by sub-cluster x sample</a></li>
  <li><a href="#construct-design-contrast-matrix-to-compare-post3rd-to-pre3rd-timepoints" id="toc-construct-design-contrast-matrix-to-compare-post3rd-to-pre3rd-timepoints" class="nav-link" data-scroll-target="#construct-design-contrast-matrix-to-compare-post3rd-to-pre3rd-timepoints"><span class="header-section-number">9.3.3</span> Construct design &amp; contrast matrix to compare Post3rd to Pre3rd timepoints</a></li>
  <li><a href="#run-differential-expression-test-via-deseq2" id="toc-run-differential-expression-test-via-deseq2" class="nav-link" data-scroll-target="#run-differential-expression-test-via-deseq2"><span class="header-section-number">9.3.4</span> Run differential expression test via DESeq2</a></li>
  <li><a href="#show-snippet-of-results-from-one-cluster-upregulated-genes-in-cluster-7" id="toc-show-snippet-of-results-from-one-cluster-upregulated-genes-in-cluster-7" class="nav-link" data-scroll-target="#show-snippet-of-results-from-one-cluster-upregulated-genes-in-cluster-7"><span class="header-section-number">9.3.5</span> Show snippet of results from one cluster: upregulated genes in cluster 7</a></li>
  <li><a href="#write-differential-expression-results-to-tsv" id="toc-write-differential-expression-results-to-tsv" class="nav-link" data-scroll-target="#write-differential-expression-results-to-tsv"><span class="header-section-number">9.3.6</span> Write differential expression results to tsv</a></li>
  <li><a href="#plot-number-of-significant-genes-for-all-clusters" id="toc-plot-number-of-significant-genes-for-all-clusters" class="nav-link" data-scroll-target="#plot-number-of-significant-genes-for-all-clusters"><span class="header-section-number">9.3.7</span> Plot number of significant genes for all clusters</a></li>
  </ul></li>
  <li><a href="#get-session-info" id="toc-get-session-info" class="nav-link" data-scroll-target="#get-session-info"><span class="header-section-number">9.4</span> Get session info</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title"><span id="sec-skin-differential-expression" class="quarto-section-identifier"><span class="chapter-number">9</span>&nbsp; <span class="chapter-title">Skin: Differential expression analysis, Post vs Pre 3rd vaccination</span></span></h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<section id="set-up-seurat-workspace" class="level2" data-number="9.1">
<h2 data-number="9.1" class="anchored" data-anchor-id="set-up-seurat-workspace"><span class="header-section-number">9.1</span> Set up Seurat workspace</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load libraries</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(presto)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Seurat)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(paletteer)</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(msigdbr)</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ComplexHeatmap)</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(circlize)</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(muscat)</span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(edgeR)</span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(SingleCellExperiment)</span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(DESeq2)</span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(scuttle)</span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(gprofiler2)</span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(msigdbr)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="load-previous-seurat-object" class="level2" data-number="9.2">
<h2 data-number="9.2" class="anchored" data-anchor-id="load-previous-seurat-object"><span class="header-section-number">9.2</span> Load previous seurat object</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>merged.<span class="fl">18279.</span>skin.singlets <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"Skin_scRNA_Part8.rds"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="run-pseudobulk-differential-expression-test-with-muscat" class="level2" data-number="9.3">
<h2 data-number="9.3" class="anchored" data-anchor-id="run-pseudobulk-differential-expression-test-with-muscat"><span class="header-section-number">9.3</span> Run pseudobulk differential expression test with <code>muscat</code></h2>
<section id="convert-seurat-object-to-singlecellexperiment" class="level3" data-number="9.3.1">
<h3 data-number="9.3.1" class="anchored" data-anchor-id="convert-seurat-object-to-singlecellexperiment"><span class="header-section-number">9.3.1</span> Convert Seurat object to SingleCellExperiment</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>merged.<span class="fl">18279.</span>skin.singlets[[<span class="st">'RNA'</span>]] <span class="ot">&lt;-</span> <span class="fu">JoinLayers</span>(merged.<span class="fl">18279.</span>skin.singlets[[<span class="st">'RNA'</span>]])</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>merged.sce <span class="ot">&lt;-</span> <span class="fu">as.SingleCellExperiment</span>(merged.<span class="fl">18279.</span>skin.singlets, <span class="at">assay=</span><span class="st">"RNA"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="prep-coldata-and-create-pseudobulk-of-raw-counts-by-sub-cluster-x-sample" class="level3" data-number="9.3.2">
<h3 data-number="9.3.2" class="anchored" data-anchor-id="prep-coldata-and-create-pseudobulk-of-raw-counts-by-sub-cluster-x-sample"><span class="header-section-number">9.3.2</span> Prep coldata and create pseudobulk of raw counts by sub-cluster x sample</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>(mergedCondition.sce <span class="ot">&lt;-</span> <span class="fu">prepSCE</span>(merged.sce,</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>    <span class="at">kid =</span> <span class="st">"sub.cluster"</span>,</span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">gid =</span> <span class="st">"Timepoint"</span>,</span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">sid =</span> <span class="st">"Sample"</span>,</span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">drop=</span><span class="cn">TRUE</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>class: SingleCellExperiment 
dim: 61217 33397 
metadata(1): experiment_info
assays(2): counts logcounts
rownames(61217): 5-8S-rRNA 5S-rRNA ... ZZEF1 ZZZ3
rowData names(0):
colnames(33397): P101_Skin_Pre3rd_2.5mgIpi_RNA_GCACATAAGTGCCATT
  P101_Skin_Pre3rd_2.5mgIpi_RNA_GGAACTTGTCGCGGTT ...
  P111_Skin_Post3rd_5mgIpi_RNA_ACGATGTTCCTAGGGC
  P111_Skin_Post3rd_5mgIpi_RNA_GAACATCAGCCACGTC
colData names(3): cluster_id sample_id group_id
reducedDimNames(4): PCA UMAP.UNINTEGRATED INTEGRATED.HARMONY
  UMAP.HARMONY
mainExpName: RNA
altExpNames(0):</code></pre>
</div>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>pb <span class="ot">&lt;-</span> <span class="fu">aggregateData</span>(mergedCondition.sce,</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>    <span class="at">assay =</span> <span class="st">"counts"</span>,<span class="at">fun=</span><span class="st">"sum"</span>,</span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"cluster_id"</span>, <span class="st">"sample_id"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="construct-design-contrast-matrix-to-compare-post3rd-to-pre3rd-timepoints" class="level3" data-number="9.3.3">
<h3 data-number="9.3.3" class="anchored" data-anchor-id="construct-design-contrast-matrix-to-compare-post3rd-to-pre3rd-timepoints"><span class="header-section-number">9.3.3</span> Construct design &amp; contrast matrix to compare Post3rd to Pre3rd timepoints</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>ei <span class="ot">&lt;-</span> <span class="fu">metadata</span>(mergedCondition.sce)<span class="sc">$</span>experiment_info</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>mm <span class="ot">&lt;-</span> <span class="fu">model.matrix</span>(<span class="sc">~</span><span class="dv">0</span> <span class="sc">+</span> ei<span class="sc">$</span>group_id)</span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>dim_name <span class="ot">&lt;-</span> <span class="fu">levels</span>(ei<span class="sc">$</span>group_id)</span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a><span class="fu">dimnames</span>(mm) <span class="ot">&lt;-</span> <span class="fu">list</span>(ei<span class="sc">$</span>sample_id, dim_name)</span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>contrast <span class="ot">&lt;-</span> <span class="fu">makeContrasts</span>(<span class="st">"Post3rd-Pre3rd"</span>, <span class="at">levels =</span> mm)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="run-differential-expression-test-via-deseq2" class="level3" data-number="9.3.4">
<h3 data-number="9.3.4" class="anchored" data-anchor-id="run-differential-expression-test-via-deseq2"><span class="header-section-number">9.3.4</span> Run differential expression test via DESeq2</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">pbDS</span>(pb, </span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>            <span class="at">design =</span> mm, </span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>            <span class="at">contrast =</span> contrast, </span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>            <span class="at">method=</span><span class="st">"DESeq2"</span>, </span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>            <span class="at">min_cells =</span> <span class="dv">20</span>,</span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>            <span class="at">verbose =</span> <span class="cn">FALSE</span>)</span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine results from all clusters into one tibble</span></span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(res<span class="sc">$</span>table<span class="sc">$</span><span class="st">`</span><span class="at">Post3rd-Pre3rd</span><span class="st">`</span>, <span class="at">.id =</span> <span class="st">"sub.cluster"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as_tibble</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="show-snippet-of-results-from-one-cluster-upregulated-genes-in-cluster-7" class="level3" data-number="9.3.5">
<h3 data-number="9.3.5" class="anchored" data-anchor-id="show-snippet-of-results-from-one-cluster-upregulated-genes-in-cluster-7"><span class="header-section-number">9.3.5</span> Show snippet of results from one cluster: upregulated genes in cluster 7</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>df <span class="sc">%&gt;%</span> </span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(p_adj.loc <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> sub.cluster<span class="sc">==</span><span class="st">"7"</span> <span class="sc">&amp;</span> logFC <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">arrange</span>(p_adj.loc) <span class="sc">%&gt;%</span> </span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(<span class="at">n=</span><span class="dv">30</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 927 Ã— 11
   sub.cluster gene     cluster_id baseMean logFC lfcSE  stat    p_val p_adj.loc
   &lt;chr&gt;       &lt;chr&gt;    &lt;chr&gt;         &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;    &lt;dbl&gt;     &lt;dbl&gt;
 1 7           APOBEC3A 7             716.   4.29 0.422 10.2  3.21e-24  2.56e-20
 2 7           ISG15    7            1378.   4.40 0.441  9.99 1.74e-23  6.95e-20
 3 7           IFI6     7             648.   4.10 0.413  9.91 3.94e-23  1.05e-19
 4 7           MX1      7             361.   3.94 0.434  9.09 1.03e-19  2.05e-16
 5 7           OAS2     7              78.2  3.31 0.370  8.95 3.43e-19  5.48e-16
 6 7           IRF7     7             310.   2.48 0.280  8.84 9.97e-19  1.33e-15
 7 7           OAS1     7             131.   3.75 0.429  8.74 2.40e-18  2.73e-15
 8 7           IFI35    7             164.   3.08 0.355  8.68 4.03e-18  3.59e-15
 9 7           LY6E     7             822.   2.92 0.337  8.68 4.05e-18  3.59e-15
10 7           OAS3     7              65.9  3.21 0.390  8.24 1.78e-16  1.29e-13
11 7           XAF1     7             125.   2.78 0.353  7.88 3.26e-15  1.45e-12
12 7           SAMD9    7              56.3  3.20 0.411  7.77 7.74e-15  3.09e-12
13 7           IL15RA   7              35.5  2.68 0.345  7.76 8.59e-15  3.26e-12
14 7           PSME2    7             791.   1.62 0.210  7.72 1.16e-14  4.01e-12
15 7           LGALS9   7             352.   1.66 0.215  7.71 1.25e-14  4.15e-12
16 7           TNFSF10  7             295.   3.25 0.422  7.70 1.32e-14  4.21e-12
17 7           EPSTI1   7             289.   2.69 0.350  7.67 1.67e-14  5.12e-12
18 7           PML      7             109.   1.87 0.243  7.67 1.76e-14  5.21e-12
19 7           IFITM1   7            1481.   4.47 0.594  7.52 5.39e-14  1.34e-11
20 7           SERPING1 7             322.   2.71 0.362  7.49 7.13e-14  1.69e-11
21 7           UBE2L6   7             247.   2.13 0.285  7.48 7.22e-14  1.69e-11
22 7           IFIT3    7              82.2  6.32 0.846  7.47 8.24e-14  1.87e-11
23 7           PPA1     7             507.   2.46 0.330  7.46 8.44e-14  1.87e-11
24 7           TYMP     7            1459.   2.09 0.281  7.45 9.52e-14  2.05e-11
25 7           NMI      7              78.9  2.24 0.302  7.43 1.07e-13  2.24e-11
26 7           LYSMD2   7              80.9  1.83 0.246  7.43 1.10e-13  2.26e-11
27 7           ENSG000â€¦ 7              25.5  3.14 0.424  7.40 1.38e-13  2.75e-11
28 7           LGALS3BP 7             147.   2.38 0.323  7.38 1.59e-13  3.09e-11
29 7           SP110    7             139.   1.98 0.269  7.36 1.81e-13  3.43e-11
30 7           IFI44L   7             107.   3.88 0.529  7.33 2.27e-13  4.22e-11
# â„¹ 897 more rows
# â„¹ 2 more variables: p_adj.glb &lt;dbl&gt;, contrast &lt;chr&gt;</code></pre>
</div>
</div>
</section>
<section id="write-differential-expression-results-to-tsv" class="level3" data-number="9.3.6">
<h3 data-number="9.3.6" class="anchored" data-anchor-id="write-differential-expression-results-to-tsv"><span class="header-section-number">9.3.6</span> Write differential expression results to tsv</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="fu">write_tsv</span>(df, <span class="st">"Skin_scRNA_PostVsPre3rd_DiffExp_results.tsv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="plot-number-of-significant-genes-for-all-clusters" class="level3" data-number="9.3.7">
<h3 data-number="9.3.7" class="anchored" data-anchor-id="plot-number-of-significant-genes-for-all-clusters"><span class="header-section-number">9.3.7</span> Plot number of significant genes for all clusters</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>all_subclusts <span class="ot">&lt;-</span> <span class="fu">enframe</span>(<span class="fu">unique</span>(merged.<span class="fl">18279.</span>skin.singlets<span class="sc">$</span>sub.cluster),<span class="at">name=</span><span class="cn">NULL</span>,<span class="at">value=</span><span class="st">"sub.cluster"</span>)</span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>df <span class="sc">%&gt;%</span></span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(p_adj.loc <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span></span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sub.cluster) <span class="sc">%&gt;%</span></span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">Up =</span> <span class="fu">count</span>(logFC <span class="sc">&gt;</span> <span class="dv">0</span>),<span class="at">Down =</span> (<span class="sc">-</span><span class="dv">1</span> <span class="sc">*</span> <span class="fu">count</span>(logFC <span class="sc">&lt;</span> <span class="dv">0</span>)),<span class="at">Total =</span> (<span class="fu">count</span>(logFC <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">+</span> <span class="fu">count</span>(logFC <span class="sc">&lt;</span> <span class="dv">0</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(all_subclusts, ., <span class="at">by=</span><span class="st">"sub.cluster"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Up =</span> <span class="fu">replace_na</span>(Up, <span class="dv">0</span>), <span class="at">Down =</span> <span class="fu">replace_na</span>(Down, <span class="dv">0</span>), <span class="at">Total =</span> <span class="fu">replace_na</span>(Total, <span class="dv">0</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">c</span>(Up,Down), <span class="at">names_to=</span><span class="st">"Direction"</span>, <span class="at">values_to=</span><span class="st">"NumberDE"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb12-9"><a href="#cb12-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">reorder</span>(sub.cluster,<span class="sc">-</span>Total), <span class="at">y =</span> NumberDE, <span class="at">fill =</span> Direction)) <span class="sc">+</span></span>
<span id="cb12-10"><a href="#cb12-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>() <span class="sc">+</span></span>
<span id="cb12-11"><a href="#cb12-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb12-12"><a href="#cb12-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">"Sub-cluster"</span>) <span class="sc">+</span></span>
<span id="cb12-13"><a href="#cb12-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"Number of Differentially Expressed Genes</span><span class="sc">\n</span><span class="st">p_adj.loc &lt; 0.05"</span>) <span class="sc">+</span></span>
<span id="cb12-14"><a href="#cb12-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">90</span>, <span class="at">vjust =</span> <span class="fl">0.5</span>, <span class="at">hjust=</span><span class="dv">1</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Skin_scRNA_Part9_files/figure-html/unnamed-chunk-9-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="get-session-info" class="level2" data-number="9.4">
<h2 data-number="9.4" class="anchored" data-anchor-id="get-session-info"><span class="header-section-number">9.4</span> Get session info</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>R version 4.3.1 (2023-06-16)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Rocky Linux 8.10 (Green Obsidian)

Matrix products: default
BLAS/LAPACK: /usr/lib64/libopenblasp-r0.3.15.so;  LAPACK version 3.9.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

time zone: America/New_York
tzcode source: system (glibc)

attached base packages:
[1] stats4    grid      stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] gprofiler2_0.2.2            scuttle_1.10.3             
 [3] DESeq2_1.40.2               SingleCellExperiment_1.22.0
 [5] SummarizedExperiment_1.30.2 Biobase_2.60.0             
 [7] GenomicRanges_1.52.1        GenomeInfoDb_1.36.4        
 [9] IRanges_2.34.1              S4Vectors_0.38.2           
[11] BiocGenerics_0.46.0         MatrixGenerics_1.12.3      
[13] matrixStats_1.2.0           edgeR_3.42.4               
[15] limma_3.56.2                muscat_1.14.0              
[17] circlize_0.4.15             ComplexHeatmap_2.16.0      
[19] msigdbr_7.5.1               paletteer_1.6.0            
[21] patchwork_1.3.0             lubridate_1.9.3            
[23] forcats_1.0.0               stringr_1.5.1              
[25] dplyr_1.1.4                 purrr_1.0.2                
[27] readr_2.1.5                 tidyr_1.3.1                
[29] tibble_3.2.1                ggplot2_3.4.4              
[31] tidyverse_2.0.0             Seurat_5.1.0               
[33] SeuratObject_5.0.2          sp_2.1-3                   
[35] presto_1.0.0                data.table_1.15.0          
[37] Rcpp_1.0.12                

loaded via a namespace (and not attached):
  [1] spatstat.sparse_3.0-3     bitops_1.0-7             
  [3] httr_1.4.7                RColorBrewer_1.1-3       
  [5] doParallel_1.0.17         numDeriv_2016.8-1.1      
  [7] backports_1.4.1           tools_4.3.1              
  [9] sctransform_0.4.1         utf8_1.2.4               
 [11] R6_2.5.1                  lazyeval_0.2.2           
 [13] uwot_0.1.16               mgcv_1.9-1               
 [15] GetoptLong_1.0.5          withr_3.0.0              
 [17] prettyunits_1.2.0         gridExtra_2.3            
 [19] SeuratWrappers_0.3.19     progressr_0.14.0         
 [21] cli_3.6.2                 spatstat.explore_3.2-6   
 [23] fastDummies_1.7.3         sandwich_3.1-0           
 [25] labeling_0.4.3            mvtnorm_1.2-4            
 [27] spatstat.data_3.0-4       blme_1.0-5               
 [29] ggridges_0.5.6            pbapply_1.7-2            
 [31] R.utils_2.12.3            scater_1.28.0            
 [33] parallelly_1.37.0         generics_0.1.3           
 [35] shape_1.4.6               vroom_1.6.5              
 [37] gtools_3.9.5              ica_1.0-3                
 [39] spatstat.random_3.2-2     Matrix_1.6-4             
 [41] ggbeeswarm_0.7.2          fansi_1.0.6              
 [43] abind_1.4-5               R.methodsS3_1.8.2        
 [45] lifecycle_1.0.4           multcomp_1.4-25          
 [47] yaml_2.3.8                gplots_3.1.3.1           
 [49] Rtsne_0.17                promises_1.2.1           
 [51] crayon_1.5.2              miniUI_0.1.1.1           
 [53] lattice_0.22-5            beachmat_2.16.0          
 [55] cowplot_1.1.3             pillar_1.9.0             
 [57] knitr_1.45                rjson_0.2.21             
 [59] boot_1.3-29               estimability_1.4.1       
 [61] future.apply_1.11.1       codetools_0.2-19         
 [63] leiden_0.4.3.1            glue_1.7.0               
 [65] remotes_2.4.2.1           vctrs_0.6.5              
 [67] png_0.1-8                 spam_2.10-0              
 [69] Rdpack_2.6                gtable_0.3.4             
 [71] rematch2_2.1.2            xfun_0.42                
 [73] rbibutils_2.2.16          S4Arrays_1.2.0           
 [75] mime_0.12                 coda_0.19-4.1            
 [77] reformulas_0.4.0          survival_3.5-8           
 [79] iterators_1.0.14          ellipsis_0.3.2           
 [81] fitdistrplus_1.1-11       TH.data_1.1-2            
 [83] ROCR_1.0-11               nlme_3.1-164             
 [85] pbkrtest_0.5.2            bit64_4.0.5              
 [87] EnvStats_2.8.1            progress_1.2.3           
 [89] RcppAnnoy_0.0.22          TMB_1.9.10               
 [91] irlba_2.3.5.1             vipor_0.4.7              
 [93] KernSmooth_2.23-22        colorspace_2.1-0         
 [95] nnet_7.3-19               tidyselect_1.2.0         
 [97] emmeans_1.10.0            bit_4.0.5                
 [99] compiler_4.3.1            BiocNeighbors_1.18.0     
[101] DelayedArray_0.26.7       plotly_4.10.4            
[103] caTools_1.18.2            scales_1.3.0             
[105] remaCor_0.0.18            lmtest_0.9-40            
[107] digest_0.6.34             goftest_1.2-3            
[109] spatstat.utils_3.0-4      minqa_1.2.6              
[111] variancePartition_1.30.2  rmarkdown_2.25           
[113] aod_1.3.3                 RhpcBLASctl_0.23-42      
[115] XVector_0.40.0            htmltools_0.5.7          
[117] pkgconfig_2.0.3           lme4_1.1-35.1            
[119] sparseMatrixStats_1.12.2  fastmap_1.1.1            
[121] rlang_1.1.3               GlobalOptions_0.1.2      
[123] htmlwidgets_1.6.4         shiny_1.8.0              
[125] DelayedMatrixStats_1.22.6 farver_2.1.1             
[127] zoo_1.8-12                jsonlite_1.8.8           
[129] BiocParallel_1.34.2       R.oo_1.26.0              
[131] BiocSingular_1.16.0       RCurl_1.98-1.14          
[133] magrittr_2.0.3            modeltools_0.2-23        
[135] GenomeInfoDbData_1.2.10   dotCall64_1.1-1          
[137] munsell_0.5.0             viridis_0.6.5            
[139] babelgene_22.9            reticulate_1.35.0        
[141] stringi_1.8.3             zlibbioc_1.46.0          
[143] MASS_7.3-60.0.1           flexmix_2.3-19           
[145] plyr_1.8.9                parallel_4.3.1           
[147] listenv_0.9.1             ggrepel_0.9.5            
[149] deldir_2.0-2              splines_4.3.1            
[151] tensor_1.5                hms_1.1.3                
[153] locfit_1.5-9.8            igraph_2.0.2             
[155] spatstat.geom_3.2-8       RcppHNSW_0.6.0           
[157] ScaledMatrix_1.8.1        reshape2_1.4.4           
[159] evaluate_0.23             BiocManager_1.30.22      
[161] nloptr_2.0.3              tzdb_0.4.0               
[163] foreach_1.5.2             httpuv_1.6.14            
[165] RANN_2.6.1                polyclip_1.10-6          
[167] future_1.33.1             clue_0.3-65              
[169] scattermore_1.2           rsvd_1.0.5               
[171] broom_1.0.5               xtable_1.8-4             
[173] RSpectra_0.16-1           later_1.3.2              
[175] viridisLite_0.4.2         lmerTest_3.1-3           
[177] glmmTMB_1.1.10            beeswarm_0.4.0           
[179] cluster_2.1.6             timechange_0.3.0         
[181] globals_0.16.2           </code></pre>
</div>
</div>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "î§‹";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="./Skin_scRNA_Part8.html" class="pagination-link" aria-label="Skin: Classification of CD4 and CD8 T cell states within T/NK lineage">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">8</span>&nbsp; <span class="chapter-title">Skin: Classification of CD4 and CD8 T cell states within T/NK lineage</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="./Skin_scRNA_Part10.html" class="pagination-link" aria-label="Skin: Deep phenotyping, cluster annotation, and exploration of Post vs Pre 3rd vaccination differences in cell proportions and expression">
        <span class="nav-page-text"><span class="chapter-number">10</span>&nbsp; <span class="chapter-title">Skin: Deep phenotyping, cluster annotation, and exploration of Post vs Pre 3rd vaccination differences in cell proportions and expression</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->




</body></html>