# IFNg: QC & Filtering {#sec-ifng-qc-filtering}

## Set up Seurat workspace

```{r }
#| warning: false
#| error: false

# Load libraries
library(data.table)
library(sctransform)
library(Seurat)
library(tidyverse)
library(miQC)
library(SeuratWrappers)
library(fishpond)
library(Matrix)
library(patchwork)
library(alevinQC)
library(harmony)
library(scDblFinder)

# Set global options for Seurat v5 objects
options(Seurat.object.assay.version = 'v5')
```

## Load alevin-fry output
Using `U+S+A` counts via `snRNA`

```{r}
#| cache: true
P108_IFNg_PostVax_5mgIpi_RNA <- loadFry(fryDir =  "/jsimonlab/projects/Wu/Melanoma_scRNA_Eryn/alevinfry_072023/P108_IFNg_PostVax_5mgIpi_RNA_alevin_quant_crlikeem/", outputFormat = "snRNA")
P103_IFNg_PostVax_2.5mgIpi_RNA <- loadFry(fryDir =  "/jsimonlab/projects/Wu/Melanoma_scRNA_Eryn/alevinfry_072023/P103_IFNg_PostVax_2.5mgIpi_RNA_alevin_quant_crlikeem/", outputFormat = "snRNA")
P101_IFNg_PostVax_2.5mgIpi_RNA <- loadFry(fryDir =  "/jsimonlab/projects/Wu/Melanoma_scRNA_Eryn/alevinfry_072023/P101_IFNg_PostVax_2.5mgIpi_RNA_alevin_quant_crlikeem/", outputFormat = "snRNA")
P104_IFNg_PostVax_2.5mgIpi_RNA <- loadFry(fryDir =  "/jsimonlab/projects/Wu/Melanoma_scRNA_Eryn/alevinfry_072023/P104_IFNg_PostVax_2.5mgIpi_RNA_alevin_quant_crlikeem/", outputFormat = "snRNA")
```

## Convert gene IDs to gene symbols, collapse transcripts to gene symbols

```{r}
tx2gene <- read.table("/jsimonlab/genomes/hg38/gencode.v43.annotation_splici/gene_id_to_name.tsv",header=F,sep="\t",col.names=c("tx","gene"))

# Use rownames of first object to apply to all
exp.txId <- rownames(P108_IFNg_PostVax_5mgIpi_RNA)
exp.geneId <- as.vector(tx2gene$gene[match(exp.txId, tx2gene$tx)])
exp.tx.grp <- t(sparse.model.matrix(~ 0 + exp.geneId))

P108_IFNg_PostVax_5mgIpi_RNA.summarized <- exp.tx.grp %*% counts(P108_IFNg_PostVax_5mgIpi_RNA)
rownames(P108_IFNg_PostVax_5mgIpi_RNA.summarized) <- rownames(P108_IFNg_PostVax_5mgIpi_RNA.summarized) %>% str_replace_all(".+.geneId","")

P103_IFNg_PostVax_2.5mgIpi_RNA.summarized <- exp.tx.grp %*% counts(P103_IFNg_PostVax_2.5mgIpi_RNA)
rownames(P103_IFNg_PostVax_2.5mgIpi_RNA.summarized) <- rownames(P103_IFNg_PostVax_2.5mgIpi_RNA.summarized) %>% str_replace_all(".+.geneId","")

P101_IFNg_PostVax_2.5mgIpi_RNA.summarized <- exp.tx.grp %*% counts(P101_IFNg_PostVax_2.5mgIpi_RNA)
rownames(P101_IFNg_PostVax_2.5mgIpi_RNA.summarized) <- rownames(P101_IFNg_PostVax_2.5mgIpi_RNA.summarized) %>% str_replace_all(".+.geneId","")

P104_IFNg_PostVax_2.5mgIpi_RNA.summarized <- exp.tx.grp %*% counts(P104_IFNg_PostVax_2.5mgIpi_RNA)
rownames(P104_IFNg_PostVax_2.5mgIpi_RNA.summarized) <- rownames(P104_IFNg_PostVax_2.5mgIpi_RNA.summarized) %>% str_replace_all(".+.geneId","")

```

## Create Seurat objects

```{r}
P108_IFNg_PostVax_5mgIpi_RNA.seurat <- CreateSeuratObject(P108_IFNg_PostVax_5mgIpi_RNA.summarized)
P103_IFNg_PostVax_2.5mgIpi_RNA.seurat <- CreateSeuratObject(P103_IFNg_PostVax_2.5mgIpi_RNA.summarized)
P101_IFNg_PostVax_2.5mgIpi_RNA.seurat <- CreateSeuratObject(P101_IFNg_PostVax_2.5mgIpi_RNA.summarized)
P104_IFNg_PostVax_2.5mgIpi_RNA.seurat <- CreateSeuratObject(P104_IFNg_PostVax_2.5mgIpi_RNA.summarized)

```

## Add sample name prefix to CBs

```{r}
P108_IFNg_PostVax_5mgIpi_RNA.seurat <- RenameCells(P108_IFNg_PostVax_5mgIpi_RNA.seurat,add.cell.id = "P108_IFNg_PostVax_5mgIpi_RNA")
P103_IFNg_PostVax_2.5mgIpi_RNA.seurat <- RenameCells(P103_IFNg_PostVax_2.5mgIpi_RNA.seurat,add.cell.id = "P103_IFNg_PostVax_2.5mgIpi_RNA")
P101_IFNg_PostVax_2.5mgIpi_RNA.seurat <- RenameCells(P101_IFNg_PostVax_2.5mgIpi_RNA.seurat,add.cell.id = "P101_IFNg_PostVax_2.5mgIpi_RNA")
P104_IFNg_PostVax_2.5mgIpi_RNA.seurat <- RenameCells(P104_IFNg_PostVax_2.5mgIpi_RNA.seurat,add.cell.id = "P104_IFNg_PostVax_2.5mgIpi_RNA")
```

## Merge objects

```{r}
merged.18279 <- merge(x = P108_IFNg_PostVax_5mgIpi_RNA.seurat, y=c(P103_IFNg_PostVax_2.5mgIpi_RNA.seurat, P101_IFNg_PostVax_2.5mgIpi_RNA.seurat, P104_IFNg_PostVax_2.5mgIpi_RNA.seurat))
dim(merged.18279)
```

## QC filter

Use relatively loose initial filters

```{r}
merged.18279 <- subset(merged.18279, subset = nCount_RNA > 1000 & nFeature_RNA > 500)
dim(merged.18279)

merged.18279 <- PercentageFeatureSet(merged.18279, pattern = "^MT-", col.name = "percent.mt")
merged.18279 <- RunMiQC(merged.18279, 
                        percent.mt = "percent.mt", 
                        nFeature_RNA = "nFeature_RNA", 
                        posterior.cutoff = 0.7, 
                        model.slot = "flexmix_model")

merged.18279 <- subset(merged.18279, miQC.keep == "keep")

dim(merged.18279)

data.frame(table(str_replace_all(colnames(merged.18279),"_RNA_.+","")))
```

## Add meta data

```{r}
patient <- str_split_i(colnames(merged.18279),"_",1)
site <- str_split_i(colnames(merged.18279),"_",2)
timepoint <- str_split_i(colnames(merged.18279),"_",3)
IpiCohort <- str_split_i(colnames(merged.18279),"_",4)
assay <- str_split_i(colnames(merged.18279),"_",5)
barcode <- str_split_i(colnames(merged.18279),"_",6)
sample <- paste0(patient,"_",site,"_",timepoint,"_",IpiCohort,"_",assay)

merged.18279 <- AddMetaData(merged.18279,patient,col.name="Patient")
merged.18279 <- AddMetaData(merged.18279,site,col.name="Site")
merged.18279 <- AddMetaData(merged.18279,timepoint,col.name="Timepoint")
merged.18279 <- AddMetaData(merged.18279,IpiCohort,col.name="IpiCohort")
merged.18279 <- AddMetaData(merged.18279,assay,col.name="Assay")
merged.18279 <- AddMetaData(merged.18279,sample,col.name="Sample")
```

## Join then re-split RNA counts layers by Sample

```{r}
merged.18279[['RNA']] <- JoinLayers(merged.18279[['RNA']])
merged.18279[["RNA"]] <- split(merged.18279[["RNA"]], f = merged.18279$Sample)
```

## Save merged object

```{r}
saveRDS(merged.18279,"IFNg_scRNA_Part1.rds")
```

## Get session info

```{r}
sessionInfo()
```
